---
title: "R Notebook for CEBPE project"
output:
  pdf_document: default
  html_notebook: default
editor_options:
  chunk_output_type: console
---

```{r global options, include = FALSE}
knitr::opts_chunk$set(echo=T, include = T, warning=FALSE, message=FALSE, tidy = TRUE, tidy.opts = list(comment = FALSE))
```

```{r ref.label=knitr::all_labels(), echo = T, eval = F}
```

# ```{r cachecontrol, cache = TRUE, cache.extra = tools::md5sum(expr_data_norm)}
# knitr::opts_chunk$set(cache.rebuild = TRUE)
# ```

#### Load libraries
```{r setup}
PWD="/home/pundhir/project/chip-seq-analysis/analysis_cebpe"
library(preprocessCore)
library(edgeR)
library(pheatmap)
library(biomaRt)
library(reshape2)
library(plyr)
library(ggplot2)
library(ggpubr)
library(doBy)
library(gridExtra)
library(ComplexHeatmap)
library(grDevices)
library(Cairo)
library(DESeq2)
library(Vennerable)
library(dplyr)
library(Hmisc)
library(rlist)
#mart = useMart(host = "grch37.ensembl.org",  biomart = "ENSEMBL_MART_ENSEMBL", dataset = "mmusculus_gene_ensembl")
library(caret)
library("psych")
library(RUVSeq)
source("~/software/myScripts/R-scripts/FUNCTIONS.R")
theme_set(theme_classic2() + theme(legend.title = element_blank()))
options(scipen=10, digits = 6)
```

#### read gene expression
```{r}
expr_data_all <- read.table("/data/nrapin/Kim/GENE_EXPRESSION.MATRIX.cebpa.cebpe", header=T)

expr_data_all <- expr_data_all[,c(1, c(grep("^lsk_", colnames(expr_data_all)), 
                                       grep("^pregm_", colnames(expr_data_all)),
                                       grep("^gmp_", colnames(expr_data_all)),
                                       grep("^gmpG_.*wt", colnames(expr_data_all)),
                                       grep("^gmpG_ko", colnames(expr_data_all)),
                                       grep("^gmpM_.*wt", colnames(expr_data_all)),
                                       grep("^epm_wt", colnames(expr_data_all)),
                                       grep("^epm_ko", colnames(expr_data_all)),
                                       grep("^pmo1_", colnames(expr_data_all)),
                                       grep("^pmo2_", colnames(expr_data_all)),
                                       grep("^mo1_", colnames(expr_data_all)),
                                       grep("^mo2_", colnames(expr_data_all)),
                                       grep("^monocytes", colnames(expr_data_all)),
                                       grep("^lpm_wt", colnames(expr_data_all)),
                                       grep("^lpm_ko", colnames(expr_data_all)),
                                       grep("^my", colnames(expr_data_all)),
                                       grep("^mmlike_wt", colnames(expr_data_all)),
                                       grep("^mmlike_ko", colnames(expr_data_all)),
                                       grep("^mm_wt", colnames(expr_data_all)),
                                       grep("^mm_ko", colnames(expr_data_all)),
                                       grep("^bc_wt", colnames(expr_data_all)),
                                       grep("^pmn_wt", colnames(expr_data_all)),
                                       grep("^granulocytes", colnames(expr_data_all)))
)]
```

#### identify robust samples
```{r}
##-----------------------------------------##
## based on correlation between replicates
##-----------------------------------------##

# APPROACH A: top three good samples
# expr_data_cor <- lapply(unique(gsub("_[Rr]+ep.*", "", colnames(expr_data_all[,c(2:ncol(expr_data_all))]))), function(x) { y <- cor(expr_data_all[,grep(sprintf("^%s", x), colnames(expr_data_all))]); y[lower.tri(y)] <- NA;  return(y); })
# robust_samples <- unlist(lapply(expr_data_cor, function(x) { y <- subset(as.data.frame(as.table(x)), Freq>0.95 & Freq<1); y <- y[order(-y$Freq),]; unique(as.vector(unlist(t(y[,c(1:2)]))))[1:3] }))
# # y <- subset(as.data.frame(as.table(expr_data_cor[[26]])), Freq>0.95 & Freq<1); y <- y[order(-y$Freq),]; y; unique(as.vector(unlist(t(y[,c(1:2)]))))[1:3]

# APPROACH B: all good samples 
expr_data_cor <- lapply(unique(gsub("_[Rr]+ep.*", "", colnames(expr_data_all[,c(2:ncol(expr_data_all))]))), function(x) { y <- cor(expr_data_all[,grep(sprintf("^%s", x), colnames(expr_data_all))]); return(y); })

names(expr_data_cor) <- unlist(lapply(expr_data_cor, function(x) gsub("_[rR]ep.*", "", colnames(x)[1])))

robust_samples <- unlist(lapply(expr_data_cor, function(y) sort(unique(as.vector(unlist(apply(y, 1, function(x) names(x[which(x>0.95 & x<1)]))))))))
#length(robust_samples) #106

bad_samples <- unlist(lapply(expr_data_cor, function(y) sort(unique(as.vector(unlist(apply(y, 1, function(x) names(x[which(x<0.95 & x<1)]))))))))
bad_samples <- bad_samples[!bad_samples %in% robust_samples]
#length(bad_samples) #14

expr_data <- cbind(expr_data_all$GeneID, expr_data_all[,robust_samples])
colnames(expr_data) <- c("GeneID", colnames(expr_data[,c(2:ncol(expr_data))]))

##-----------------------------------------##
## based on manual annotation of hiercarchical clustering tree of individual replicates
##-----------------------------------------##

# tree_all <- plotCustomPCA(expr_data[,c(29:ncol(expr_data))], ncol(expr_data[,c(29:ncol(expr_data))]))
# plot(tree_all)

expr_data_norm_with_replicates <- cbind(expr_data$GeneID, as.data.frame(normalize.quantiles(do.call(cbind, lapply(unique(gsub("_[Rr]+ep.*", "", colnames(expr_data[,c(2:ncol(expr_data))]))), function(x) (rle_normalize(expr_data[,grep(sprintf("^%s", x), colnames(expr_data))], "none")))))))
colnames(expr_data_norm_with_replicates) <- colnames(expr_data)
sample_info <- as.data.frame(cbind(gsub("_cebpa", "", gsub("_[0-9]+$", "", gsub("_[rR]+ep[0-9]+", "", colnames(expr_data[,c(2:ncol(expr_data))])))), ifelse(grepl("cebpa", colnames(expr_data))[2:ncol(expr_data)], 1, 2)))

## remove batch effects
colnames(sample_info) <- c("sample", "batch")
design <- model.matrix(~ 0 + sample, data=sample_info)
expr_data_norm_with_replicates <- cbind(expr_data_norm_with_replicates$GeneID, as.data.frame(removeBatchEffect(as.matrix(expr_data_norm_with_replicates[,c(2:ncol(expr_data_norm_with_replicates))]), sample_info$batch, design=design)))
colnames(expr_data_norm_with_replicates) <- colnames(expr_data)

pdf(sprintf("%s/pca_plot_manual_annotation.pdf", PWD), width=15, height=10)
tree_with_replicates <- plotCustomPCA(expr_data_norm_with_replicates[,grep("wt", colnames(expr_data_norm_with_replicates))], ncol(expr_data_norm_with_replicates[,grep("wt", colnames(expr_data_norm_with_replicates))]))
plot(tree_with_replicates)
dev.off()

## add least BAD samples, if less than three replicates are available per cell type
# replicate_info <- as.data.frame(table(gsub("_[rR]ep.*", "", colnames(expr_data_norm_with_replicates)[2:ncol(expr_data_norm_with_replicates)])))
# colnames(replicate_info) <- c("sample", "replicates")
# replicate_info[which(replicate_info$replicates<3),]$sample
# 
# test <- lapply(which(names(expr_data_cor) %in% replicate_info[which(replicate_info$replicates<3),]$sample), function(x) expr_data_cor[[x]])

##-----------------------------------------##
## remove BAD + EXTRA replicates from ROBUST samples based on manually annotation (see tree in pca_plot_to_filter_samples.pdf)
##-----------------------------------------##

## we want three replicates per cell type
## epm_wt, gmpM_cebpa_wt, mmlike_wt, mo1_wt, pmo1_wt, pmo2_wt, pregm_cebpa_ko
bad_samples <- c(bad_samples, "epm_wt_rep9", "gmpM_cebpa_wt_Rep2", "gmpG_wt_rep9_1", "gmpG_wt_rep9", "gmpG_wt_rep3", "pregm_cebpa_ko_Rep3",
                 "epm_wt_rep3", "pmo1_wt_rep3", "pmo1_wt_rep9", "pmo2_wt_rep4", "pmo2_wt_rep9", "mo1_wt_rep9_1", "mo1_wt_rep9",
                 "mo2_wt_rep9", "mmlike_wt_rep4", "mmlike_wt_rep1", "lpm_wt_rep13")
# length(bad_samples) #14+17
extra_replicates <- c("gmpG_ko_rep6", "gmpM_wt_rep9", "epm_ko_rep5", "mmlike_ko_rep6", "mm_ko_rep5", "mm_wt_rep1", 
                      "mm_wt_rep4", "my_wt_rep9", "lpm_wt_rep4", "lpm_ko_rep5", "lpm_ko_rep7", "pmn_wt_rep10", "pmn_wt_rep3", "bc_wt_rep9")
# length(extra_replicates) #14
robust_samples <- robust_samples[!robust_samples %in% c(bad_samples, extra_replicates)]
# length(robust_samples) #75

##-----------------------------------------##
## select only GOOD samples
##-----------------------------------------##

all_samples <- colnames(expr_data_all[,c(2:ncol(expr_data_all))])
info_samples <- merge(as.data.frame(table((gsub("_[Rr]+ep.*", "", all_samples)))), as.data.frame(table((gsub("_[Rr]+ep.*", "", robust_samples)))), by="Var1")

expr_data <- cbind(expr_data$GeneID, expr_data_all[,robust_samples])
colnames(expr_data) <- c("GeneID", colnames(expr_data[,c(2:ncol(expr_data))]))
```

#### analysis on individual replicates
```{r}
expr_data_norm_with_replicates <- cbind(expr_data$GeneID, as.data.frame(normalize.quantiles(do.call(cbind, lapply(unique(gsub("_[Rr]+ep.*", "", colnames(expr_data[,c(2:ncol(expr_data))]))), function(x) (rle_normalize(expr_data[,grep(sprintf("^%s", x), colnames(expr_data))], "none")))))))
colnames(expr_data_norm_with_replicates) <- colnames(expr_data)
sample_info <- as.data.frame(cbind(gsub("_cebpa", "", gsub("_[0-9]+$", "", gsub("_[rR]+ep[0-9]+", "", colnames(expr_data[,c(2:ncol(expr_data))])))), ifelse(grepl("cebpa", colnames(expr_data))[2:ncol(expr_data)], 1, 2)))

## remove batch effects
colnames(sample_info) <- c("sample", "batch")
design <- model.matrix(~ 0 + sample, data=sample_info)
expr_data_norm_with_replicates <- cbind(expr_data_norm_with_replicates$GeneID, as.data.frame(removeBatchEffect(as.matrix(expr_data_norm_with_replicates[,c(2:ncol(expr_data_norm_with_replicates))]), sample_info$batch, design=design)))
colnames(expr_data_norm_with_replicates) <- colnames(expr_data)
```

#### analysis on mean of replicates
```{r}
expr_data_norm <- cbind(expr_data$GeneID, as.data.frame(normalize.quantiles(do.call(cbind, lapply(unique(gsub("_[Rr]+ep.*", "", colnames(expr_data[,c(2:ncol(expr_data))]))), function(x) rowMeans(rle_normalize(expr_data[,grep(sprintf("^%s", x), colnames(expr_data))], "none")))))))
colnames(expr_data_norm) <- c("GeneID", unique(gsub("_[Rr]+ep.*", "", colnames(expr_data[,c(2:ncol(expr_data))]))))
sample_info <- as.data.frame(cbind(gsub("_cebpa", "", gsub("_[0-9]+$", "", gsub("_[rR]+ep[0-9]+", "", colnames(expr_data_norm[,c(2:ncol(expr_data_norm))])))), ifelse(grepl("cebpa", colnames(expr_data_norm))[2:ncol(expr_data_norm)], 1, 2)))

## remove batch effects
colnames(sample_info) <- c("sample", "batch")
design <- model.matrix(~ 0 + sample, data=sample_info)
expr_data_norm <- cbind(expr_data_norm$GeneID, as.data.frame(removeBatchEffect(as.matrix(expr_data_norm[,c(2:ncol(expr_data_norm))]), sample_info$batch, design=design)))
```

#### plot PCA and heatmaps
```{r}
pdf(sprintf("%s/pdf/pca_plot.pdf", PWD), width=10)
mat <- expr_data_norm_with_replicates[,c(2:4,8:10,13:44,48:56,62:73)]

tree_with_replicates <- plotCustomPCA(mat, ncol(mat), arg_cex = 1.5, customGroups = 1)
#pheatmap(cor((expr_data_norm_with_replicates[,c(2:ncol(expr_data_norm_with_replicates))])), color=col, border_color = NA, scale="none", cluster_rows = T, cluster_cols = T, legend=T, show_rownames = T, show_colnames = F, display_numbers = F)
plot(tree_with_replicates)

tree_with_replicates_only_wt <- plotCustomPCA(mat[,grep("wt", colnames(mat))], ncol(mat[,grep("wt", colnames(mat))]), arg_cex = 1.5, customGroups = 1)
plot(tree_with_replicates_only_wt)

mat <- expr_data_norm[,c(2,4,6:17,19:27)]
tree <- plotCustomPCA(mat, ncol(mat), arg_cex = 1.5, customGroups = 1)
dev.off()
```

#### write to file information on ROBUST, BAD and EXTRA samples
```{r}
## sample statistics
## total: 120; bad: 31; extra: 14; analyzed: 75 (total-bad-extra)

expr_data_all_marked <- expr_data_all

colnames(expr_data_all_marked) <- unlist(lapply(colnames(expr_data_all_marked), function(x) { x <- ifelse(x %in% bad_samples, sprintf("%s#BAD", x), x); ifelse(x %in% extra_replicates, sprintf("%s#EXTRA", x), x); }))
# expr_data_all_marked[colnames(expr_data_all_marked) %in% colnames(expr_data_norm_with_replicates)] <- expr_data_norm_with_replicates

write.table(expr_data_all_marked, "/home/pundhir/project/chip-seq-analysis/data/gene/cebpe/GENE_EXPRESSION.MATRIX.ALLSAMPLES", sep="\t", quote = F, row.names = F, col.names = T)

## normalized - merged replicates
dim(expr_data_norm)

## normalized - all replicates
dim(expr_data_norm_with_replicates)

## BAD samples as defined by Kim (all are excluded in the final samples, except "gmpG_wt_rep4", "gmpG_cebpa_wt_Rep1", "gmpG_cebpa_wt_Rep2", "gmpM_cebpa_wt_Rep1", "gmpM_cebpa_wt_Rep3" - leads to only replicates of gmpG_cebpa_wt)
bad_samples_kim <- c("gmpG_wt_rep4", "gmpG_cebpa_wt_Rep1", "gmpG_cebpa_wt_Rep2", "gmpM_cebpa_wt_Rep2", "gmpG_wt_rep3", "gmpG_wt_rep9",
                     "gmpG_wt_rep9_1", "epm_wt_rep3", "epm_wt_rep9", "gmpM_cebpa_wt_Rep1", "gmpM_cebpa_wt_Rep3", "gmpM_wt_rep9",
                     "pmo1_wt_rep9", "pmo1_wt_rep3", "pmo2_wt_rep4", "pmo2_wt_rep9", "mo2_wt_rep9", "mo1_wt_rep9", "mo1_wt_rep9_1",
                     "bc_wt_rep9", "pmn_wt_rep3", "pmn_wt_rep10", "mm_wt_rep4", "mm_wt_rep1", "mmlike_wt_rep4", "lpm_wt_rep13", "lpm_wt_rep4",
                     "my_wt_rep9")
bad_samples_kim[bad_samples_kim %in% colnames(expr_data_norm_with_replicates)]
```

#### define cluster based on NORMALIZED gene expression
```{r, cache=TRUE, cache.vars="KC_CEBPE_NORM", cache.comments=FALSE}
## normalized gene expression values (quantile normalized and batch corrected)
reqSamples <- c("lsk_cebpa_wt", "pregm_cebpa_wt", "gmp_cebpa_wt", "gmpG_wt", "epm_wt", "lpm_wt", "my_wt", "mm_wt", "bc_wt", "pmn_wt", "gmpM_wt", "pmo1_wt", "pmo2_wt", "mo1_wt", "mo2_wt", "gmpG_ko", "epm_ko", "lpm_ko", "mm_ko")

mat_norm <- expr_data_norm[,c(reqSamples)]
row.names(mat_norm) <- expr_data_norm$`expr_data_norm$GeneID`
setdiff(colnames(expr_data_norm), colnames(mat_norm))
mat_norm <- mat_norm[apply(mat_norm, 1, function(x) length(x[x>1]) >=2),]
mat_norm <- mat_norm[apply(mat_norm[,grep("wt", colnames(mat_norm))], 1, function(x) ifelse(sd(x, na.rm=TRUE)>0, T, F)),]
#cluster_info_matrix(mat_norm[,grep("wt", colnames(mat_norm))])
set.seed(1)
#KC_CEBPE_NORM <- cluster_matrix(mat_norm[,grep("wt", colnames(mat_norm))],15)
#KC_CEBPE_NORM$cluster <- factor(KC_CEBPE_NORM$cluster, levels=c(3,6,7,4,1,2,8,5), ordered = T)
#KC_CEBPE_NORM$cluster <- factor(KC_CEBPE_NORM$cluster, levels=c(4,3,7,6,9,2,5,8,10,1), ordered = T)
mat_norm$cluster <- sprintf("%d", as.numeric(KC_CEBPE_NORM$cluster))
mat_norm <- cbind(row.names(mat_norm), mat_norm)
colnames(mat_norm)[1] <- "gene"
mat_norm <- mat_norm[order(mat_norm$cluster),]
table(KC_CEBPE_NORM$cluster)
table(mat_norm$cluster)
unique(mat_norm$cluster)

## raw gene expression values
mat_raw <- cbind(expr_data$GeneID, as.data.frame((do.call(cbind, lapply(unique(gsub("_[Rr]+ep.*", "", colnames(expr_data[,c(2:ncol(expr_data))]))), function(x) rowMeans(rle_normalize(expr_data[,grep(sprintf("^%s", x), colnames(expr_data))], "none")))))))
colnames(mat_raw) <- c("GeneID", unique(gsub("_[Rr]+ep.*", "", colnames(expr_data[,c(2:ncol(expr_data))]))))
mat_raw <- merge(mat_raw, mat_norm[,c(1,ncol(mat_norm))], by.x="GeneID", by.y="gene")
mat_raw <- cbind(mat_raw$GeneID, mat_raw[,grep("wt", colnames(mat_raw))], mat_raw[,grep("ko", colnames(mat_raw))], mat_raw$cluster)
colnames(mat_raw)[c(1,29)] <- c("gene", "cluster")
mat_raw <- mat_raw[,colnames(mat_norm)]
mat_raw <- mat_raw[order(mat_raw$cluster),]
mat_raw$cluster <- as.character(mat_raw$cluster)
colnames(mat_norm)
colnames(mat_raw)

mat_raw <- mat_raw[match(mat_norm$gene, mat_raw$gene),]

colnames(mat_raw)[2:(ncol(mat_raw)-1)] <- sprintf("%s_raw", colnames(mat_raw)[2:(ncol(mat_raw)-1)])
colnames(mat_norm)[2:(ncol(mat_norm)-1)] <- sprintf("%s_norm", colnames(mat_norm)[2:(ncol(mat_norm)-1)])

table(mat_raw$cluster)
table(mat_norm$cluster)
```

#### define cluster based on RAW gene expression
```{r, cache=TRUE, cache.vars="KC_CEBPE_RAW", cache.comments=FALSE}
reqSamples <- c("lsk_cebpa_wt", "pregm_cebpa_wt", "gmp_cebpa_wt", "gmpG_wt", "epm_wt", "lpm_wt", "my_wt", "mm_wt", "bc_wt", "pmn_wt", "gmpM_wt", "pmo1_wt", "pmo2_wt", "mo1_wt", "mo2_wt", "gmpG_ko", "epm_ko", "lpm_ko", "mm_ko")

mat_raw <- cbind(expr_data$GeneID, as.data.frame((do.call(cbind, lapply(unique(gsub("_[Rr]+ep.*", "", colnames(expr_data[,c(2:ncol(expr_data))]))), function(x) rowMeans(rle_normalize(expr_data[,grep(sprintf("^%s", x), colnames(expr_data))], "none")))))))
colnames(mat_raw) <- c("gene", unique(gsub("_[Rr]+ep.*", "", colnames(expr_data[,c(2:ncol(expr_data))]))))
mat_raw <- mat_raw[,c(reqSamples)]
row.names(mat_raw) <- expr_data$GeneID
mat_raw <- mat_raw[apply(mat_raw, 1, function(x) length(x[x>1]) >=2),]
mat_raw <- mat_raw[apply(mat_raw[,grep("wt", colnames(mat_raw))], 1, function(x) ifelse(sd(x, na.rm=TRUE)>0, T, F)),]
#cluster_info_matrix(mat_raw[,grep("wt", colnames(mat_raw))])
set.seed(1)
# KC_CEBPE_RAW <- cluster_matrix(mat_raw[,grep("wt", colnames(mat_raw))],15)
mat_raw$cluster <- as.numeric(factor(KC_CEBPE_RAW$cluster, levels=c(5,1,3,6,13,12,15,9,2,7,4,14,10,11,8), ordered = T))
#mat_raw$cluster <- KC_CEBPE_RAW$cluster
mat_raw <- cbind(row.names(mat_raw), mat_raw)
colnames(mat_raw)[1] <- "gene"
mat_raw <- mat_raw[order(mat_raw$cluster),]
# mat_raw[which(mat_raw$cluster=="4" | mat_raw$cluster=="5" | mat_raw$cluster=="6"),]$cluster <- "4"
# mat_raw[which(mat_raw$cluster=="7" | mat_raw$cluster=="8"),]$cluster <- "7"
colnames(mat_norm)
colnames(mat_raw)
table(KC_CEBPE_RAW$cluster)
table(mat_raw$cluster)

mat_norm <- expr_data_norm[,c(reqSamples)]
row.names(mat_norm) <- expr_data_norm$`expr_data_norm$GeneID`
mat_norm <- cbind(row.names(mat_norm), mat_norm)
colnames(mat_norm)[1] <- "gene"
mat_norm <- merge(mat_norm[,c(1:20)], mat_raw[,c(1,21)], by.x="gene", by.y="gene")
mat_norm <- mat_norm[match(mat_raw$gene, mat_norm$gene),]

colnames(mat_raw)[2:(ncol(mat_raw)-1)] <- sprintf("%s_raw", colnames(mat_raw)[2:(ncol(mat_raw)-1)])
colnames(mat_norm)[2:(ncol(mat_norm)-1)] <- sprintf("%s_norm", colnames(mat_norm)[2:(ncol(mat_norm)-1)])

table(mat_norm$cluster)
table(mat_raw$cluster)
```

#### identify differentially expressed genes upon CEBPE KD
```{r}
## RAW expression for 75 samples selected for analysis
#colnames(expr_data_all_marked[,which(!grepl("BAD|EXTRA", colnames(expr_data_all_marked)))])
## NORMALIZED expression for 75 samples selected for analysis
#colnames(expr_data_norm_with_replicates)

df <- expr_data_all_marked[,which(!grepl("BAD|EXTRA", colnames(expr_data_all_marked)))]
df <- df[,c(1,unlist(lapply(reqSamples, function(x) grep(x, colnames(df)))))]

## gmpG (WT vs CEBPE KO)
mat <- as.matrix(df[,grep("gmpG_", colnames(df))])
rownames(mat) <- df$GeneID
mat <- mat[apply(mat, 1, function(x) length(x[x>5]) >= 2),]
colnames(mat)
plotCustomPCA(mat, ncol(mat))
res_gmpG <- matrix2deseq2(mat, "wt,wt,wt,ko,ko,ko")
res_gmpG$class <- "neutral"
res_gmpG[which(res_gmpG$padj<0.05 & res_gmpG$log2FoldChange<0),]$class <- "down"
res_gmpG[which(res_gmpG$padj<0.05 & res_gmpG$log2FoldChange>0),]$class <- "up"

## epm (WT vs CEBPE KO)
mat <- as.matrix(df[,grep("epm_", colnames(df))])
rownames(mat) <- df$GeneID
mat <- mat[apply(mat, 1, function(x) length(x[x>5]) >= 2),]
colnames(mat)
plotCustomPCA(mat, ncol(mat))
res_epm <- matrix2deseq2(mat, "wt,wt,ko,ko,ko")
res_epm$class <- "neutral"
res_epm[which(res_epm$padj<0.05 & res_epm$log2FoldChange<0),]$class <- "down"
res_epm[which(res_epm$padj<0.05 & res_epm$log2FoldChange>0),]$class <- "up"

## lpm (WT vs CEBPE KO)
mat <- as.matrix(df[,grep("lpm_", colnames(df))])
rownames(mat) <- df$GeneID
mat <- mat[apply(mat, 1, function(x) length(x[x>5]) >= 2),]
colnames(mat)
plotCustomPCA(mat, ncol(mat))
res_lpm <- matrix2deseq2(mat, "wt,wt,wt,ko,ko,ko")
res_lpm$class <- "neutral"
res_lpm[which(res_lpm$padj<0.05 & res_lpm$log2FoldChange<0),]$class <- "down"
res_lpm[which(res_lpm$padj<0.05 & res_lpm$log2FoldChange>0),]$class <- "up"

## mm-like (WT vs CEBPE KO; KIM don't want to include these samples anymore - )
# mat <- as.matrix(df[,grep("mmlike_", colnames(df))])
# rownames(mat) <- df$GeneID
# mat <- mat[apply(mat, 1, function(x) length(x[x>5]) >= 2),]
# colnames(mat)
# plotCustomPCA(mat, ncol(mat))
# res_mmlike <- matrix2deseq2(mat, "wt,wt,ko,ko,ko")
# res_mmlike$class <- "neutral"
# res_mmlike[which(res_mmlike$padj<0.05 & res_mmlike$log2FoldChange<0),]$class <- "down"
# res_mmlike[which(res_mmlike$padj<0.05 & res_mmlike$log2FoldChange>0),]$class <- "up"

## mm (WT vs CEBPE KO)
mat <- as.matrix(df[,grep("mm_", colnames(df))])
rownames(mat) <- df$GeneID
mat <- mat[apply(mat, 1, function(x) length(x[x>5]) >= 2),]
colnames(mat)
plotCustomPCA(mat, ncol(mat))
res_mm <- matrix2deseq2(mat, "wt,wt,wt,ko,ko,ko")
res_mm$class <- "neutral"
res_mm[which(res_mm$padj<0.05 & res_mm$log2FoldChange<0),]$class <- "down"
res_mm[which(res_mm$padj<0.05 & res_mm$log2FoldChange>0),]$class <- "up"

## overlap between DE genes
# input <- list(gmpG=rownames(res_gmpG[which(res_gmpG$class!="neutral"),]), epm=rownames(res_epm[which(res_epm$class!="neutral"),]), 
#               lpm=rownames(res_lpm[which(res_lpm$class!="neutral"),]), mmlike=rownames(res_mmlike[which(res_mmlike$class!="neutral"),]), 
#               mm=rownames(res_mm[which(res_mm$class!="neutral"),]))
input <- list(gmpG=rownames(res_gmpG[which(res_gmpG$class!="neutral"),]), epm=rownames(res_epm[which(res_epm$class!="neutral"),]),
              lpm=rownames(res_lpm[which(res_lpm$class!="neutral"),]), mm=rownames(res_mm[which(res_mm$class!="neutral"),]))
pdf("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/pdf/de_genes_venn.pdf")
plot(Venn(input))
dev.off()
#attr(gplots::venn(input, show.plot=FALSE), "intersections")
GENES_DE <- unique(unlist(attr(gplots::venn(input, show.plot=FALSE), "intersections")))
```

#### combine gene expression/cluster information with their proximal and distal regulatory complexity
```{r}
rp <- read.table("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/allGenes.RP")
colnames(rp) <- c("chr", "start", "end", "gene", "score", "strand", "ensemblId", 
                  "pu1_lsk_rep1", "pu1_lsk_rep2", "pu1_pregm_rep1", "pu1_pregm_rep2", "pu1_gmp_rep1", "pu1_gmp_rep2", 
                  "cebpa_lsk_rep1", "cebpa_lsk_rep2", "cebpa_pregm_rep1", "cebpa_pregm_rep2", "cebpa_gmp_rep1", "cebpa_gmp_rep2",
                  "cebpe_lpmmy_rep1", "cebpe_lpmmy_rep2", "cebpe_mm_rep1", "cebpe_mm_rep2", "cebpe_bc_rep1", "cebpe_bc_rep2", 
                  "cebpe_pmn_rep1", "cebpe_pmn_rep2", "e2f", "myc",
                  "pu1_lsk_rep1_dis", "pu1_lsk_rep2_dis", "pu1_pregm_rep1_dis", "pu1_pregm_rep2_dis", "pu1_gmp_rep1_dis", "pu1_gmp_rep2_dis", 
                  "cebpa_lsk_rep1_dis", "cebpa_lsk_rep2_dis", "cebpa_pregm_rep1_dis", "cebpa_pregm_rep2_dis", "cebpa_gmp_rep1_dis", "cebpa_gmp_rep2_dis",
                  "cebpe_lpmmy_rep1_dis", "cebpe_lpmmy_rep2_dis", "cebpe_mm_rep1_dis", "cebpe_mm_rep2_dis", "cebpe_bc_rep1_dis", "cebpe_bc_rep2_dis", 
                  "cebpe_pmn_rep1_dis", "cebpe_pmn_rep2_dis", "e2f_dis", "myc_dis",
                  "pu1_lsk_rep1_ws", "pu1_lsk_rep2_ws", "pu1_pregm_rep1_ws", "pu1_pregm_rep2_ws", "pu1_gmp_rep1_ws", "pu1_gmp_rep2_ws", 
                  "cebpa_lsk_rep1_ws", "cebpa_lsk_rep2_ws", "cebpa_pregm_rep1_ws", "cebpa_pregm_rep2_ws", "cebpa_gmp_rep1_ws", "cebpa_gmp_rep2_ws",
                  "cebpe_lpmmy_rep1_ws", "cebpe_lpmmy_rep2_ws", "cebpe_mm_rep1_ws", "cebpe_mm_rep2_ws", "cebpe_bc_rep1_ws", "cebpe_bc_rep2_ws", 
                  "cebpe_pmn_rep1_ws", "cebpe_pmn_rep2_ws", "e2f_ws", "myc_ws", "enhancer_count")
rp[is.na(rp)] <- 0

rp_mean <- cbind(rp[,c(1:7)],
                 as.data.frame(lapply(seq(8,26,2), function(i) rowMeans(rp[,c(i,i+1)]))), rp[,c(28)], rp[,c(29)],
                 as.data.frame(lapply(seq(30,48,2), function(i) rowMeans(rp[,c(i,i+1)]))), rp[,c(50)], rp[,c(51)],
                 as.data.frame(lapply(seq(52,70,2), function(i) rowMeans(rp[,c(i,i+1)]))), rp[,c(72)], rp[,c(73)],
                 rp[,c(74)])
colnames(rp_mean) <- c("chr", "start", "end", "gene", "score", "strand", "ensemblId", 
                       "pu1_lsk", "pu1_pregm", "pu1_gmp", "cebpa_lsk", "cebpa_pregm", "cebpa_gmp", 
                       "cebpe_lpmmy", "cebpe_mm", "cebpe_bc", "cebpe_pmn", "e2f", "myc",
                       "pu1_lsk_dis", "pu1_pregm_dis", "pu1_gmp_dis", "cebpa_lsk_dis", "cebpa_pregm_dis", "cebpa_gmp_dis", 
                       "cebpe_lpmmy_dis", "cebpe_mm_dis", "cebpe_bc_dis", "cebpe_pmn_dis", "e2f_dis", "myc_dis",
                       "pu1_lsk_ws", "pu1_pregm_ws", "pu1_gmp_ws", "cebpa_lsk_ws", "cebpa_pregm_ws", "cebpa_gmp_ws", 
                       "cebpe_lpmmy_ws", "cebpe_mm_ws", "cebpe_bc_ws", "cebpe_pmn_ws", "e2f_ws", "myc_ws", "enhancer_count")

## RAW EXPRESSION ONLY
# genes_cebpe <- merge(mat_raw, rp_mean, by.x="gene", by.y="gene")
#genes_cebpe <- genes_cebpe[,c(22:24,1,25:27,2:21,28:ncol(genes_cebpe))]

## NORMALIZED EXPRESSION ONLY
#genes_cebpe <- merge(mat_norm, rp_mean, by.x="gene", by.y="gene")
#genes_cebpe <- genes_cebpe[,c(22:24,1,25:27,2:21,28:ncol(genes_cebpe))]

## RAW AND NORMALIZED EXPRESSION
genes_cebpe <- merge(mat_raw, mat_norm[,c(1:ncol(mat_norm)-1)], by.x="gene", by.y="gene")
genes_cebpe <- merge(genes_cebpe, rp_mean, by.x="gene", by.y="gene")
genes_cebpe <- genes_cebpe[,c(41:43,1,44:46,2:40,47:ncol(genes_cebpe))]

genes_cebpe <- genes_cebpe[order(genes_cebpe$cluster),]
table(genes_cebpe$cluster)
class(genes_cebpe$cluster)
dim(genes_cebpe)
genes_cebpe$cluster <- as.factor(genes_cebpe$cluster)
#genes_cebpe$cluster <- factor(genes_cebpe$cluster, levels = test[order(-test$per_e2f_binding),]$cluster, ordered = T)

# test <- melt(table(genes_cebpe[which(genes_cebpe$e2f>0),]$cluster)/table(genes_cebpe$cluster))
# colnames(test) <- c("cluster", "per_e2f_binding")
# test$cluster <- as.factor(test$cluster)
# #test$cluster <- factor(test$cluster, levels=test[order(-test$per_e2f_binding),]$cluster, ordered = T)
# test$per_e2f_binding_distal <- melt(table(genes_cebpe[which(genes_cebpe$e2f_dis>0),]$cluster)/table(genes_cebpe$cluster))$value
# #impGenes <- read.table("/home/pundhir/project/chip-seq-analysis/data/gene/cebpe/important_genes")
# test$impGenes <- as.vector(table(genes_cebpe[which(genes_cebpe$gene %in% impGenes$V1),]$cluster))

##------------------------------------------------##
## add differential expression information
genes_cebpe <- merge(genes_cebpe, res_gmpG[,c("log2FoldChange","class","gene")], by.x="gene", by.y="gene", all.x=T)
colnames(genes_cebpe)[(ncol(genes_cebpe)-1):ncol(genes_cebpe)] <- c("log2FC_gmpG", "de_gmpG")
genes_cebpe <- merge(genes_cebpe, res_epm[,c("log2FoldChange","class","gene")], by.x="gene", by.y="gene", all.x=T)
colnames(genes_cebpe)[(ncol(genes_cebpe)-1):ncol(genes_cebpe)] <- c("log2FC_epm", "de_epm")
genes_cebpe <- merge(genes_cebpe, res_lpm[,c("log2FoldChange","class","gene")], by.x="gene", by.y="gene", all.x=T)
colnames(genes_cebpe)[(ncol(genes_cebpe)-1):ncol(genes_cebpe)] <- c("log2FC_lpm", "de_lpm")
# genes_cebpe <- merge(genes_cebpe, res_mmlike[,c("log2FoldChange","class","gene")], by.x="gene", by.y="gene", all.x=T)
# colnames(genes_cebpe)[(ncol(genes_cebpe)-1):ncol(genes_cebpe)] <- c("log2FC_mmlike", "de_mmlike")
genes_cebpe <- merge(genes_cebpe, res_mm[,c("log2FoldChange","class","gene")], by.x="gene", by.y="gene", all.x=T)
colnames(genes_cebpe)[(ncol(genes_cebpe)-1):ncol(genes_cebpe)] <- c("log2FC_mm", "de_mm")

genes_cebpe[,c("de_gmpG", "de_epm", "de_lpm", "de_mm")] <- apply(genes_cebpe[,c("de_gmpG", "de_epm", "de_lpm", "de_mm")], 2, function(x) 
  ifelse(is.na(x), "neutral", x))

genes_cebpe <- genes_cebpe[,c(2:4,1,5:ncol(genes_cebpe))]

genes_cebpe$de_count <- apply(genes_cebpe[,c("de_gmpG", "de_epm", "de_lpm", "de_mm")], 1, function(x) length(which(!grepl("neutral", x)==T)))

##------------------------------------------------##
## add genes of interest information
genesInterest <- read.table("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/genes_of_interest.txt")
colnames(genesInterest) <- c("gene", "funcClass")

genes_cebpe <- merge(genes_cebpe, genesInterest, by.x="gene", by.y="gene", all=T)
genes_cebpe <- genes_cebpe[,c(2:4,1,5:ncol(genes_cebpe))]
table(genes_cebpe$funcClass)

##------------------------------------------------##
## add TF binding information
rpClass <- read.table("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/allGenes.RPClass", header=T)

genes_cebpe <- merge(genes_cebpe, rpClass[,c(4,9:13,15:19)], by.x="gene", by.y="name")

## add gene cluster information based upon the stage at which they are differentially expressed
geneSet <- list()

geneSet[["A_epm"]] <- genes_cebpe[which((genes_cebpe$cluster==3 | genes_cebpe$cluster==4 | genes_cebpe$cluster==5) & 
                                    (genes_cebpe$de_lpm=="up" | genes_cebpe$de_mm=="up")),]
geneSet[["A_epm"]]$cluster_de <- "A_epm"

geneSet[["B_lpm"]] <- genes_cebpe[which((genes_cebpe$cluster==6) & (genes_cebpe$de_mm=="up")),]
geneSet[["B_lpm"]]$cluster_de <- "B_lpm"

geneSet[["C_gran"]] <- genes_cebpe[which((genes_cebpe$cluster==10 | genes_cebpe$cluster==11) & 
                                   (genes_cebpe$de_lpm=="down" | genes_cebpe$de_mm=="down")),]
geneSet[["C_gran"]]$cluster_de <- "C_gran"

geneSet[["D_gran_mono"]] <- genes_cebpe[which((genes_cebpe$cluster==12) &
                                    (genes_cebpe$de_lpm=="up" | genes_cebpe$de_mm=="up")),]
geneSet[["D_gran_mono"]]$cluster_de <- "D_gran_mono"

geneSet[["E_mono"]] <- genes_cebpe[which((genes_cebpe$cluster==14 | genes_cebpe$cluster==15) & 
                                           (genes_cebpe$de_lpm=="up" | genes_cebpe$de_mm=="up")),]
geneSet[["E_mono"]]$cluster_de <- "E_mono"

geneSet[["F_rest"]] <- genes_cebpe[!genes_cebpe$gene %in% do.call(rbind, geneSet)$gene,]
geneSet[["F_rest"]]$cluster_de <- "F_rest"

#plot(Venn(lapply(geneSet[1:5], function(x) x$gene)))
input <- attr(gplots::venn(lapply(geneSet[1:length(geneSet)], function(x) x$gene), show.plot=FALSE), "intersections")

genes_cebpe$cluster_de <- "none"
#for(i in grep(":", names(input), value=T, invert = T)) { genes_cebpe[genes_cebpe$gene %in% input[[i]],]$cluster_de <- i; }
for(i in names(input)) { genes_cebpe[genes_cebpe$gene %in% input[[i]],]$cluster_de <- i; }

##------------------------------------------------##
## add refined gene cluster information
genes_cebpe$cluster_deRefined <- "F_rest-neutral"
genes_cebpe[which((genes_cebpe$cluster_de!="F_rest") &
                    (genes_cebpe$cluster==3 | genes_cebpe$cluster==4 | genes_cebpe$cluster==5)),]$cluster_deRefined <- "A_epm-de"
genes_cebpe[which((genes_cebpe$cluster_de=="F_rest") &
                    (genes_cebpe$cluster==4 | genes_cebpe$cluster==4 | genes_cebpe$cluster==5)),]$cluster_deRefined <- "A_epm-neutral"
genes_cebpe[which((genes_cebpe$cluster_de!="F_rest") & 
                    (genes_cebpe$cluster==6)),]$cluster_deRefined <- "B_lpm-de"
genes_cebpe[which((genes_cebpe$cluster_de=="F_rest") & 
                    (genes_cebpe$cluster==6)),]$cluster_deRefined <- "B_lpm-neutral"
genes_cebpe[which((genes_cebpe$cluster_de!="F_rest") & 
                    (genes_cebpe$cluster==10 | genes_cebpe$cluster==11)),]$cluster_deRefined <- "C_gran-de"
genes_cebpe[which((genes_cebpe$cluster_de=="F_rest") & 
                    (genes_cebpe$cluster==10 | genes_cebpe$cluster==11)),]$cluster_deRefined <- "C_gran-neutral"
genes_cebpe[which((genes_cebpe$cluster_de!="F_rest") & 
                    (genes_cebpe$cluster==12)),]$cluster_deRefined <- "D_gran_mono-de"
genes_cebpe[which((genes_cebpe$cluster_de=="F_rest") & 
                    (genes_cebpe$cluster==12)),]$cluster_deRefined <- "D_gran_mono-neutral"
genes_cebpe[which((genes_cebpe$cluster_de!="F_rest") & 
                    (genes_cebpe$cluster==14 | genes_cebpe$cluster==15)),]$cluster_deRefined <- "E_mono-de"
genes_cebpe[which((genes_cebpe$cluster_de=="F_rest") & 
                    (genes_cebpe$cluster==14 | genes_cebpe$cluster==15)),]$cluster_deRefined <- "E_mono-neutral"

table(genes_cebpe$cluster_de)
table(genes_cebpe$cluster_deRefined)

##------------------------------------------------##
## add motif information
motifAna <- read.table("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/gene_clusters/motifAna/REGIONS_INTEREST.find.unique", header=T, sep="\t")
genes_cebpe <- merge(genes_cebpe, motifAna[grep("E2F1_MOUSE.H10MO.A", motifAna$Motif.Name),c("PositionID", "MotifScore")], by.x="gene", by.y="PositionID", all=T)
colnames(genes_cebpe)[ncol(genes_cebpe)] <- "motifScore_E2f1"
genes_cebpe[is.na(genes_cebpe$motifScore_E2f1),]$motifScore_E2f1 <- 0

genes_cebpe <- merge(genes_cebpe, motifAna[grep("NFYB", motifAna$Motif.Name),c("PositionID", "MotifScore")], by.x="gene", by.y="PositionID", all=T)
colnames(genes_cebpe)[ncol(genes_cebpe)] <- "motifScore_Nfyb"
genes_cebpe[is.na(genes_cebpe$motifScore_Nfyb),]$motifScore_Nfyb <- 0

genes_cebpe <- merge(genes_cebpe, motifAna[grep("LIN54", motifAna$Motif.Name),c("PositionID", "MotifScore")], by.x="gene", by.y="PositionID", all=T)
colnames(genes_cebpe)[ncol(genes_cebpe)] <- "motifScore_Lin54"
genes_cebpe[is.na(genes_cebpe$motifScore_Lin54),]$motifScore_Lin54 <- 0

genes_cebpe <- merge(genes_cebpe, motifAna[grep("MYCN_MOUSE.H10MO.B", motifAna$Motif.Name),c("PositionID", "MotifScore")], by.x="gene", by.y="PositionID", all=T)
colnames(genes_cebpe)[ncol(genes_cebpe)] <- "motifScore_Myc"
genes_cebpe[is.na(genes_cebpe$motifScore_Myc),]$motifScore_Myc <- 0

genes_cebpe <- merge(genes_cebpe, motifAna[grep("CEBPA", motifAna$Motif.Name),c("PositionID", "MotifScore")], by.x="gene", by.y="PositionID", all=T)
colnames(genes_cebpe)[ncol(genes_cebpe)] <- "motifScore_Cebpa"
genes_cebpe[is.na(genes_cebpe$motifScore_Cebpa),]$motifScore_Cebpa <- 0

genes_cebpe <- merge(genes_cebpe, motifAna[grep("CEBPE", motifAna$Motif.Name),c("PositionID", "MotifScore")], by.x="gene", by.y="PositionID", all=T)
colnames(genes_cebpe)[ncol(genes_cebpe)] <- "motifScore_Cebpe"
  genes_cebpe[is.na(genes_cebpe$motifScore_Cebpe),]$motifScore_Cebpe <- 0

##------------------------------------------------##
## add nfyb peak information

peaks_nfyb <- read.table("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/genes_cebpe.bed.nfyb_signal", header=F)
colnames(peaks_nfyb) <- c("chr", "start", "end", "gene", "score", "strand", "log2FC_mm", "cluster_deRefined", "numTagsNfyb")

genes_cebpe <- merge(genes_cebpe, peaks_nfyb[,c("gene","numTagsNfyb")], by.x="gene", by.y="gene", all=T)
genes_cebpe$nfyb_bound <- ifelse(genes_cebpe$numTagsNfyb>0, "Y", "N")

THRESHOLD = summary(genes_cebpe[which(genes_cebpe$de_count>0),]$cebpe_lpmmy)[3]

genes_cebpe$peak_class <- "bkg"
genes_cebpe[which(genes_cebpe$e2f > 0 & genes_cebpe$cebpe_lpmmy > THRESHOLD),]$peak_class <- "e2f1+cebpe"
genes_cebpe[which(genes_cebpe$e2f > 0 & genes_cebpe$cebpe_lpmmy > THRESHOLD &
                    genes_cebpe$motifScore_Lin54>0 &
                    (genes_cebpe$motifScore_Nfyb>0 | genes_cebpe$nfyb_bound=="Y")),]$peak_class <- "e2f1+cebpe+nfyb+lin54"
# genes_cebpe[which(genes_cebpe$e2f > 0 & genes_cebpe$cebpe_lpmmy > THRESHOLD & genes_cebpe$motifScore_Lin54>0 &
#                     ((genes_cebpe$motifScore_Nfyb>0 | genes_cebpe$nfyb_bound=="Y") |
#                        (genes_cebpe$motifScore_Nfyb==0 & genes_cebpe$nfyb_bound=="N"))),]$peak_class <- "e2f1+cebpe+nfyb+lin54"

#genes_cebpe[grep("^Ccnb1$", genes_cebpe$gene),c("e2f", "e2f_bound", "cebpe_lpmmy", "motifScore_Nfyb", "numTagsNfyb", "nfyb_bound", "peak_class")]
genes_cebpe[grep("^Ccnb1$", genes_cebpe$gene),]$peak_class <- "e2f1+cebpe+nfyb+lin54"
genes_cebpe$peak_class <- factor(genes_cebpe$peak_class, levels=c("bkg", "e2f1+cebpe", "e2f1+cebpe+nfyb+lin54"), ordered=T)
table(genes_cebpe$peak_class)
table(genes_cebpe[grep("B_lpm", genes_cebpe$cluster_deRefined),]$peak_class)

##------------------------------------------------##
## add cebpA/E class information
genes_cebpe$cebp_class <- sprintf("%s_%s", genes_cebpe$cebpa_bound, genes_cebpe$cebpe_bound)
genes_cebpe$cebp_class <- factor(genes_cebpe$cebp_class, levels=c("N_Y", "Y_Y", "Y_N", "N_N"), ordered = T)

genes_cebpe$cebp_class_dis <- sprintf("%s_%s", genes_cebpe$cebpa_bound_dis, genes_cebpe$cebpe_bound_dis)
genes_cebpe$cebp_class_dis <- factor(genes_cebpe$cebp_class_dis, levels=c("N_Y", "Y_Y", "Y_N", "N_N"), ordered = T)

##------------------------------------------------##
## cell cycle genes information
## source: https://academic.oup.com/jmcb/article/11/8/703/5188008
cell_cycle <- c("Atad2","Bora","Btg3","Casp8ap2","Ccdc134","Ccna2","Ccnb1","Ccnb2","Ccne1","Ccne2","Cdc25a","Cdc25b","Cdc25c","Cdc5l","Cdc7","Cdca3","Cdca4","Cdca7","Cdk1","Cdk19","Cdk4","Cdkn2c","Cdkn2d","Cdkn3","Cks1b","Cks2","Crlf3","Dbf4b","Dlgap5","E2f1","E2f7","E2f8","Fam83d","Fbxo5","Foxm1","Fzr1","Gmnn","Lin54","Lin9","Mastl","Melk","Mybl2","Npat","Odf2","Pa2g4","Pbk","Pimreg","Pkmyt1","Prr11","Rbl1","Stil","Tcf19","Tfdp1","Ticrr","Toe1","Trim28","Ttf2","Ube2c","Ube2s","Uhrf2","Usp37","Wee1","Cdk2","Aurka","Kif14","Dtl","Gtse1","Plk1")
## source: Kim
cell_cycle <- c(cell_cycle, as.character(genes_cebpe[grepl("Cell_Div_Cycle|Cyclin|Cyclin_Dep_Kinase|G1_S|G2_M", genes_cebpe$funcClass),]$gene))
length(cell_cycle)
genes_cebpe$cell_cycle <- "not_cell_cycle"
genes_cebpe[genes_cebpe$gene %in% cell_cycle,]$cell_cycle <- "cell_cycle"

##------------------------------------------------##
## myc target information
myc_target <- read.table("/home/pundhir/project/genome_annotations/cell_cycle/myc_target_genes")$V1

genes_cebpe$myc_target <- "not_myc_target"
genes_cebpe[genes_cebpe$gene %in% myc_target,]$myc_target <- "myc_target"

##------------------------------------------------##
## add e2f1 and myc ChIP-seq signal at gene promoters
df <- read.table(pipe("cut -f 4,8,9,10,11 /home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/allGenes.E2f1_Myc_E2f4_Nfyb"))
colnames(df) <- c("gene", "e2f_chip", "myc_chip", "e2f4_chip", "nfyb_chip")
genes_cebpe <- merge(genes_cebpe, df, by.x="gene", by.y="gene")

##------------------------------------------------##
## add information on e2f1 and cebpe peak distance to each other

df <- read.table("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/cebpe_e2f1_distance", stringsAsFactors = F)
colnames(df) <- c("gene", "distance_cebpe_e2f1", "distance_cebpe_e2f1_class")
genes_cebpe <- merge(genes_cebpe, df, by.x="gene", by.y="gene")

##------------------------------------------------##
## add information on cebp peak class (cebpa_enhancer cebpa_promoter cebpe_enhancer cebpe_promoter)

df <- read.table(pipe("cut -f 4,7,14 /home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/genes_cebpe.bed.cebpPeakClass"), stringsAsFactors = F)
colnames(df) <- c("gene", "cluster_de", "cebp_closestPeakClass")
genes_cebpe <- merge(genes_cebpe, df[,c("gene", "cebp_closestPeakClass")], by.x="gene", by.y="gene")

##------------------------------------------------##
## properly format final data frame

genes_cebpe <- genes_cebpe[match(mat_raw$gene, genes_cebpe$gene),]
genes_cebpe <- genes_cebpe[,c(2:4,1,5:ncol(genes_cebpe))]

#genes_cebpe[grep("Cdkn3",genes_cebpe$gene),c(4,27,89,99,82,84,86,88)]

#(table(genes_cebpe[which(!is.na(genes_cebpe$funcClass) & genes_cebpe$cluster_de=="F_rest" & genes_cebpe$de_count>0),]$funcClass))
#(table(genes_cebpe[which(!is.na(genes_cebpe$funcClass)  & genes_cebpe$de_count>0),]$funcClass))

##------------------------------------------------##
## statistics on cell cycle genes

plot(Venn(list(gioti=c("Atad2","Bora","Btg3","Casp8ap2","Ccdc134","Ccna2","Ccnb1","Ccnb2","Ccne1","Ccne2","Cdc25a","Cdc25b","Cdc25c","Cdc5l","Cdc7","Cdca3","Cdca4","Cdca7","Cdk1","Cdk19","Cdk4","Cdkn2c","Cdkn2d","Cdkn3","Cks1b","Cks2","Crlf3","Dbf4b","Dlgap5","E2f1","E2f7","E2f8","Fam83d","Fbxo5","Foxm1","Fzr1","Gmnn","Lin54","Lin9","Mastl","Melk","Mybl2","Npat","Odf2","Pa2g4","Pbk","Pimreg","Pkmyt1","Prr11","Rbl1","Stil","Tcf19","Tfdp1","Ticrr","Toe1","Trim28","Ttf2","Ube2c","Ube2s","Uhrf2","Usp37","Wee1","Cdk2","Aurka","Kif14","Dtl","Gtse1","Plk1"), manual=as.character(genes_cebpe[grepl("Cell_Div_Cycle|Cyclin|Cyclin_Dep_Kinase|G1_S|G2_M", genes_cebpe$funcClass),]$gene))))

plot(Venn(list(c(genes_cebpe[grepl("Cell_Div_Cycle|Cyclin|Cyclin_Dep_Kinase|G1_S|G2_M", genes_cebpe$funcClass),]$gene), genes_cebpe[which(genes_cebpe$cell_cycle=="cell_cycle"),]$gene)))

plot(Venn(list(genes_cebpe[which(grepl("Cell_Div_Cycle|Cyclin|Cyclin_Dep_Kinase|G1_S|G2_M", genes_cebpe$funcClass) & genes_cebpe$cluster==4),]$gene, genes_cebpe[which(genes_cebpe$cell_cycle=="cell_cycle" & genes_cebpe$cluster==4),]$gene)))

plot(Venn(list(genes_cebpe[which(genes_cebpe$e2f > 0 & genes_cebpe$cebpe_lpmmy > THRESHOLD &
                    genes_cebpe$motifScore_Lin54>0 &
                    (genes_cebpe$motifScore_Nfyb>0 | genes_cebpe$nfyb_bound=="Y") & grepl("B_lpm", genes_cebpe$cluster_deRefined)),]$gene, 
       genes_cebpe[which(genes_cebpe$e2f > 0 & genes_cebpe$cebpe_lpmmy > THRESHOLD &
                    (genes_cebpe$motifScore_Nfyb>0 | genes_cebpe$nfyb_bound=="Y") & grepl("B_lpm", genes_cebpe$cluster_deRefined)),]$gene)))
```

#### write gene expression matrix to output file
```{r}
write.table(genes_cebpe, "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/genes_cebpe.bed", sep="\t", quote = F, row.names = F, col.names = T)
# write.table(genes_cebpe, "/home/pundhir/project/chip-seq-analysis/data/gene/cebpe/GENE_EXPRESSION.MATRIX", sep="\t", quote = F, row.names = F, col.names = T)
# write.table(genes_cebpe, "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/GENE_EXPRESSION.MATRIX", sep="\t", quote = F, row.names = F, col.names = T)
write.table(genes_cebpe[,c(1:58,83:93,104:114,117:124)], "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/xls/data_genes.xls", sep="\t", quote = F, row.names = F, col.names = T)
```

#### plot gene expression profile as heatmaps
```{r, eval=FALSE}
heatmap_lst <- list()
heatmap_lst[[1]] <- matrix2pheatmap(genes_cebpe[,c(8:9,10,11:26)], genes_cebpe$cluster, "row", colorRampPalette(c("dodgerblue4", "#2171b5", "grey97", "#fc8d59", "sienna3"), bias=1)(256))[[4]]
heatmap_lst[[2]] <- matrix2pheatmap(genes_cebpe[,c(28:29,30,31:46)], genes_cebpe$cluster, "row", colorRampPalette(c("dodgerblue4", "#2171b5", "grey97", "#fc8d59", "sienna3"), bias=1)(256))[[4]]
heatmap_lst[[3]] <- matrix2pheatmap(genes_cebpe[grep("Cebpa|Cebpe", genes_cebpe$gene),c(8:9,10,11:22)], c(1,1), "row", colorRampPalette(c("dodgerblue4", "#2171b5", "grey97", "#fc8d59", "sienna3"), bias=1)(256))[[4]]
heatmap_lst[[4]] <- matrix2pheatmap(genes_cebpe[grep("Cebpa|Cebpe", genes_cebpe$gene),c(28:29,30,31:42)], c(1,1), "row", colorRampPalette(c("dodgerblue4", "#2171b5", "grey97", "#fc8d59", "sienna3"), bias=1)(256))[[4]]
# heatmap_lst[[3]] <- matrix2pheatmap(mat_raw[(rownames(mat_raw) %in% GENES_DE), c(2,3,5:(ncol(mat_raw)-1))], mat_raw[rownames(mat_raw) %in% GENES_DE,]$cluster, "row", colorRampPalette(c("dodgerblue4", "#2171b5", "grey97", "#fc8d59", "sienna3"), bias=1)(256))[[4]]

pdf("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/pdf/heatmap_gene_expression.pdf", width=10)
gHeatmap <- grid.arrange(arrangeGrob(grobs= heatmap_lst[c(1,2)],ncol=2))
matrix2Heatmap(genes_cebpe[,c(8:9,10,11:26)], genes_cebpe$cluster, "row", colorRampPalette(c("dodgerblue4", "#2171b5", "grey97", "#fc8d59", "sienna3"), bias=1)(256))
dev.off()

pdf("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/pdf/heatmap_cebpAE_expression.pdf", width=10)
gHeatmap <- grid.arrange(arrangeGrob(grobs= heatmap_lst[c(3,4)], ncol=1))
dev.off()

pdf("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/pdf/heatmap_gene_expression_refined.pdf", width=10)
  for(i in 1:3) {
    mat_int <- genes_cebpe
    if(i==2 | i==1) {
      if(i==2) {
        mat_int <- genes_cebpe[which(genes_cebpe$cluster_de!="F_rest"),]
      }
      mat_int <- mat_int[order(mat_int$cluster_de, mat_int$cluster),]
      #mat_int <- mat_int[order(mat_int$cluster_de, mat_int$cluster, mat_int$funcClass),]
      
      mat_int$cluster <- factor(mat_int$cluster, levels=c(which(table(mat_int$cluster)>0)), ordered = T)
      col_funcClass <- c("#bf5b17", "#bf5b17", "#f0027f", "#fdc086", "#ffff99", "#beaed4", "#6a3d9a", "#386cb0", "#7fc97f")
      funcClass_Int <- c("DNA_replication", "G1_S", "G2_M_Mitosis", "GP1", "GP2", "GP3", "TF", "SURFACE", "SCHLAFEN")
      col_funcClass <- c(col_funcClass, rep("#999999", length(names(which(table(mat_int[!mat_int$funcClass %in% funcClass_Int,]$funcClass)>0)))))
      names(col_funcClass) <- c(funcClass_Int, names(which(table(mat_int[!mat_int$funcClass %in% funcClass_Int,]$funcClass)>0)))
      funcClass_Int <- c("G1_S", "G2_M", "GP1", "GP2", "GP3", "GRAN", "MONO")
    
      if(i==1) {
        funcClass_Int <- c("GP1", "GP2", "GP3")
      }

      col_cluster <- colorRampPalette(brewer.pal(length(unique(mat_int$cluster_de)), "BrBG"), bias=1)(length(unique(mat_int$cluster_de)))
      names(col_cluster) <- levels(as.factor((mat_int$cluster_de)))
    
      labels_gp = gpar(fontsize = rep(1, length(mat_int[which(!is.na(mat_int$funcClass)),]$gene)))
      labels_at <- which(mat_int$funcClass %in% funcClass_Int)
      labels_genes <- sprintf("%s_%s", mat_int[which(mat_int$funcClass %in% funcClass_Int),]$gene, mat_int[which(mat_int$funcClass %in% funcClass_Int),]$funcClass)
      la <- rowAnnotation(cluster_de = as.factor(mat_int$cluster_de),
                        # cluster = as.factor(mat_int$cluster),
                        col = list(cluster_de = col_cluster))
    
      ra <- rowAnnotation(na_col = NA,
                        de_lpm = mat_int$de_lpm,
                        de_mm = mat_int$de_mm,
                        bar = mat_int$funcClass,
                        # pu1 = mat_int$pu1_bound,
                        # cebpa = mat_int$cebpa_bound,
                        # cebpe = mat_int$cebpe_bound,
                        # e2f = mat_int$e2f_bound,
                        # pu1_dis = mat_int$pu1_bound_dis,
                        # cebpa_dis = mat_int$cebpa_bound_dis,
                        # cebpe_dis = mat_int$cebpe_bound_dis,
                        # e2f_dis = mat_int$e2f_bound_dis,
                        foo = anno_mark(at = labels_at, labels = labels_genes),
                        col = list(bar = col_funcClass,
                                   de_lpm = c("up" = "red", "neutral" = "white", "down" = "blue"),
                                   de_mm = c("up" = "red", "neutral" = "white", "down" = "blue"),
                                   pu1 = c("Y" = "#ff7f00", "N" = "#ffffff"),
                                   cebpa = c("Y" = "#1f78b4", "N" = "#ffffff"),
                                   cebpe = c("Y" = "#6a3d9a", "N" = "#ffffff"),
                                   #e2f = c("Y" = "#33a02c", "N" = "#ffffff"),
                                   pu1_dis = c("1" = "#ff7f00", "0" = "#ffffff"),
                                   cebpa_dis = c("1" = "#1f78b4", "0" = "#ffffff"),
                                   cebpe_dis = c("1" = "#6a3d9a", "0" = "#ffffff"),
                                   e2f_dis = c("1" = "#33a02c", "0" = "#ffffff")
                                   )
                        )
      print(Heatmap(as.matrix(t(scale(t(mat_int[,c(8:9,10,11:22)])))), cluster_rows=F, cluster_columns=F, col=colorRampPalette(c("dodgerblue4","grey97", "sienna3"), bias=1.5)(250), use_raster=T, show_row_names = F, raster_device="tiff", raster_quality = 10, left_annotation = la, right_annotation = ra))
  } else if(i==3) {
      mat_int <- genes_cebpe[which(genes_cebpe$cell_cycle=="cell_cycle" & grepl("A_epm|B_lpm", genes_cebpe$cluster_deRefined)),]
      #mat_int <- mat_int[!is.na(mat_int$funcClass),]
      
      set.seed(1)
      bkg <- genes_cebpe[which(genes_cebpe$cluster_deRefined=="C_gran-de"),]
      mat_int <- rbind(mat_int, bkg[sample(nrow(bkg), 20),])
      
      mat_int <- mat_int[order(mat_int$cluster_deRefined, mat_int$cluster),]
      #mat_int <- mat_int[order(mat_int$cluster_de, mat_int$cluster, mat_int$funcClass),]
      
      mat_int$cluster <- factor(mat_int$cluster, levels=c(which(table(mat_int$cluster)>0)), ordered = T)
      col_funcClass <- c("#bf5b17", "#bf5b17", "#f0027f", "#fdc086", "#ffff99", "#beaed4", "#6a3d9a", "#386cb0", "#7fc97f")
      funcClass_Int <- c("DNA_replication", "G1_S", "G2_M_Mitosis", "GP1", "GP2", "GP3", "TF", "SURFACE", "SCHLAFEN")
      col_funcClass <- c(col_funcClass, rep("#999999", length(names(which(table(mat_int[!mat_int$funcClass %in% funcClass_Int,]$funcClass)>0)))))
      names(col_funcClass) <- c(funcClass_Int, names(which(table(mat_int[!mat_int$funcClass %in% funcClass_Int,]$funcClass)>0)))
      funcClass_Int <- c("G1_S", "G2_M", "GP1", "GP2", "GP3", "GRAN", "MONO")
    
      col_cluster <- colorRampPalette(brewer.pal(length(unique(mat_int$cluster_deRefined)), "BrBG"), bias=1)(length(unique(mat_int$cluster_deRefined)))
      names(col_cluster) <- levels(as.factor((mat_int$cluster_deRefined)))
    
      labels_gp = gpar(fontsize = rep(1, length(mat_int$gene)))
      labels_at <- seq(1, length(mat_int$gene))
      labels_genes <- mat_int$gene
      la <- rowAnnotation(cluster_de = as.factor(mat_int$cluster_deRefined),
                        # cluster = as.factor(mat_int$cluster),
                        col = list(cluster_de = col_cluster))
    
      ra <- rowAnnotation(na_col = NA,
                        de_lpm = mat_int$de_lpm,
                        de_mm = mat_int$de_mm,
                        # bar = mat_int$funcClass,
                        # pu1 = mat_int$pu1_bound,
                        # cebpa = mat_int$cebpa_bound,
                        # cebpe = mat_int$cebpe_bound,
                        # myc = mat_int$myc_bound,
                        # e2f = mat_int$e2f_bound,
                        # myc_chip = log2(mat_int$myc_chip),
                        # e2f_chip = log2(mat_int$e2f_chip),
                        # pu1_dis = mat_int$pu1_bound_dis,
                        # cebpa_dis = mat_int$cebpa_bound_dis,
                        # cebpe_dis = mat_int$cebpe_bound_dis,
                        # e2f_dis = mat_int$e2f_bound_dis,
                        foo = anno_mark(at = labels_at, labels = labels_genes),
                        col = list(bar = col_funcClass,
                                   de_lpm = c("up" = "red", "neutral" = "white", "down" = "blue"),
                                   de_mm = c("up" = "red", "neutral" = "white", "down" = "blue"),
                                   pu1 = c("Y" = "#ff7f00", "N" = "#ffffff"),
                                   cebpa = c("Y" = "#1f78b4", "N" = "#ffffff"),
                                   cebpe = c("Y" = "#6a3d9a", "N" = "#ffffff"),
                                   myc = c("Y" = "#33a02c", "N" = "#ffffff"),
                                   e2f = c("Y" = "#33a02c", "N" = "#ffffff"),
                                   myc_chip = circlize::colorRamp2(c(min(log2(mat_int$myc_chip)), 
                                                                     1, max(log2(mat_int$myc_chip))), 
                                                                   c("#998ec3", "#f7f7f7", "#f1a340")),
                                   e2f_chip = circlize::colorRamp2(c(min(log2(mat_int$e2f_chip)), 
                                                                     1, max(log2(mat_int$e2f_chip))), 
                                                                   c("#998ec3", "#f7f7f7", "#f1a340")),
                                   pu1_dis = c("1" = "#ff7f00", "0" = "#ffffff"),
                                   cebpa_dis = c("1" = "#1f78b4", "0" = "#ffffff"),
                                   cebpe_dis = c("1" = "#6a3d9a", "0" = "#ffffff"),
                                   e2f_dis = c("1" = "#33a02c", "0" = "#ffffff")
                                   )
                        )
      #print(Heatmap(as.matrix(t(scale(t(mat_int[,c(8:9,11:22)])))), cluster_rows=F, cluster_columns=F, col=colorRampPalette(c("dodgerblue4","grey97", "sienna3"), bias=1.5)(250), use_raster=T, show_row_names = F, raster_device="tiff", raster_quality = 10, left_annotation = la, right_annotation = ra))
      ha = HeatmapAnnotation(summary = anno_summary(gp = gpar(fill = c(2,2,3,3)),  height = unit(2, "cm")))
      print(Heatmap(as.matrix(t(scale(t(mat_int[,c(8:9,10,11:22)])))), cluster_rows=F, cluster_columns=F,
              col=colorRampPalette(c("dodgerblue4","grey97", "sienna3"), bias=1.5)(250), use_raster=T, show_row_names = F,
              raster_device="tiff", raster_quality = 10, left_annotation = la, right_annotation = ra, 
              row_split = mat_int$cluster_deRefined) +
        Heatmap(log2(mat_int$e2f_chip), name = "mat", top_annotation = ha, width = unit(2, "cm"), 
                row_split = mat_int$cluster_deRefined, cluster_rows=F, 
                col=rev(colorBrewer2palette(name="PuOr", bias = 0.8))) +
        Heatmap(log2(mat_int$e2f4_chip), name = "mat", top_annotation = ha, width = unit(2, "cm"), 
                row_split = mat_int$cluster_deRefined, cluster_rows=F, 
                col=rev(colorBrewer2palette(name="PuOr", bias = 0.8))) +
        Heatmap(log2(mat_int$myc_chip), name = "mat", top_annotation = ha, width = unit(2, "cm"), 
                row_split = mat_int$cluster_deRefined, cluster_rows=F,
                col=rev(colorBrewer2palette(name="PuOr", bias = 0.8))) + 
          Heatmap(log2(mat_int$cebpe_lpmmy), name = "mat", top_annotation = ha, width = unit(2, "cm"), 
                row_split = mat_int$cluster_deRefined, cluster_rows=F,
                col=rev(colorBrewer2palette(name="PuOr", bias = 0.8))))
    }
  }
dev.off()

# df.melt <- melt(mat_int[,c("cluster_deRefined", "e2f_chip", "e2f4_chip", "myc_chip")])
# ggboxplot(df.melt, x="cluster_deRefined", y="value", notch = F, notchwidth = 0.6, yscale="log2") +
#   facet_wrap(~variable, scales="free") +
#   theme_bw() +
#   scale_fill_manual(values = c("#a6cee3", "#a6cee3")) +
#   stat_compare_means(comparisons = list(c("A_epm-de", "A_epm-neutral"),
#                                         c("A_epm-de", "B_lpm-de"),
#                                         c("B_lpm-de", "B_lpm-neutral")), 
#                      method.args = list(alternative = "g")) +
#   theme(legend.position = "none") + ylab("ChIP signal (log2)")


# library(gplots)
# cols <- rep('black', nrow(mat[1:100,]))
# cols[!is.na(mat$funcClass)] <- 'red'
# heatmap.2(as.matrix(mat[1:100,c(2,3,5:(ncol(mat)-2))]), Colv=F, Rowv=F, scale="row", dendrogram="none", colRow = cols, trace="none", col=colorRampPalette(c("dodgerblue4","grey97", "sienna3"), bias=1.5)(250))
```

#### clustering for differentially expressed genes
```{r, cache=TRUE, cache.vars="KC_CEBPE_RAW", cache.comments=FALSE}
reqSamples <- c("lsk_cebpa_wt", "pregm_cebpa_wt", "gmp_cebpa_wt", "gmpG_wt", "epm_wt", "lpm_wt", "my_wt", "mm_wt", "bc_wt", "pmn_wt", "gmpM_wt", "pmo1_wt", "pmo2_wt", "mo1_wt", "mo2_wt", "gmpG_ko", "epm_ko", "lpm_ko", "mm_ko")

mat_raw_de <- cbind(expr_data$GeneID, as.data.frame((do.call(cbind, lapply(unique(gsub("_[Rr]+ep.*", "", colnames(expr_data[,c(2:ncol(expr_data))]))), function(x) rowMeans(rle_normalize(expr_data[,grep(sprintf("^%s", x), colnames(expr_data))], "none")))))))
colnames(mat_raw_de) <- c("gene", unique(gsub("_[Rr]+ep.*", "", colnames(expr_data[,c(2:ncol(expr_data))]))))
mat_raw_de <- mat_raw_de[,c(reqSamples)]
row.names(mat_raw_de) <- expr_data$GeneID
mat_raw_de <- mat_raw_de[rownames(mat_raw_de) %in% genes_cebpe[which(genes_cebpe$de_count>0),]$gene,]
mat_raw_de <- mat_raw_de[apply(mat_raw_de, 1, function(x) length(x[x>1]) >=2),]
mat_raw_de <- mat_raw_de[apply(mat_raw_de[,grep("wt", colnames(mat_raw_de))], 1, function(x) ifelse(sd(x, na.rm=TRUE)>0, T, F)),]
cluster_info_matrix(mat_raw_de[,grep("wt", colnames(mat_raw_de))])
set.seed(1)
#KC_CEBPE_RAW_DE <- cluster_matrix(mat_raw_de[,grep("wt", colnames(mat_raw_de))],9)
#mat_raw_de$cluster <- as.numeric(factor(KC_CEBPE_RAW_DE$cluster, levels=c(5,1,3,6,13,12,9,2,15,7,4,14,10,11,8), ordered = T))
mat_raw_de$cluster <- KC_CEBPE_RAW_DE$cluster
mat_raw_de <- cbind(row.names(mat_raw_de), mat_raw_de)
colnames(mat_raw_de)[1] <- "gene"
mat_raw_de <- mat_raw_de[order(mat_raw_de$cluster),]
colnames(mat_raw_de)
table(KC_CEBPE_RAW_DE$cluster)
table(mat_raw_de$cluster)

#matrix2pheatmap(mat_raw_de[,c(2,3,5:(ncol(mat_raw)-1))], mat_raw_de$cluster, "row", colorRampPalette(c("dodgerblue4", "#2171b5", "grey97", "#fc8d59", "sienna3"), bias=1)(256))[[4]]
```

#### analyze gene clusters for the enrichment of differentially expressed genes
```{r}
GENES_ALL <- (unique(c(rownames(res_gmpG), rownames(res_epm), rownames(res_lpm), rownames(res_mm))))
GENES_DE <- (unique(c(rownames(res_gmpG[which(res_gmpG$class!="neutral"),]), rownames(res_epm[which(res_epm$class!="neutral"),]), 
                rownames(res_lpm[which(res_lpm$class!="neutral"),]), rownames(res_mm[which(res_mm$class!="neutral"),]))))
GENES_UP <- (unique(c(rownames(res_gmpG[which(res_gmpG$class=="up"),]), rownames(res_epm[which(res_epm$class=="up"),]), 
                rownames(res_lpm[which(res_lpm$class=="up"),]), rownames(res_mm[which(res_mm$class=="up"),]))))
GENES_DOWN <- (unique(c(rownames(res_gmpG[which(res_gmpG$class=="down"),]), rownames(res_epm[which(res_epm$class=="down"),]), 
                rownames(res_lpm[which(res_lpm$class=="down"),]), rownames(res_mm[which(res_mm$class=="down"),]))))
# input <- list(up=GENES_UP, down=GENES_DOWN, all=GENES_ALL)
# plot(Venn(input))
# names(attr(gplots::venn(input, show.plot=FALSE), "intersections"))
# GENES_DOWN <- attr(gplots::venn(input, show.plot=FALSE), "intersections")$"down:all"
# GENES_UP <- attr(gplots::venn(input, show.plot=FALSE), "intersections")$"up:all"
# GENES_DE <- c(GENES_DOWN, GENES_UP)

# input <- list(gmpG=rownames(res_gmpG[which(res_gmpG$class!="neutral"),]), epm=rownames(res_epm[which(res_epm$class!="neutral"),]),
#               lpm=rownames(res_lpm[which(res_lpm$class!="neutral"),]), mm=rownames(res_mm[which(res_mm$class!="neutral"),]))
# GENES_DE_LIST <- attr(gplots::venn(input, show.plot=FALSE), "intersections")

df <- as.data.frame(cbind(unique(mat_raw$cluster), table(mat_raw[rownames(mat_raw) %in% GENES_DE,]$cluster), table(mat_raw$cluster),
                          table(mat_raw[rownames(mat_raw) %in% GENES_UP,]$cluster),
                          table(mat_raw[rownames(mat_raw) %in% GENES_DOWN,]$cluster)))
colnames(df) <- c("cluster", "genes_de", "genes_all", "genes_up", "genes_down")
df$cluster <- factor(df$cluster, levels=levels(genes_cebpe$cluster), ordered = T)
df$perDE <- (df$genes_de*100)/df$genes_all
df$perUP <- (df$genes_up*100)/df$genes_all
df$perDOWN <- (df$genes_down*100)/df$genes_all
df$pVal <- apply(df[,c(2:3)], 1, function(x) binom.test(c(x[1], x[2]), p=apply(df[,c(2:3)], 2, function(x) sum(x))[1]/apply(df[,c(2:3)], 2, function(x) sum(x))[2], alternative="g")$p.value)

# df$cluster <- factor(df$cluster, levels=df[order(-df$perDE),]$cluster, ordered = T)
# df$cluster <- factor(df$cluster, levels=df[order(-df$genes_up, -df$genes_down),]$cluster, ordered = T)

p1 <- ggplot(NULL, aes(cluster, value)) + theme_bw() +
  geom_bar(aes(fill="genes_all"), data=melt(df[,c(1,3)]), stat="identity", alpha=0.4) +
  geom_bar(aes(fill=variable), data=melt(df[,c(1,4,5)]), stat="identity", alpha=0.8) + 
  scale_y_continuous(position="right") +
  scale_fill_manual(values=c("#999999", "#67a9cf", "#ef8a62")) +
  theme(legend.position="none", legend.text=element_text(size=6), axis.text.y = element_text(angle = -90, hjust = 1)) +
  ylab("")
  #geom_bar(aes(fill="genes_de"), data=melt(df[,c(1,2)]), stat="identity", alpha=0.8)

cluster_de <- lapply(c("down", "up", "neutral"), function(z) as.data.frame(t(as.data.frame(lapply(levels(genes_cebpe$cluster), function(x) apply(genes_cebpe[which(genes_cebpe$cluster==x),c("de_gmpG", "de_epm", "de_lpm", "de_mm")], 2, function(y) length(which(y==z))))))))
cluster_de <- lapply(cluster_de, function(x) { row.names(x) <- levels(genes_cebpe$cluster); return(x) })

# cluster_cebpe <- lapply(c("down", "up", "neutral"), function(z) as.data.frame(t(as.data.frame(lapply(levels(genes_cebpe$cluster), function(y) unlist(lapply(c("de_gmpG", "de_epm", "de_lpm", "de_mm"), function(x) 
#   as.numeric(sprintf("%0.2f", -log10(binom.test(c(nrow(genes_cebpe[which(genes_cebpe$cluster==y & genes_cebpe[,x]==z & genes_cebpe$cebpe_lpmmy > median(genes_cebpe$cebpe_lpmmy)),]), nrow(genes_cebpe[which(genes_cebpe$cluster==y & genes_cebpe[,x]==z & genes_cebpe$cebpe_lpmmy <= median(genes_cebpe$cebpe_lpmmy)),])), nrow(genes_cebpe[which(genes_cebpe$cebpe_lpmmy > median(genes_cebpe$cebpe_lpmmy)),])/nrow(genes_cebpe), alternative="g")$p.value)))
#   )))))))

cluster_cebpe <- lapply(c("down", "up", "neutral"), function(z) as.data.frame(t(as.data.frame(lapply(levels(genes_cebpe$cluster), function(y) unlist(lapply(c("de_gmpG", "de_epm", "de_lpm", "de_mm"), function(x) 
  as.numeric(sprintf("%0.2f", -log10(wilcox.test(genes_cebpe[which(genes_cebpe$cluster==y & genes_cebpe[,x]==z),]$cebpe_lpmmy, genes_cebpe$cebpe_lpmmy, alternative = "g")$p.value)))
  )))))))

# cluster_cebpe <- lapply(c("down", "up", "neutral"), function(z) as.data.frame(t(as.data.frame(lapply(levels(genes_cebpe$cluster), function(y) unlist(lapply(c("de_gmpG", "de_epm", "de_lpm", "de_mm"), function(x) as.numeric(sprintf("%0.2f", median(genes_cebpe[which(genes_cebpe$cluster==y & genes_cebpe[,x]==z),]$cebpe_lpmmy))))))))))
cluster_cebpe <- lapply(cluster_cebpe, function(x) { colnames(x) <- c("de_gmpG", "de_epm", "de_lpm", "de_mm"); row.names(x) <- levels(genes_cebpe$cluster); return(x) })

p <- lapply(seq(1,3,1), function(i) {
  if(i==1) { col="Blues";} else if(i==2) { col="Reds"} else { col="Greys"; }
  x <- cluster_de[[i]]
  y <- cluster_cebpe[[i]]
  x$cluster <- levels(genes_cebpe$cluster);
  y$cluster <- levels(genes_cebpe$cluster);
  # x$cluster <- factor(x$cluster, levels=df[order(-df$perDE),]$cluster, ordered = T);
  x$cluster <- factor(x$cluster, levels=levels(genes_cebpe$cluster), ordered = T)
  y$cluster <- factor(y$cluster, levels=levels(genes_cebpe$cluster), ordered = T)
  rownames(x) <- NULL;
  rownames(y) <- NULL;
  x.melt <- cbind(melt(x), melt(y)[,3])
  colnames(x.melt)[4] <- "pVal"
  #x.melt$pVal <- ifelse(x.melt$pVal > 10, 1, 0)
  ggplot(x.melt, aes(x = cluster, y = variable, label = value)) +
    geom_point(aes(size = value, alpha=.02, fill = pVal), shape=21) + 
    geom_text(hjust = 1, size = 2) +
    scale_size(range = c(1,10)) +
    theme_bw() + scale_y_discrete(position="right") + ylab("") + 
    scale_fill_gradientn(colours = (colorRampPalette(brewer.pal(9, col), bias=1)(256))) + xlab("") +
    theme(legend.position="none", legend.text = element_text(size=6), axis.text.y = element_text(angle = -90, hjust = 1))
})

p[[4]] <- ggplot(as.data.frame(table(genes_cebpe[which(genes_cebpe$myc_target=="myc_target"),c("cluster", "myc_target")])), aes(cluster, Freq, label=Freq)) + geom_bar(stat="identity") + facet_wrap(~myc_target, scales="free") + theme_bw() +
  geom_text(size = 3, position = position_stack(vjust = 0.5)) + scale_y_continuous(position = "right") + xlab("") + ylab("") +
  theme(legend.position="none", legend.text = element_text(size=6), axis.text.y = element_text(angle = -90, hjust = 1))

p[[5]] <- ggplot(as.data.frame(table(genes_cebpe[grepl("G1_S|G2_M", genes_cebpe$funcClass),c("cluster", "cell_cycle", "funcClass")])), 
                 aes(cluster, Freq, label=Freq, fill=funcClass)) + geom_bar(position="stack", stat="identity") + theme_bw() +
  geom_text(size = 3, position = position_stack(vjust = 0.5)) + scale_y_continuous(position = "right") + xlab("") + ylab("") +
  theme(legend.position="none", legend.text = element_text(size=6), axis.text.y = element_text(angle = -90, hjust = 1))

# ggplot(as.data.frame(table(genes_cebpe[which(genes_cebpe$cell_cycle=="cell_cycle"),c("cluster", "cell_cycle", "funcClass")])), 
#        aes(cluster, Freq, label=Freq, fill=funcClass)) + geom_bar(position="stack", stat="identity") + facet_wrap(~cell_cycle, scales="free") + theme_bw() +
#    geom_text(size = 3, position = position_stack(vjust = 0.5)) + scale_y_continuous(position = "right") + xlab("") + ylab("") +
#    theme(legend.position="none", legend.text = element_text(size=6), axis.text.y = element_text(angle = -90, hjust = 1))

p[[6]] <- ggplot((melt(summaryBy(myc_chip ~ cluster, data = genes_cebpe, FUN=list(median)))), aes(variable, cluster, fill= value)) + 
  geom_tile() + ylab("") + scale_x_discrete(position="right") +
  scale_fill_gradientn(colours = (colorRampPalette(brewer.pal(9, "Blues"), bias=0.2)(256))) + theme_bw() +
  theme(legend.position="none", legend.text = element_text(size=6), axis.text.y = element_text(angle = -90, hjust = 1)) +
  coord_flip() + xlab("")

p[[7]] <- ggplot((melt(summaryBy(e2f_chip ~ cluster, data = genes_cebpe, FUN=list(median)))), aes(variable, cluster, fill= value)) + 
  geom_tile() + ylab("") + scale_x_discrete(position="right") +
  scale_fill_gradientn(colours = (colorRampPalette(brewer.pal(9, "Blues"), bias=0.2)(256))) + theme_bw() +
  theme(legend.position="none", legend.text = element_text(size=6), axis.text.y = element_text(angle = -90, hjust = 1)) +
  coord_flip() + xlab("")

p[[6]] <- ggline(genes_cebpe, x = "cluster", y = "myc_chip", add = c("mean_se"))

p[[7]] <- ggline(genes_cebpe, x = "cluster", y = "e2f_chip", add = c("mean_se"))

ggarrange(ggplot(genes_cebpe[grep("A_epm|B_lpm", genes_cebpe$cluster_deRefined),], aes(cluster_deRefined, log(myc_chip))) + geom_boxplot() + 
            theme(legend.position="none", legend.text = element_text(size=6), axis.text.x = element_text(angle = 45, hjust = 1)),
          ggplot(genes_cebpe[grep("A_epm|B_lpm", genes_cebpe$cluster_deRefined),], aes(cluster_deRefined, log(e2f_chip))) + geom_boxplot() +
            theme(legend.position="none", legend.text = element_text(size=6), axis.text.x = element_text(angle = 45, hjust = 1)), nrow=1, ncol=2)

gDE <- ggarrange(p[[1]], p[[2]], p[[3]], p[[4]], p[[5]], p[[6]], p[[7]], ncol = 1, nrow=7)

# # cluster_cebpe <- lapply(c("down", "up", "neutral"), function(z) as.data.frame(t(as.data.frame(lapply(levels(genes_cebpe$cluster), function(y) unlist(lapply(c("de_gmpG", "de_epm", "de_lpm", "de_mm"), function(x) as.numeric(sprintf("%0.2f", median(genes_cebpe[which(genes_cebpe$cluster==y & genes_cebpe[,x]==z),]$cebpe_lpmmy))))))))))
# 
# # cluster_cebpe <- lapply(c("down", "up", "neutral"), function(z) as.data.frame(t(as.data.frame(lapply(levels(genes_cebpe$cluster), function(y) unlist(lapply(c("de_gmpG", "de_epm", "de_lpm", "de_mm"), function(x) nrow(genes_cebpe[which(genes_cebpe$cluster==y & genes_cebpe[,x]==z & genes_cebpe$cebpe_lpmmy > median(genes_cebpe$cebpe_lpmmy)),]))))))))
# 
# cluster_cebpe <- lapply(c("down", "up", "neutral"), function(z) as.data.frame(t(as.data.frame(lapply(levels(genes_cebpe$cluster), function(y) unlist(lapply(c("de_gmpG", "de_epm", "de_lpm", "de_mm"), function(x) 
#   as.numeric(sprintf("%0.2f", -log10(binom.test(c(nrow(genes_cebpe[which(genes_cebpe$cluster==y & genes_cebpe[,x]==z & genes_cebpe$cebpe_lpmmy > median(genes_cebpe$cebpe_lpmmy)),]), nrow(genes_cebpe[which(genes_cebpe$cluster==y & genes_cebpe[,x]==z & genes_cebpe$cebpe_lpmmy <= median(genes_cebpe$cebpe_lpmmy)),])), nrow(genes_cebpe[which(genes_cebpe$cebpe_lpmmy > median(genes_cebpe$cebpe_lpmmy)),])/nrow(genes_cebpe), alternative="g")$p.value)))
#   )))))))
# 
# p <- lapply(seq(1,3,1), function(i) {
#   if(i==1) { col="Blues";} else if(i==2) { col="Reds"} else { col="Greys"; }
#   x <- cluster_cebpe[[i]]
#   colnames(x) <- c("de_gmpG", "de_epm", "de_lpm", "de_mm")
#   x$cluster <- levels(genes_cebpe$cluster);
#   # x$cluster <- factor(x$cluster, levels=df[order(-df$perDE),]$cluster, ordered = T);
#   x$cluster <- factor(x$cluster, levels=levels(genes_cebpe$cluster), ordered = T)
#   rownames(x) <- NULL;
#   ggplot(melt(x), aes(x = cluster, y = variable, label = value)) +
#     geom_point(aes(size = value, colour = value, alpha=.02)) + 
#     geom_text(hjust = 1, size = 2) +
#     scale_size(range = c(1,10)) +
#     theme_bw() + scale_y_discrete(position="right") + ylab("") + 
#     scale_color_gradientn(colours = (colorRampPalette(brewer.pal(9, col), bias=1)(256))) + xlab("") +
#     theme(legend.position="none", legend.text = element_text(size=6), axis.text.y = element_text(angle = -90, hjust = 1))
# })

```

#### analyze interesting gene clusters via plotting their gene expression profiles as line plots
```{r}
p <- lapply(names(geneSet), function(x) {
  # t1 <- summaryBy(gmpG_wt + epm_wt + lpm_wt + mmlike_wt + mm_wt ~ cluster, data = x, FUN=list(median))
  # t1$class <- "wt"
  # colnames(t1) <- gsub("_wt", "", colnames(t1))
  # t2 <- summaryBy(gmpG_ko + epm_ko + lpm_ko + mmlike_ko + mm_ko ~ cluster, data = x, FUN=list(median))
  # t2$class <- "ko"
  # colnames(t2) <- gsub("_ko", "", colnames(t2))
  # t <- rbind(t1, t2)
  # t.melt <- melt(t)
  # ggplot(t.melt, aes(variable, log(value), group=1)) +
  #   stat_summary(data = t.melt[which(t.melt$class=="wt"),], fun.y=mean, geom="point", size=3, color="red") +
  #   stat_summary(data = t.melt[which(t.melt$class=="wt"),], fun.data=mean_cl_normal, geom="smooth", alpha=1) +
  #   stat_summary(data = t.melt[which(t.melt$class=="ko"),], fun.y=mean, geom="point", size=3, color="blue") +
  #   stat_summary(data = t.melt[which(t.melt$class=="ko"),], fun.data=mean_cl_normal, geom="smooth", alpha=1) + 
  #   ylim(c(4,8)) + theme_bw()
  #t.melt <- (melt(genes_cebpe[which(genes_cebpe$cluster_de==x),c(11,12,13,15,23,24,25,26)]))
  t.melt <- (melt(genes_cebpe[which(genes_cebpe$cluster_de==x),c(31,32,33,35,43,44,45,46)]))
  #t.melt <- (melt(log(x[,c(11,12,13,15,16,24,25,26,27,28)]+1)))
  t.melt$class <- gsub("_.*", "", (t.melt$variable))
  t.melt$variable <- gsub("^[^\\_]*_", "", t.melt$variable)
  t.melt <- summarySE(t.melt, measurevar = "value", groupvars = c("class", "variable"))
  t.melt$class <- factor(t.melt$class, levels=c("gmpG", "epm", "lpm", "mm"))
  ggplot(t.melt, aes(class, value, color=variable)) +
    geom_errorbar(aes(ymin=value-se, ymax=value+se), width=.1, alpha=1) +
    geom_line(aes(group=factor(variable), color=factor(variable)), alpha=1) +
    geom_point(shape = 21, size = 2, fill="white") +
    scale_color_manual(values=c("#313695", "#a50026")) +
    theme_bw() + theme(legend.position="top")
})
names(p) <- names(geneSet)

# table(ifelse(genes_cebpe$cluster_de_count>0, "Deregulated", "Neutral"))
# ggplot(melt(genes_cebpe[,c(68,69)]), aes(cluster_de, (value))) + geom_boxplot() + theme_bw()

# p[[8]] <- ggplot(melt(genes_cebpe[,c("pu1_pregm", "cebpa_gmp", "cebpe_lpmmy", "cluster_de")]), aes(cluster_de, log(value), fill=cluster_de)) + 
#   geom_boxplot(notch=TRUE, notchwidth = 0.3) + facet_wrap(~variable, scales="free") + theme_bw()

p[["e2f"]] <- freqPlot(melt(table(genes_cebpe[,c("cluster_de", "e2f_bound")])), "density", "e2f") +
  #freqPlot(melt(table(as.data.frame(cbind(genes_cebpe$cluster_de, ifelse(genes_cebpe$e2f>0, "Y", "N"))))), "density", "e2f") + 
  theme_bw() + xlab("") + theme(legend.position="top")

p[["myc"]] <- freqPlot(melt(table(genes_cebpe[,c("cluster_de", "myc_bound")])), "density", "myc") +
  #freqPlot(melt(table(as.data.frame(cbind(genes_cebpe$cluster_de, ifelse(genes_cebpe$myc>0, "Y", "N"))))), "density", "myc") + 
  theme_bw() + xlab("") + theme(legend.position="top")

p[["cebpe"]] <- freqPlot(melt(table(genes_cebpe[,c("cluster_de","cebpe_bound")])), "density", "cebpe") + theme_bw() + theme(legend.position="top")

gDEProfile <- ggarrange(p[["A_epm"]], p[["B_lpm"]], p[["C_gran"]], p[["D_gran_mono"]], p[["E_mono"]], p[["F_rest"]], 
                        p[["e2f"]], p[["myc"]], p[["cebpe"]], ncol = 3, nrow=3)

#---------------------------------------------------------------------#

p <- lapply(sort(grep("neutral", unique(genes_cebpe$cluster_deRefined), value=TRUE)), function(x) {
  t.melt <- (melt(genes_cebpe[which(genes_cebpe$cluster_deRefined==x),c(31,32,33,35,43,44,45,46)]))
  #t.melt <- (melt(log(x[,c(11,12,13,15,16,24,25,26,27,28)]+1)))
  t.melt$class <- gsub("_.*", "", (t.melt$variable))
  t.melt$variable <- gsub("^[^\\_]*_", "", t.melt$variable)
  t.melt <- summarySE(t.melt, measurevar = "value", groupvars = c("class", "variable"))
  t.melt$class <- factor(t.melt$class, levels=c("gmpG", "epm", "lpm", "mm"))
  ggplot(t.melt, aes(class, value, color=variable)) +
    geom_errorbar(aes(ymin=value-se, ymax=value+se), width=.1, alpha=1) +
    geom_line(aes(group=factor(variable), color=factor(variable)), alpha=1) +
    geom_point(shape = 21, size = 2, fill="white") +
    scale_color_manual(values=c("#313695", "#a50026")) +
    theme_bw() + theme(legend.position="top")
})
names(p) <- sort(grep("neutral", unique(genes_cebpe$cluster_deRefined), value=TRUE))
gDEProfileNeutral <- ggarrange(plotlist = p, ncol=3, nrow=3)

```

##### analyze cebpe, e2f, lin54, nfyb binding enrichment at proximal and distal regions of interesting gene clusters
```{r}
# t.melt <- summarySE(melt(genes_cebpe[,c("cebpe_lpmmy", "cluster_de")]), measurevar = "value", groupvars = c("cluster_de", "variable"))
# p1 <- ggplot(t.melt, aes(cluster_de, value, color=variable)) +
#   geom_errorbar(aes(ymin=value-se, ymax=value+se), width=.1, alpha=1) +
#   geom_line(aes(group=factor(variable), color=factor(variable)), alpha=1) +
#   geom_point(shape = 21, size = 2, fill="white") +
#   #scale_color_manual(values=c("#313695", "#a50026")) +
#   theme(legend.position="top", axis.text.x = element_text(angle = 45, hjust = 1))  + ylab("proximal (TPM)") +
#   facet_wrap(~variable, scales="free") + theme_bw() +
#   geom_hline(yintercept = summary(genes_cebpe[which(genes_cebpe$cluster_de!="F_rest"),]$cebpe_lpmmy)[3] , color="black", linetype=2) + 
#   theme(legend.position="top")
# 
# test <- genes_cebpe[,c("cebpe_lpmmy_dis", "cluster_de", "enhancer_count")]
# test$cebpe_lpmmy_dis_norm <- test$cebpe_lpmmy_dis/test$enhancer_count
# test[is.na(test)] <- 0
# t.melt <- summarySE(melt(test[,c("cebpe_lpmmy_dis_norm", "cluster_de")]), measurevar = "value", groupvars = c("cluster_de", "variable"))
# p2 <- ggplot(t.melt, aes(cluster_de, value, color=variable)) +
#   geom_errorbar(aes(ymin=value-se, ymax=value+se), width=.1, alpha=1) +
#   geom_line(aes(group=factor(variable), color=factor(variable)), alpha=1) +
#   geom_point(shape = 21, size = 2, fill="white") +
#   #scale_color_manual(values=c("#313695", "#a50026")) +
#   theme(legend.position="top", axis.text.x = element_text(angle = 45, hjust = 1)) + ylab("distal per enhancer (TPM)") +
#   facet_wrap(~variable, scales="free") + theme_bw() +
#   geom_hline(yintercept = summary(genes_cebpe[which(genes_cebpe$cluster_de!="F_rest"),]$cebpe_lpmmy_dis)[3] , color="black", linetype=2) + 
#   theme(legend.position="top")

test <- genes_cebpe[,c("cebpe_lpmmy", "e2f_chip", "myc_chip", "cluster", "cluster_de", "cluster_deRefined")]
test <- test[which(test$cluster_deRefined!="F_rest-neutral"),]
test$cluster_de <- gsub("-.*", "", test$cluster_deRefined)
test$cluster_deRefined <- gsub(".*-", "", test$cluster_deRefined)
test <- test[which(test$cluster_de!="F_rest"),]

t.melt <- summarySE(test, measurevar = "e2f_chip", groupvars = c("cluster_de", "cluster_deRefined"))
t.melt <- t.melt[which(t.melt$cluster_deRefined!="neutral"),]
p1 <- ggplot(t.melt, aes(cluster_de, e2f_chip, color=cluster_deRefined)) +
    geom_errorbar(aes(ymin=e2f_chip-se, ymax=e2f_chip+se), width=.1, alpha=1) +
    geom_line(aes(group=factor(cluster_deRefined), color=factor(cluster_deRefined)), alpha=1) +
    geom_point(shape = 21, size = 2, fill="white") +
    scale_color_manual(values=c("#a50026", "#000000")) +
    theme_bw() + theme(legend.position="top")

t.melt <- summarySE(test, measurevar = "myc_chip", groupvars = c("cluster_de", "cluster_deRefined"))
t.melt <- t.melt[which(t.melt$cluster_deRefined!="neutral"),]
p2 <- ggplot(t.melt, aes(cluster_de, myc_chip, color=cluster_deRefined)) +
    geom_errorbar(aes(ymin=myc_chip-se, ymax=myc_chip+se), width=.1, alpha=1) +
    geom_line(aes(group=factor(cluster_deRefined), color=factor(cluster_deRefined)), alpha=1) +
    geom_point(shape = 21, size = 2, fill="white") +
    scale_color_manual(values=c("#a50026", "#000000")) +
    theme_bw() + theme(legend.position="top")

t.melt <- summarySE(test, measurevar = "cebpe_lpmmy", groupvars = c("cluster_de", "cluster_deRefined"))
p3 <- ggplot(t.melt, aes(cluster_de, cebpe_lpmmy, color=cluster_deRefined)) +
    geom_errorbar(aes(ymin=cebpe_lpmmy-se, ymax=cebpe_lpmmy+se), width=.1, alpha=1) +
    geom_line(aes(group=factor(cluster_deRefined), color=factor(cluster_deRefined)), alpha=1) +
    geom_point(shape = 21, size = 2, fill="white") +
    scale_color_manual(values=c("#a50026", "#000000")) +
    theme_bw() + theme(legend.position="top") + ylim(c(0,6))

# ggline(test[which(test$cebpe_lpmmy<6),], x="cluster_de", y="cebpe_lpmmy", add = c("mean_se"), color="cluster_deRefined",
#                  palette=c("#a50026", "#000000"), point.color="gray")

test <- genes_cebpe[,c("cebpe_lpmmy_dis", "cluster_de", "cluster_deRefined", "enhancer_count")]
test$cebpe_lpmmy_dis <- test$cebpe_lpmmy_dis/test$enhancer_count
test[is.na(test)] <- 0
test <- test[which(test$cluster_deRefined!="F_rest-neutral"),]
test$cluster_de <- gsub("-.*", "", test$cluster_deRefined)
test$cluster_deRefined <- gsub(".*-", "", test$cluster_deRefined)
test <- test[which(test$cluster_de!="F_rest"),]

t.melt <- summarySE(test, measurevar = "cebpe_lpmmy_dis", groupvars = c("cluster_de", "cluster_deRefined"))
p4 <- ggplot(t.melt, aes(cluster_de, cebpe_lpmmy_dis, color=cluster_deRefined)) +
    geom_errorbar(aes(ymin=cebpe_lpmmy_dis-se, ymax=cebpe_lpmmy_dis+se), width=.1, alpha=1) +
    geom_line(aes(group=factor(cluster_deRefined), color=factor(cluster_deRefined)), alpha=1) +
    geom_point(shape = 21, size = 2, fill="white") +
    scale_color_manual(values=c("#a50026", "#000000")) +
    theme_bw() + theme(legend.position="top")

# df_motif <- read.table("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/gene_clusters/motifDynAna/DENOVO_motif_dynamics.pdf.txt", header=T)
# ggplot(melt(df_motif[,c(1:6)]), aes(variable, value)) + geom_bar(stat="identity") + facet_wrap(~Row.names, scales="free") + theme_bw() + 
#   theme(legend.position="none")

# p4 <- ggplot(melt(genes_cebpe[grep("^E2f1|Myc$|Lin54$|Nfyb$", genes_cebpe$gene),c(4,28:29,31:37)]), aes(variable, (value), fill=variable)) + geom_bar(stat="identity") + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + facet_wrap(~gene, scales="free") + theme_bw() +
#   scale_fill_manual(values=colorRampPalette(brewer.pal(9, "RdYlGn"), bias=1)(9)) + theme(legend.position="none")

#ggplot(melt(test[which(test$cluster=="4" & test$e2f_bound=="Y"),c(100,81,83,85,87)]), aes(V10, value)) + geom_boxplot() + facet_grid(.~variable)

df.melt <- melt(genes_cebpe[grep("A_epm|B_lpm", genes_cebpe$cluster_deRefined),c("cebpe_lpmmy", "cluster_deRefined")])
df.melt$class <- gsub(".*-", "", df.melt$cluster_deRefined)
df.melt$cluster_deRefined <- gsub("-.*", "", df.melt$cluster_deRefined)
p5 <- ggplot(df.melt, aes(class, log(value), fill=class)) + geom_boxplot(notch = T, notchwidth = 0.6) + 
  facet_wrap(~cluster_deRefined, scales="free") + theme_bw() + scale_fill_manual(values = c("#a6cee3", "#a6cee3"))
wilcox.test(genes_cebpe[which(genes_cebpe$cluster_deRefined=="A_epm-de"),]$cebpe_lpmmy, genes_cebpe[which(genes_cebpe$cluster_deRefined=="A_epm-neutral"),]$cebpe_lpmmy, alternative = "g")$p.value
sprintf("%E", wilcox.test(genes_cebpe[which(genes_cebpe$cluster_deRefined=="B_lpm-de"),]$cebpe_lpmmy, genes_cebpe[which(genes_cebpe$cluster_deRefined=="B_lpm-neutral"),]$cebpe_lpmmy, alternative = "g")$p.value)

df.melt <- (genes_cebpe[grep("A_epm", genes_cebpe$cluster_deRefined),c("cluster_deRefined", "e2f_bound")])
p6 <- freqPlot(melt(table(df.melt)), "density", "e2f binding") + theme_bw() + xlab("") + theme(legend.position = "top") 

df.melt <- (genes_cebpe[grep("B_lpm", genes_cebpe$cluster_deRefined),c("cluster_deRefined", "e2f_bound")])
p7 <- freqPlot(melt(table(df.melt)), "density", "e2f binding") + theme_bw() + xlab("") + theme(legend.position = "top")

gDEWithCebpe <- ggarrange(ggarrange(p1, p2, p3, p4, ncol=4, nrow=1), 
                          ggarrange(p5, ggarrange(p6, p7, ncol=2, nrow=1)), 
                          ncol=1, nrow=2)

#---------------------------------------------------------------------#
## sanity check (is cebpe, myc and e2f1 enriched at G1_S genes)
# df.melt <- melt(genes_cebpe[which(genes_cebpe$funcClass=="G1_S" & (genes_cebpe$cluster_deRefined=="A_epm-de" | genes_cebpe$cluster_deRefined=="A_epm-neutral")),c("cebpe_lpmmy", "cluster_deRefined", "funcClass", "myc_bound", "e2f_bound")])
# df.melt$class <- gsub(".*-", "", df.melt$cluster_deRefined)
# df.melt$cluster_deRefined <- gsub("-.*", "", df.melt$cluster_deRefined)
# p1 <- ggboxplot(df.melt, x="class", y="value", fill="class", notch = F, notchwidth = 0.6) +
#   #facet_wrap(~cluster_deRefined, scales="free") +
#   theme_bw() +
#   scale_fill_manual(values = c("#a6cee3", "#a6cee3")) +
#   stat_compare_means(comparisons = list(c("de", "neutral")), method.args = list(alternative = "two.sided")) +
#   theme(legend.position = "none") + ylab("CEBPE signal")

# df.melt <- melt(genes_cebpe[which((genes_cebpe$funcClass=="G2_M" & genes_cebpe$cluster_deRefined=="B_lpm-de") | (genes_cebpe$cluster_deRefined=="B_lpm-neutral")),c("cebpe_lpmmy", "cluster_deRefined", "funcClass", "myc_bound", "e2f_bound")])
df.melt <- melt(genes_cebpe[which((genes_cebpe$funcClass=="G1_S" & genes_cebpe$cluster_deRefined=="A_epm-de") | (genes_cebpe$cluster_deRefined=="A_epm-neutral")),c("cebpe_lpmmy", "cluster_deRefined", "funcClass", "myc_bound", "e2f_bound")])
df.melt$class <- gsub(".*-", "", df.melt$cluster_deRefined)
df.melt$cluster_deRefined <- gsub("-.*", "", df.melt$cluster_deRefined)
table(df.melt$class)
p1 <- ggboxplot(df.melt, x="class", y="value", fill="class", notch = F, notchwidth = 0.6) +
  #facet_wrap(~cluster_deRefined, scales="free") +
  theme_bw() +
  scale_fill_manual(values = c("#a6cee3", "#a6cee3")) +
  stat_compare_means(comparisons = list(c("de", "neutral")), method.args = list(alternative = "two.sided")) +
  theme(legend.position = "none") + ylab("CEBPE signal") + xlab("")

p2 <- freqPlot(melt(table(df.melt[,c("class", "myc_bound")])), "density", "Myc bound") + theme_bw() + theme(legend.position = "none") + xlab("")
p3 <- freqPlot(melt(table(df.melt[,c("class", "e2f_bound")])), "density", "E2F1 bound") +  theme_bw() + theme(legend.position = "none") + xlab("")
gDEWithCebpeG1S <- ggarrange(p1, p2, p3, ncol=3)
gDEWithCebpeG1S

#---------------------------------------------------------------------#

PVALUE_CEBPE <- cbind(p.adjust(lapply(unique(genes_cebpe$cluster_deRefined)[!grepl("F_rest|-neutral", unique(genes_cebpe$cluster_deRefined))],
                                      function(x) {
                                        wilcox.test(genes_cebpe[which(genes_cebpe$cluster_deRefined==x),]$cebpe_lpmmy, genes_cebpe[which(genes_cebpe$de_count>0),]$cebpe_lpmmy, alternative = "g")$p.value
                                        }
                                      ), method="BH"
                               ),
                      p.adjust(lapply(unique(genes_cebpe$cluster_deRefined)[!grepl("F_rest|-de", unique(genes_cebpe$cluster_deRefined))],
                                      function(x) {
                                        wilcox.test(genes_cebpe[which(genes_cebpe$cluster_deRefined==x),]$cebpe_lpmmy, genes_cebpe[which(genes_cebpe$de_count>0),]$cebpe_lpmmy, alternative = "g")$p.value
                                        }
                                      ), method="BH"
                               ),
                      p.adjust(lapply(unique(genes_cebpe$cluster_deRefined)[!grepl("F_rest|-neutral", unique(genes_cebpe$cluster_deRefined))],
                                      function(x) {
                                        wilcox.test(genes_cebpe[which(genes_cebpe$cluster_deRefined==x),]$cebpe_lpmmy_dis, genes_cebpe[which(genes_cebpe$de_count>0),]$cebpe_lpmmy_dis, alternative = "g")$p.value
                                        }
                                      ), method="BH"
                               ),
                      p.adjust(lapply(unique(genes_cebpe$cluster_deRefined)[!grepl("F_rest|-de", unique(genes_cebpe$cluster_deRefined))],
                                      function(x) {
                                        wilcox.test(genes_cebpe[which(genes_cebpe$cluster_deRefined==x),]$cebpe_lpmmy_dis, genes_cebpe[which(genes_cebpe$de_count>0),]$cebpe_lpmmy_dis, alternative = "g")$p.value
                                        }
                                      ), method="BH"
                               ))
row.names(PVALUE_CEBPE) <- unique(genes_cebpe$cluster_de)[!grepl("F_rest", unique(genes_cebpe$cluster_de))]
colnames(PVALUE_CEBPE) <- c("promoter_de", "promoter_neutral", "enhancer_de", "enhancer_neutral")

# median CEBPE levels at promoters
#summary(genes_cebpe[which(genes_cebpe$de_count>0),]$cebpe_lpmmy)
# median CEBPE levels at enhancers (normalized)
#summary(genes_cebpe[which(genes_cebpe$de_count>0),]$cebpe_lpmmy_dis/genes_cebpe[which(genes_cebpe$de_count>0),]$enhancer_count)

#---------------------------------------------------------------------#
p <- list()
p[["cebpe_promoter"]] <- matrix2pheatmap(-log10(PVALUE_CEBPE[,c(1,2)]), scale = "none", bias = 2.5)[[4]]
p[["cebpe_enhancer"]] <- matrix2pheatmap(-log10(PVALUE_CEBPE[,c(3,4)]), scale = "none", bias = 4)[[4]]

p[["e2f1"]] <- freqPlot(melt(table(as.data.frame(cbind(genes_cebpe$cluster_de, ifelse(genes_cebpe$motifScore_E2f1 > 0, "Y", "N"))))), "density", "E2f1") + 
  theme_bw() + xlab("") + theme(legend.position="top")
p[["myc"]] <- freqPlot(melt(table(as.data.frame(cbind(genes_cebpe$cluster_de, ifelse(genes_cebpe$motifScore_Myc > 0, "Y", "N"))))), "density", "Myc") + 
  theme_bw() + xlab("") + theme(legend.position="top")
p[["lin54"]] <- freqPlot(melt(table(as.data.frame(cbind(genes_cebpe$cluster_de, ifelse(genes_cebpe$motifScore_Lin54 > 0, "Y", "N"))))), "density", "Lin54") + 
  theme_bw() + xlab("") + theme(legend.position="top")
p[["nfyb"]] <- freqPlot(melt(table(as.data.frame(cbind(genes_cebpe$cluster_de, ifelse(genes_cebpe$motifScore_Nfyb > 0, "Y", "N"))))), "density", "Nfyb") + 
  theme_bw() + xlab("") + theme(legend.position="top")

gDEWithCebpeEnrich <- ggarrange(arrangeGrob(grobs=p,ncol=2))

#---------------------------------------------------------------------#

METHOD="glm"

## classification model
p <- lapply(c("A_epm", "B_lpm"), function(x) {
  set.seed(6)
  mat <- genes_cebpe
  mat$cebpe_lpmmy <- ifelse(mat$cebpe_lpmmy > THRESHOLD, 1, 0)
  mat$pu1_pregm <- ifelse(mat$pu1_pregm > summary(genes_cebpe[which(genes_cebpe$de_count>0),]$pu1_pregm)[3], 1, 0)
  mat$cebpa_gmp <- ifelse(mat$cebpa_gmp > summary(genes_cebpe[which(genes_cebpe$de_count>0),]$cebpa_gmp)[3], 1, 0)
  mat$myc <- ifelse(mat$myc > 0, 1, 0)
  mat$e2f <- ifelse(mat$e2f > 0, 1, 0)
  mat <- mat[grep(x, mat$cluster_deRefined),c("cluster_de", "cebpe_lpmmy", "myc", "e2f", "pu1_pregm", "cebpa_gmp")]
  mat$cluster_de <- as.factor(mat$cluster_de)
  table(mat$cluster_de)
  model <- train(cluster_de~., data=mat, method=METHOD, preProcess=c("center", "scale"),
                   trControl=trainControl(method="repeatedcv", number=10, repeats=3))
  importance <- varImp(model, scale=F)
  #print(importance)
  plot(importance, main=sprintf("classification (de vs neutral): %s", x))
  })

## regression model
set.seed(6)
mat <- genes_cebpe[grep("A_epm", genes_cebpe$cluster_de),c("log2FC_epm", "cebpe_lpmmy", "myc", "e2f", "pu1_pregm", "cebpa_gmp")]
mat <- mat[!is.na(mat$log2FC_epm),]
lmFit <- train(log2FC_epm~., data = mat, method = METHOD, preProcess=c("center", "scale"),
               trControl=trainControl(method="repeatedcv", number=10, repeats=3))
importance <- varImp(lmFit, scale=F)
#print(importance)
p[[3]] <- plot(importance, main="regression (log2FC): A_epm")

set.seed(6)
mat <- genes_cebpe[grep("B_lpm", genes_cebpe$cluster_de),c("log2FC_lpm", "cebpe_lpmmy", "myc", "e2f", "pu1_pregm", "cebpa_gmp")]
mat <- mat[!is.na(mat$log2FC_lpm),]
lmFit <- train(log2FC_lpm~., data = mat, method = METHOD, preProcess=c("center", "scale"),
               trControl=trainControl(method="repeatedcv", number=10, repeats=3))
importance <- varImp(lmFit, scale=F)
#print(importance)
p[[4]] <- plot(importance, main="regression (log2FC): B_lpm")

gDEWithTFRank <- ggarrange(arrangeGrob(grobs=p, ncol=2))

#---------------------------------------------------------------------#
#^Chd4$|^H3f3a$|^H2afz$|^Ezh2$|^Smarca4$|^Tet2$|^Rb1$|^Cdk12$|^Dnmt3a$|^Med1$|^Ep400$|^Ep300$|^Ppp2ca$
#enhancer candidates: ^Pik3c2b$|^Cpne3$|^Zdhhc18$|^Zcwpw1$|^Lrrc28$|^Igf1r$|^Tgm4$|^Traf4$|^Tst$|^Igsf5$
df.melt <- melt(expr_data_norm_with_replicates[grep("^Cebpa$|^Cebpe$|^E2f1$|^E2f7$|^E2f8$|^Lin54$|^Myc$|^Nfyb$|^Rb1$", expr_data_norm_with_replicates$GeneID),
                               c("GeneID",
                                 grep("lsk_.*wt", colnames(expr_data_norm_with_replicates), value = T),
                                 grep("pregm_.*wt", colnames(expr_data_norm_with_replicates), value = T),
                                 grep("gmp_.*wt|gmpG.*wt", colnames(expr_data_norm_with_replicates), value = T),
                                 grep("gmpG.*wt", colnames(expr_data_norm_with_replicates), value = T),
                                 grep("epm_.*wt", colnames(expr_data_norm_with_replicates), value = T),
                                 grep("lpm_.*wt", colnames(expr_data_norm_with_replicates), value = T),
                                 grep("my_.*wt", colnames(expr_data_norm_with_replicates), value = T),
                                 grep("mm_.*wt", colnames(expr_data_norm_with_replicates), value = T),
                                 grep("bc_.*wt", colnames(expr_data_norm_with_replicates), value = T),
                                 grep("pmn_.*wt", colnames(expr_data_norm_with_replicates), value = T),
                                 grep("gmpG_.*ko", colnames(expr_data_norm_with_replicates), value = T),
                                 grep("epm_.*ko", colnames(expr_data_norm_with_replicates), value = T),
                                 grep("lpm_.*ko", colnames(expr_data_norm_with_replicates), value = T),
                                 grep("mm_.*ko", colnames(expr_data_norm_with_replicates), value = T)
                                 )])
LEVELS <- unique(gsub("_[Rr]ep.*|_cebpa", "", df.melt$variable))
df.melt$variable <- gsub("_[Rr]ep.*|_cebpa", "", df.melt$variable)
df.melt$value <- df.melt$value
df.melt <- summarySE(df.melt, measurevar = "value", groupvars = c("GeneID", "variable"))
df.melt$variable <- factor(df.melt$variable, levels=LEVELS, ordered = T)

gXExprProfile <- ggplot(df.melt, aes(variable, value)) +
  geom_errorbar(aes(ymin=value-se, ymax=value+se), width=.1, alpha=1) +
  geom_point(shape = 21, size = 2, aes(fill=GeneID)) +
  #scale_fill_manual(values = c(rep("#000000",12))) +
  geom_line(aes(group=1, color=GeneID), alpha=1) +
  theme_classic() + theme(legend.position="top", axis.text.x = element_text(angle = 45, hjust = 1)) + ylab("Expr.") +
  facet_wrap(~GeneID, scales="free")

gXExprProfile

df.melt <- melt(expr_data[grep("^Myc$|^Mxd1$|^Max$", expr_data$GeneID),
                                               c("GeneID",
                                                 grep("gmpG.*wt", colnames(expr_data), value = T),
                                                 grep("epm_.*wt", colnames(expr_data), value = T),
                                                 grep("lpm_.*wt", colnames(expr_data), value = T),
                                                 grep("mm_.*wt", colnames(expr_data), value = T),
                                                 grep("gmpG_.*ko", colnames(expr_data), value = T),
                                                 grep("epm_.*ko", colnames(expr_data), value = T),
                                                 grep("lpm_.*ko", colnames(expr_data), value = T),
                                                 grep("mm_.*ko", colnames(expr_data), value = T)
                                               )])

df.melt$class <- gsub("_.*", "", (df.melt$variable))
df.melt$variable <- gsub("^.*_", "", gsub("_[Rr]ep.*", "", df.melt$variable))
df.melt <- summarySE(df.melt, measurevar = "value", groupvars = c("class", "variable", "GeneID"))
df.melt$class <- factor(df.melt$class, levels=c("gmpG", "epm", "lpm", "mm"))
gXExprProfileMyc <- ggplot(df.melt, aes(class, value, color=variable)) +
  geom_errorbar(aes(ymin=value-se, ymax=value+se), width=.1, alpha=1) +
  geom_line(aes(group=factor(variable), color=factor(variable)), alpha=1) +
  geom_point(shape = 21, size = 2, fill="white") +
  scale_color_manual(values=c("#313695", "#a50026")) +
  theme_bw() + theme(legend.position="top") + 
  facet_wrap(~GeneID, scales="free")

gXExprProfileMyc

df.melt <- melt(expr_data_norm_with_replicates[grep("^Nfyb$|^Lin54$", expr_data_norm_with_replicates$GeneID),
                               c("GeneID",
                                 grep("lpm_.*wt", colnames(expr_data_norm_with_replicates), value = T),
                                 grep("lpm_.*ko", colnames(expr_data_norm_with_replicates), value = T)
                                 )])
LEVELS <- unique(gsub("_[Rr]ep.*|_cebpa", "", df.melt$variable))
df.melt$variable <- gsub("_[Rr]ep.*|_cebpa", "", df.melt$variable)
df.melt$value <- df.melt$value

gXExprProfileNfybLin54 <- ggboxplot(df.melt, x="variable", y="value", notch=F, notchwdth=0.6, facet.by="GeneID", fill="GeneID", outlier.shape = NA) +
  xlab("") + ylab("Gene expr. (norm)") +
  scale_fill_manual(values=c("#8da0cb", "#66c2a5")) + 
  stat_compare_means(comparisons = list(c("lpm_wt", "lpm_ko")), method.args = list(alternative = "two.sided")) +
  theme(legend.position = "top")

res_lpm[grep("^Nfyb$|^Lin54$", res_lpm$gene),]
genes_cebpe[grep("^Rb1$|^E2f7$|^E2f8$|^Cdkn1b$", genes_cebpe$gene),c(4,84:124)]
pdf("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/pdf/myc_mad_max.pdf")
barplot(genes_cebpe[grep("^Mxd1$|^Myc$|^Max$", genes_cebpe$gene),c(4,84:91)]$log2FC_mm, 
        names.arg=genes_cebpe[grep("^Mxd1$|^Myc$|^Max$", genes_cebpe$gene),c(4,84:91)]$gene, las=2)
res_mm[grep("^Mxd1$|^Myc$|^Max$", res_mm$gene),]
sprintf("%E", res_mm[grep("^Mxd1$|^Myc$|^Max$", res_mm$gene),]$padj)

pdf("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/pdf/genes_expr_se_points_plots.pdf")
df.melt <- melt(expr_data_norm_with_replicates[grep("^Cebpa$|^Cebpe$|^E2f1$|^E2f7$|^E2f8$|^Lin54$|^Myc$|^Nfyb$|^Rb1$", expr_data_norm_with_replicates$GeneID),
                               c("GeneID",
                                 grep("lsk_.*wt", colnames(expr_data_norm_with_replicates), value = T),
                                 grep("pregm_.*wt", colnames(expr_data_norm_with_replicates), value = T),
                                 grep("gmp_.*wt|gmpG.*wt", colnames(expr_data_norm_with_replicates), value = T),
                                 grep("gmpG.*wt", colnames(expr_data_norm_with_replicates), value = T),
                                 grep("epm_.*wt", colnames(expr_data_norm_with_replicates), value = T),
                                 grep("lpm_.*wt", colnames(expr_data_norm_with_replicates), value = T),
                                 grep("my_.*wt", colnames(expr_data_norm_with_replicates), value = T),
                                 grep("mm_.*wt", colnames(expr_data_norm_with_replicates), value = T),
                                 grep("bc_.*wt", colnames(expr_data_norm_with_replicates), value = T),
                                 grep("pmn_.*wt", colnames(expr_data_norm_with_replicates), value = T),
                                 grep("gmpG_.*ko", colnames(expr_data_norm_with_replicates), value = T),
                                 grep("epm_.*ko", colnames(expr_data_norm_with_replicates), value = T),
                                 grep("lpm_.*ko", colnames(expr_data_norm_with_replicates), value = T),
                                 grep("mm_.*ko", colnames(expr_data_norm_with_replicates), value = T)
                                 )])
LEVELS <- unique(gsub("_[Rr]ep.*|_cebpa", "", df.melt$variable))
df.melt$variable <- gsub("_[Rr]ep.*|_cebpa", "", df.melt$variable)
df.melt$value <- df.melt$value
df.melt$variable <- factor(df.melt$variable, levels=LEVELS, ordered = T)
ggarrange(ggline(df.melt[which(df.melt$GeneID=="Cebpa" | df.melt$GeneID=="Cebpe"),], x="variable", y="value", add = c("mean_se", "jitter"), color="GeneID",
                 palette=c("#2a7ab9", "#d52221"), point.color="gray"),
          ggline(df.melt[which(df.melt$GeneID=="Myc"),], x="variable", y="value", add = c("mean_se", "jitter"), color="GeneID", 
                 palette=c("#c77cff"), point.color="gray"),
          ggline(df.melt[which(df.melt$GeneID=="E2f1"),], x="variable", y="value", add = c("mean_se", "jitter"), color="GeneID", 
                 palette=c("#3caa3c"), point.color="gray"), nrow=2, ncol=2)

df.melt <- melt(expr_data[grep("^Myc$|^Mxd1$|^Max$", expr_data$GeneID),
                                               c("GeneID",
                                                 grep("gmpG.*wt", colnames(expr_data), value = T),
                                                 grep("epm_.*wt", colnames(expr_data), value = T),
                                                 grep("lpm_.*wt", colnames(expr_data), value = T),
                                                 grep("mm_.*wt", colnames(expr_data), value = T),
                                                 grep("gmpG_.*ko", colnames(expr_data), value = T),
                                                 grep("epm_.*ko", colnames(expr_data), value = T),
                                                 grep("lpm_.*ko", colnames(expr_data), value = T),
                                                 grep("mm_.*ko", colnames(expr_data), value = T)
                                               )])

df.melt$class <- gsub("_.*", "", (df.melt$variable))
df.melt$variable <- gsub("^.*_", "", gsub("_[Rr]ep.*", "", df.melt$variable))
df.melt$class <- factor(df.melt$class, levels=c("gmpG", "epm", "lpm", "mm"))
ggline(df.melt[which(df.melt$GeneID=="Myc"),], x="class", y="value", add = c("mean_se", "jitter"), color="variable", palette=c("#2a7ab9", "#d52221"))
# table(df.melt[which(df.melt$GeneID=="Myc"),c("variable", "class")])

df.melt <- melt(expr_data_norm_with_replicates[grep("^Myc$|^Mxd1$|^Max$|^Nfyb$|^Lin54$", expr_data_norm_with_replicates$GeneID),
                               c("GeneID",
                                 grep("lsk_.*wt", colnames(expr_data_norm_with_replicates), value = T),
                                 grep("pregm_.*wt", colnames(expr_data_norm_with_replicates), value = T),
                                 grep("gmp_.*wt|gmpG.*wt", colnames(expr_data_norm_with_replicates), value = T),
                                 grep("gmpG.*wt", colnames(expr_data_norm_with_replicates), value = T),
                                 grep("epm_.*wt", colnames(expr_data_norm_with_replicates), value = T),
                                 grep("lpm_.*wt", colnames(expr_data_norm_with_replicates), value = T),
                                 grep("my_.*wt", colnames(expr_data_norm_with_replicates), value = T),
                                 grep("mm_.*wt", colnames(expr_data_norm_with_replicates), value = T),
                                 grep("bc_.*wt", colnames(expr_data_norm_with_replicates), value = T),
                                 grep("pmn_.*wt", colnames(expr_data_norm_with_replicates), value = T)
                                 )])
LEVELS <- unique(gsub("_[Rr]ep.*|_cebpa", "", df.melt$variable))
df.melt$variable <- gsub("_[Rr]ep.*|_cebpa", "", df.melt$variable)
df.melt$value <- df.melt$value
df.melt$variable <- factor(df.melt$variable, levels=LEVELS, ordered = T)
ggarrange(ggline(df.melt[which(df.melt$GeneID=="Myc"),], x="variable", y="value", add = c("mean_se", "jitter"), color="GeneID"),
          ggline(df.melt[which(df.melt$GeneID=="Mxd1"),], x="variable", y="value", add = c("mean_se", "jitter"), color="GeneID"),
          ggline(df.melt[which(df.melt$GeneID=="Max"),], x="variable", y="value", add = c("mean_se", "jitter"), color="GeneID"),
          ggline(df.melt[which(df.melt$GeneID=="Nfyb" | df.melt$GeneID=="Lin54"),], x="variable", y="value", add = c("mean_se", "jitter"), color="GeneID",
                 palette=c("#00b9e3", "#db72fb"), point.color="gray"), nrow=2, ncol=2)

dev.off()
```

#### plot results showing analysis on gene expression and TF binding profile for interesting gene clusters
```{r}
pdf("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/pdf/de_genes_analysis.pdf")
gDE
gDEProfile
gDEProfileNeutral
gDEWithCebpe
gDEWithCebpeEnrich
gDEWithTFRank
gXExprProfile
gXExprProfileMyc
gXExprProfileNfybLin54
# gDEWithTF
# gDEWithTFRefined
dev.off()
```

#### refined analysis on A_lpm gene class (MYC TF binding and cell cycle analysis)
```{r}
dev.off()
pdf("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/pdf/class_aEPM_analysis.pdf")

# relationship between TF co-binding and logFC in gene expression
test <- genes_cebpe[grep("A_epm", genes_cebpe$cluster_deRefined),]
t <- cbind(ifelse(test$cebpe_lpmmy > THRESHOLD, "cebpe", "bkg"),
           ifelse(test$myc>0, "myc", "bkg"),
           ifelse(test$e2f>0, "e2f1", "bkg"))
                  

test$peak_class <- paste(t[,1], t[,2], t[,3], sep="_")
table(test$peak_class)

test$motifScore_Cebpe_binary <- ifelse(test$motifScore_Cebpe>0, "Y", "N")

PVALUE_EPM <- as.data.frame(cbind(
  names(table(test$peak_class)),
  table(test$peak_class),
  t(as.data.frame(lapply(names(table(test$peak_class)), function(x) 
    wilcox.test(test[which(test$peak_class==x),c("lpm_ko_norm")], test[which(test$peak_class==x),c("lpm_wt_norm")], alternative = "g")$p.value))), 
  t(as.data.frame(lapply(names(table(test$peak_class)), function(x) 
    mean(!is.na(test[which(test$peak_class==x),c("log2FC_mm")]))))),
  t(as.data.frame(lapply(names(table(test$peak_class)), function(x) 
    median(test[which(test$peak_class==x),c("cebpe_lpmmy")])))),
  t(as.data.frame(lapply(names(table(test$peak_class)), function(x) 
    median(test[which(test$peak_class==x),c("e2f_chip")])))),
  t(as.data.frame(lapply(names(table(test$peak_class)), function(x) 
    median(test[which(test$peak_class==x),c("myc_chip")])))),
  t(as.data.frame(lapply(names(table(test$peak_class)), function(x) 
    mean(test[which(test$peak_class==x),c("numTagsNfyb")])))),
  t(as.data.frame(lapply(names(table(test$peak_class)), function(x) 
    mean(test[which(test$peak_class==x),c("motifScore_Cebpe")])))),
  t(as.data.frame(lapply(names(table(test$peak_class)), function(x) 
    mean(test[which(test$peak_class==x),c("motifScore_E2f1")])))),
  t(as.data.frame(lapply(names(table(test$peak_class)), function(x) 
    mean(test[which(test$peak_class==x),c("motifScore_Myc")])))),
  t(as.data.frame(lapply(names(table(test$peak_class)), function(x) 
    mean(test[which(test$peak_class==x),c("motifScore_Nfyb")])))),
  t(as.data.frame(lapply(names(table(test$peak_class)), function(x) 
    mean(test[which(test$peak_class==x),c("motifScore_Lin54")])))),
  t(as.data.frame(lapply(names(table(test$peak_class)), function(x) 
    sprintf("%0.1f", (nrow(test[which(test$peak_class==x & test$motifScore_Cebpe>0),])*100)/nrow(test[which(test$peak_class==x),])))))
  ))

colnames(PVALUE_EPM) <- c("tf", "gene_count", "sig", "medianLog2FC", "cebpe_lpmmy", "e2f_chip", "myc_chip", "numTagsNfyb", 
                         "motifScore_Cebpe", "motifScore_E2f1", "motifScore_Myc", "motifScore_Nfyb", "motifScore_Lin54", "per_genes_with_cebpe_motif")
PVALUE_EPM[,c(2:ncol(PVALUE_EPM))] <- apply(PVALUE_EPM[,c(2:ncol(PVALUE_EPM))], 2, function(x) as.numeric(as.character(x)))
#PVALUE_EPM <- PVALUE_EPM[grepl("myc", PVALUE_EPM$tf),]

PVALUE_EPM$tf <- factor(PVALUE_EPM$tf, levels = PVALUE_EPM[order(PVALUE_EPM$sig),]$tf, ordered = T)
PVALUE_EPM$sig <- p.adjust(PVALUE_EPM$sig, "BH")

PVALUE_EPM[levels(PVALUE_EPM$tf),]$tf
PVALUE_EPM[levels(PVALUE_EPM$tf),]$gene_count

# ggplot(PVALUE_EPM, aes(tf, -log10(sig))) + geom_bar(stat="identity") + theme_bw() +
#   #theme(axis.text.x = element_blank()) + 
#   geom_hline(yintercept = -log10(0.05), lty=2)
# plot(Venn(list(myc=test[which(test$myc>0),]$gene, e2f1=test[which(test$e2f>0),]$gene)))
# freqPlot(melt(table(genes_cebpe[,c("cluster_de", "myc_bound")])), "density") + theme_bw()

test$myc_class <- "neutral"
test[which(test$cluster_deRefined!="A_epm-neutral"),]$myc_class <- "de"
test[which(test$cluster_deRefined!="A_epm-neutral" & (test$cell_cycle=="cell_cycle")),]$myc_class <- "cell_cycle"
#test[which(test$cluster_deRefined!="A_epm-neutral" & (test$cell_cycle=="cell_cycle" | grepl("Apoptosis_TP53", test$funcClass))),]$myc_class <- "cell_cycle"
table(test$myc_class)
test$myc_class <- factor(test$myc_class, levels=c("neutral", "de", "cell_cycle"), ordered=T)
test <- test[order(-test$myc_chip),]

# p1 <- ggboxplot(test[,c("myc_target", "log2FC_mm")], x="myc_target", y="log2FC_mm", notch=T, notchwdth=0.6, outlier.shape = NA, yscale="none", fill="myc_target") +
#   xlab("") + ylab("Gene expr. FC (mm)") +
#   scale_fill_manual(values=c("#8da0cb", "#66c2a5", "#fc8d62")) +
#   stat_compare_means(comparisons = list(c("myc_target", "not_myc_target")), method.args = list(alternative = "g")) +
#   theme(legend.position = "top")

p1 <- ggboxplot(test[,c("myc_class", "log2FC_mm")], x="myc_class", y="log2FC_mm", notch=T, notchwdth=0.6, outlier.shape = NA, yscale="none", fill="myc_class") +
  xlab("") + ylab("Gene expr. FC (mm)") +
  scale_fill_manual(values=c("#8da0cb", "#66c2a5", "#fc8d62")) +
  stat_compare_means(comparisons = list(c("cell_cycle", "de"), c("de", "neutral"), c("cell_cycle", "neutral")), 
                     method.args = list(alternative = "g")) +
  theme(legend.position = "top")

p2 <- ggboxplot(test[,c("myc_class", "myc_chip")], x="myc_class", y="myc_chip", notch=T, notchwdth=0.6, outlier.shape = NA, yscale="log2", fill="myc_class") +
  xlab("") + ylab("MYC Signal") +
  scale_fill_manual(values=c("#8da0cb", "#66c2a5", "#fc8d62")) +
  stat_compare_means(comparisons = list(c("cell_cycle", "de"), c("de", "neutral"), c("cell_cycle", "neutral")), 
                     method.args = list(alternative = "g")) +
  theme(legend.position = "top")

p3 <- ggboxplot(test[,c("myc_class", "e2f_chip")], x="myc_class", y="e2f_chip", notch=T, notchwdth=0.6, outlier.shape = NA, yscale="log2", fill="myc_class") +
  xlab("") + ylab("E2F Signal") +
  scale_fill_manual(values=c("#8da0cb", "#66c2a5", "#fc8d62")) +
  stat_compare_means(comparisons = list(c("cell_cycle", "de"), c("de", "neutral"), c("cell_cycle", "neutral")), 
                     method.args = list(alternative = "g")) +
  theme(legend.position = "top")

p4 <- ggboxplot(test[,c("myc_class", "cebpe_lpmmy")], x="myc_class", y="cebpe_lpmmy", notch=T, notchwdth=0.6, outlier.shape = NA, yscale="log2", fill="myc_class") +
  xlab("") + ylab("CEBPE Signal") +
  scale_fill_manual(values=c("#8da0cb", "#66c2a5", "#fc8d62")) +
  stat_compare_means(comparisons = list(c("cell_cycle", "de"), c("de", "neutral"), c("cell_cycle", "neutral")), 
                     method.args = list(alternative = "g")) +
  theme(legend.position = "top")

# t <- test[which(test$myc_class=="cell_cycle"),]
# t[order(-t$cebpe_lpmmy),]$gene

ggarrange(p1, p2, p3, p4, nrow=2, ncol=2)

mat <- merge(res_mm, test[which(test$myc_class=="cell_cycle" | (test$funcClass=="Apoptosis_TP53" & test$myc_bound=="Y")),c("gene", "myc", "myc_chip", "myc_bound", "funcClass", "cell_cycle", "log2FC_lpm", "log2FC_mm", "myc_target", "e2f_chip", "cebpe_lpmmy")], by.x="gene", by.y="gene")
colVals <- rep("#8da0cb", nrow(mat))
names(colVals) <- rep("bkg", nrow(mat))
colVals[which(mat$myc_target=="myc_target")] <- "#fc8d62"
names(colVals)[which(mat$myc_target=="myc_target")] <- "myc-target/bound"
colVals[which(mat$myc > 0)] <- "#fc8d62"
names(colVals)[which(mat$myc > 0)] <- "myc-target/bound"
table(names(colVals))
table((colVals))

EnhancedVolcano::EnhancedVolcano(mat, lab=mat$gene, x='log2FoldChange', y='padj', 
                                 pCutoff = 1,
                                 FCcutoff = 0,
                                 selectLab = mat[which(mat$myc_target=="myc_target" | mat$myc > 0),]$gene,
                                 drawConnectors = TRUE,
                                 widthConnectors = 0.2,
                                 colConnectors = '#000000',
                                 colCustom = colVals,
                                 transcriptPointSize=10,
                                 title="",
                                 subtitle = "")

mat <- test[which(test$myc_class=="cell_cycle" | (test$funcClass=="Apoptosis_TP53" & test$myc_bound=="Y")),c("gene", "myc", "myc_chip", "myc_bound", "funcClass", "cell_cycle", "log2FC_lpm", "log2FC_mm", "myc_target")]
row.names(mat) <- mat$gene
mat$gene <- factor(mat$gene, levels=mat$gene, ordered = T)
mat <- mat[order(-mat$log2FC_mm),]

matrix2pheatmap(mat[,c("log2FC_lpm", "log2FC_mm")], bias=3)
matrix2pheatmap(cbind(ifelse(mat$myc_target=="myc_target", 1, 0), (ifelse(mat$myc>0, 1, 0))))
ggplot(mat, aes(gene, myc_chip)) + geom_bar(stat="identity") + theme_bw() + coord_flip()

par(mfrow=c(2,2))
test <- genes_cebpe[which(genes_cebpe$funcClass=="G1_S"),]
#test <- genes_cebpe[grep("A_epm", genes_cebpe$cluster_deRefined),]
table(test$cluster_deRefined)
smoothScatter(log(test$e2f_chip), log(test$myc_chip), xlab="E2F1 signal (log)", ylab="MYC signal (log)", main=sprintf("Spearman corr: %0.2f",cor(test$myc_chip, test$e2f_chip, method="spearman")), pch=19, cex=.2, useRaster=T, colramp=colorRampPalette((brewer.pal(9, "BuPu")), bias=1))
lines(lowess(log(test$e2f_chip), log(test$myc_chip)), col='black', lty="dashed")

smoothScatter(log(test$e2f_chip), log(test$cebpe_lpmmy), xlab="E2F1 signal (log)", ylab="CEBPE signal (log)", main=sprintf("Spearman corr: %0.2f",cor(test$e2f_chip, test$cebpe_lpmmy, method="spearman")), pch=19, cex=.2, useRaster=T, colramp=colorRampPalette((brewer.pal(9, "BuPu")), bias=1))
lines(lowess(log(test$e2f_chip), log(test$cebpe_lpmmy)), col='black', lty="dashed")

test <- genes_cebpe[which(genes_cebpe$funcClass=="G2_M"),]
#test <- genes_cebpe[grep("B_lpm", genes_cebpe$cluster_deRefined),]
table(test$cluster_deRefined)
smoothScatter(log(test$e2f_chip), log(test$myc_chip), xlab="E2F1 signal (log)", ylab="MYC signal (log)", main=sprintf("Spearman corr: %0.2f",cor(test$myc_chip, test$e2f_chip, method="spearman")), pch=19, cex=.2, useRaster=T, colramp=colorRampPalette((brewer.pal(9, "BuPu")), bias=1))
lines(lowess(log(test$e2f_chip), log(test$myc_chip)), col='black', lty="dashed")

smoothScatter(log(test$e2f_chip), log(test$cebpe_lpmmy), xlab="E2F1 signal (log)", ylab="CEBPE signal (log)", main=sprintf("Spearman corr: %0.2f",cor(test$e2f_chip, test$cebpe_lpmmy, method="spearman")), pch=19, cex=.2, useRaster=T, colramp=colorRampPalette((brewer.pal(9, "BuPu")), bias=1))
lines(lowess(log(test$e2f_chip), log(test$cebpe_lpmmy)), col='black', lty="dashed")
dev.off()
```


#### refined analysis on B_lpm gene class (NFYB and LIN54 TF binding analysis)
```{r}

dev.off()
pdf("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/pdf/class_bLPM_analysis.pdf")

#---------------------------------------------------------------------#
# correlation between binding affnities of CEBPE, E2F1, NFYB and LIN54
test <- genes_cebpe[grep("B_lpm", genes_cebpe$cluster_deRefined),]
test$cebp_class <- cut2(test$cebpe_lpmmy, m=10)

test.melt <- melt(summaryBy(motifScore_E2f1 ~ cebp_class, data=test, FUN=list(mean)))
p1 <- ggplot(test.melt, aes(cebp_class, variable, fill=(value))) + geom_tile() + 
  scale_fill_gradientn(colours = rev(colorRampPalette(brewer.pal(11, "RdBu"), bias=1)(256))) +
  theme(axis.text.x = element_blank())

test.melt <- melt(summaryBy(motifScore_Cebpe ~ cebp_class, data=test, FUN=list(mean)))
p2 <- ggplot(test.melt, aes(cebp_class, variable, fill=(value))) + geom_tile() + 
  scale_fill_gradientn(colours = rev(colorRampPalette(brewer.pal(11, "RdBu"), bias=0.5)(256))) +
  theme(axis.text.x = element_blank())

test.melt <- melt(summaryBy(motifScore_Lin54 ~ cebp_class, data=test, FUN=list(mean)))
p3 <- ggplot(test.melt, aes(cebp_class, variable, fill=(value))) + geom_tile() + 
  scale_fill_gradientn(colours = rev(colorRampPalette(brewer.pal(11, "RdBu"), bias=1)(256))) +
  theme(axis.text.x = element_blank())

test.melt <- melt(summaryBy(motifScore_Nfyb ~ cebp_class, data=test, FUN=list(mean)))
p4 <- ggplot(test.melt, aes(cebp_class, variable, fill=(value))) + geom_tile() + 
  scale_fill_gradientn(colours = rev(colorRampPalette(brewer.pal(11, "RdBu"), bias=0.5)(256))) +
  theme(axis.text.x = element_text(size=10, angle=45))

ggarrange(p1, p2, p3, p4, nrow=4)

plotCustomPCA(scale_df(test[,c("motifScore_Nfyb", "numTagsNfyb", "motifScore_Lin54", "motifScore_E2f1", 
                               "motifScore_Cebpe", "cebpe_lpmmy", "e2f", "motifScore_Myc")], "zo"), 8)

mat <- corr.test(test[,c("cebpe_lpmmy", "motifScore_E2f1", "motifScore_Nfyb", "motifScore_Lin54", "numTagsNfyb", "e2f")], adjust = "BH", method = "spearman")
diag(mat$p) <- 1e-256
p_lst <- list(a=pheatmap(cor(scale_df(test[,c("cebpe_lpmmy", "motifScore_E2f1", "motifScore_Nfyb", "motifScore_Lin54", "numTagsNfyb", "e2f")], "none"), method="spearman"), scale="none", col=colorRampPalette(c("dodgerblue4","grey97", "sienna3"), bias=2)(250))[[4]],
    b=pheatmap(-log10(mat$p), col = rev(colorBrewer2palette("RdBu", bias=0.1, count=256)))[[4]])
ggarrange(plotlist = p_lst, ncol = 2)

#---------------------------------------------------------------------#
# relationship between TF co-binding and logFC in gene expression
test <- genes_cebpe[grep("B_lpm", genes_cebpe$cluster_deRefined),]
# test$peak_class_broad <- "bkg"
# test[which(test$e2f > 0 & test$cebpe_lpmmy > THRESHOLD),]$peak_class_broad <- "e2f1+cebpe"
# #test[which(test$e2f > 0 & test$cebpe_lpmmy > THRESHOLD & test$motifScore_Lin54>0),]$peak_class_broad <- "e2f1+cebpe+lin54"
# test[which(test$e2f > 0 & test$cebpe_lpmmy > THRESHOLD & (test$motifScore_Lin54>0 | test$motifScore_Nfyb>0 | test$nfyb_bound=="Y")),]$peak_class_broad <- "e2f1+cebpe+nfyb+lin54"
# #test[grep("^Ccnb1$", test$gene),c("e2f", "e2f_bound", "cebpe_lpmmy", "motifScore_Nfyb", "numTagsNfyb", "nfyb_bound", "peak_class_broad")]
# test[grep("^Ccnb1$", test$gene),]$peak_class_broad <- "e2f1+cebpe+nfyb+lin54"
# table(test$peak_class_broad)

t <- cbind(ifelse(test$cebpe_lpmmy > THRESHOLD, "cebpe", "bkg"),
           ifelse(test$e2f>0, "e2f1", "bkg"),
           ifelse((test$motifScore_Nfyb>0 | test$nfyb_bound=="Y"), "nfyb", "bkg"),
           ifelse(test$motifScore_Lin54>0, "lin54", "bkg"))
                  
test$peak_class <- paste(t[,1], t[,2], t[,3], t[,4], sep="_")
#test$peak_class <- paste(t[,1], t[,2], t[,3], sep="_")
table(test$peak_class)

#wilcox.test(test[which(test$peak_class=="e2f1_cebpe_lin54_nfyb"),c("cebpe_lpmmy")], test[which(test$peak_class=="e2f1_cebpe_lin54_bkg"),c("cebpe_lpmmy")], alternative = "g")$p.value

test$motifScore_Cebpe_binary <- ifelse(test$motifScore_Cebpe>0, "Y", "N")

PVALUE_TF <- as.data.frame(cbind(
  names(table(test$peak_class)),
  table(test$peak_class),
  t(as.data.frame(lapply(names(table(test$peak_class)), function(x) 
    wilcox.test(test[which(test$peak_class==x),c("mm_ko_norm")], test[which(test$peak_class==x),c("mm_wt_norm")], alternative = "g")$p.value))), 
  t(as.data.frame(lapply(names(table(test$peak_class)), function(x) 
    mean(!is.na(test[which(test$peak_class==x),c("log2FC_mm")]))))),
  t(as.data.frame(lapply(names(table(test$peak_class)), function(x) 
    median(test[which(test$peak_class==x),c("cebpe_lpmmy")])))),
  t(as.data.frame(lapply(names(table(test$peak_class)), function(x) 
    mean(test[which(test$peak_class==x),c("e2f")])))),
  t(as.data.frame(lapply(names(table(test$peak_class)), function(x) 
    mean(test[which(test$peak_class==x),c("myc")])))),
  t(as.data.frame(lapply(names(table(test$peak_class)), function(x) 
    mean(test[which(test$peak_class==x),c("nfyb_chip")])))),
  t(as.data.frame(lapply(names(table(test$peak_class)), function(x) 
    mean(test[which(test$peak_class==x),c("motifScore_Cebpe")])))),
  t(as.data.frame(lapply(names(table(test$peak_class)), function(x) 
    mean(test[which(test$peak_class==x),c("motifScore_E2f1")])))),
  t(as.data.frame(lapply(names(table(test$peak_class)), function(x) 
    mean(test[which(test$peak_class==x),c("motifScore_Myc")])))),
  t(as.data.frame(lapply(names(table(test$peak_class)), function(x) 
    mean(test[which(test$peak_class==x),c("motifScore_Nfyb")])))),
  t(as.data.frame(lapply(names(table(test$peak_class)), function(x) 
    mean(test[which(test$peak_class==x),c("motifScore_Lin54")])))),
  t(as.data.frame(lapply(names(table(test$peak_class)), function(x) 
    sprintf("%0.1f", (nrow(test[which(test$peak_class==x & test$motifScore_Cebpe>0),])*100)/nrow(test[which(test$peak_class==x),])))))
  ))

colnames(PVALUE_TF) <- c("tf", "gene_count", "sig", "medianLog2FC", "cebpe_lpmmy", "e2f", "myc", "numTagsNfyb", 
                         "motifScore_Cebpe", "motifScore_E2f1", "motifScore_Myc", "motifScore_Nfyb", "motifScore_Lin54", "per_genes_with_cebpe_motif")
PVALUE_TF[,c(2:ncol(PVALUE_TF))] <- apply(PVALUE_TF[,c(2:ncol(PVALUE_TF))], 2, function(x) as.numeric(as.character(x)))
PVALUE_TF <-  PVALUE_TF[which(PVALUE_TF$gene_count>30),]

PVALUE_TF$tf <- factor(PVALUE_TF$tf, levels = PVALUE_TF[order(PVALUE_TF$sig),]$tf, ordered = T)
PVALUE_TF$sig <- p.adjust(PVALUE_TF$sig, "BH")

PVALUE_TF[levels(PVALUE_TF$tf),]$tf
PVALUE_TF[levels(PVALUE_TF$tf),]$gene_count

p1 <- ggplot(PVALUE_TF, aes(tf, -log10(sig))) + geom_bar(stat="identity") + theme_bw() +
  #theme(axis.text.x = element_blank()) + 
  geom_hline(yintercept = -log10(0.05), lty=2)

p2 <- ggplot(PVALUE_TF, aes(x = tf, y = 1, label = -log10(sig))) +
  geom_point(aes(size = -log10(sig), alpha=.02, fill = medianLog2FC), shape=21) + 
  #geom_text(hjust = 1, size = 2) +
  scale_size(range = c(1,20)) +
  theme_bw() + scale_y_discrete(position="right") + ylab("") + 
  scale_fill_gradientn(colours = (colorRampPalette(brewer.pal(9, "Blues"), bias=1)(256))) + xlab("") + 
  theme(axis.text.x = element_blank())

# ggplot(PVALUE_TF, aes(tf, per_genes_with_cebpe_motif)) + geom_bar(stat="identity") +
#   geom_text(data=PVALUE_TF, aes(tf, per_genes_with_cebpe_motif, label=per_genes_with_cebpe_motif), vjust=0)
# 
# ggplot(test, aes(motifScore_Cebpe, color=e2f_bound)) + stat_ecdf()
# freqPlot(melt(table(test[,c("e2f_bound", "motifScore_Cebpe_binary")])), "density")
# fisher.test(table(test[,c("e2f_bound", "motifScore_Cebpe_binary")]))$p.value

ggarrange(p1, p2, nrow=2)

p1 <- ggplot(melt(PVALUE_TF[,c("tf", "motifScore_Lin54")]), aes(tf, variable, fill=(value))) + geom_tile() + 
  scale_fill_gradientn(colours = rev(colorRampPalette(brewer.pal(11, "RdBu"), bias=2.5)(256))) +
  theme_bw() + theme(axis.text.x = element_blank())

p2 <- ggplot(melt(PVALUE_TF[,c("tf", "numTagsNfyb")]), aes(tf, variable, fill=(value))) + geom_tile() + 
  scale_fill_gradientn(colours = rev(colorRampPalette(brewer.pal(11, "RdBu"), bias=1.5)(256))) +
  theme_bw() + theme(axis.text.x = element_blank())

p3 <- ggplot(melt(PVALUE_TF[,c("tf", "e2f")]), aes(tf, variable, fill=(value))) + geom_tile() + 
  scale_fill_gradientn(colours = rev(colorRampPalette(brewer.pal(11, "RdBu"), bias=2.5)(256))) +
  theme_bw() + theme(axis.text.x = element_blank())

p4 <- ggplot(melt(PVALUE_TF[,c("tf", "cebpe_lpmmy")]), aes(tf, variable, fill=(value))) + geom_tile() + 
  scale_fill_gradientn(colours = rev(colorRampPalette(brewer.pal(11, "RdBu"), bias=1)(256))) +
  theme_bw() + theme(axis.text.x = element_text(size=6, angle=45))

# mat <- t(as.data.frame(lapply(levels(PVALUE_TF$tf), function(x) ifelse(unlist(strsplit(as.character(x), "_"))=="bkg", 0, 1))))
# row.names(mat) <- levels(PVALUE_TF$tf)
# matrix2pheatmap(x=mat, col=colorBrewer2palette("Greys", count=9))

ggarrange(p1, p2, p3, p4, nrow=4)

#---------------------------------------------------------------------#
# TF peak class analysis
test <- as.data.frame(cbind(names(table(genes_cebpe$cluster_de)),
                            (table(genes_cebpe[which(genes_cebpe$motifScore_Nfyb>0),]$cluster_de)*100)/table(genes_cebpe$cluster_de),
                            (table(genes_cebpe[which(genes_cebpe$numTagsNfyb>0),]$cluster_de)*100)/table(genes_cebpe$cluster_de),
                            (table(genes_cebpe[which(genes_cebpe$motifScore_Lin54>0),]$cluster_de)*100)/table(genes_cebpe$cluster_de),
                            (table(genes_cebpe[which(genes_cebpe$peak_class=="e2f1+cebpe+nfyb+lin54"),]$cluster_de)*100)/table(genes_cebpe$cluster_de),
                            (table(genes_cebpe[which(genes_cebpe$peak_class=="e2f1+cebpe"),]$cluster_de)*100)/table(genes_cebpe$cluster_de),
                            (table(genes_cebpe[which(genes_cebpe$e2f>0 & genes_cebpe$myc>0),]$cluster_de)*100)/table(genes_cebpe$cluster_de)
                            ))
colnames(test) <- c("cluster_de", "nfyb_motif", "nfyb_peak", "lin54_motif", "peak_all", "e2f1_cebpe", "e2f1_myc")
test$nfyb_motif <- as.numeric(as.character(test$nfyb_motif))
test$nfyb_peak <- as.numeric(as.character(test$nfyb_peak))
test$lin54_motif <- as.numeric(as.character(test$lin54_motif))
test$peak_all <- as.numeric(as.character(test$peak_all))
test$e2f1_cebpe <- as.numeric(as.character(test$e2f1_cebpe))
test$e2f1_myc <- as.numeric(as.character(test$e2f1_myc))
p1 <- ggplot(melt(test), aes(cluster_de, value)) + geom_bar(stat="identity") + facet_wrap(.~variable, scales="free") + theme_bw() + ylab("% of promoters")

test <- genes_cebpe[grep("B_lpm", genes_cebpe$cluster_deRefined),]
table(test$peak_class)
table(test$cebp_class)

p2 <- freqPlot(melt(table(test[,c("cluster_deRefined", "peak_class")])), "density", "") + 
  scale_fill_manual(values=c("#8da0cb", "#66c2a5", "#fc8d62")) +
  theme_bw() + theme(legend.position = "top")
# ggplot(melt(test[,c(114,28:37)]), aes(variable, (value))) + geom_boxplot() + facet_grid(.~peak_class)
# freqPlot(melt(table(test[,c("peak_class", "de_mm")])), "density") + theme_bw()
# table(test[,c("peak_class", "de_mm")])
# test.sub <- rbind(test[(grepl("e2f1\\+cebpe$", test$peak_class) & grepl("down", test$de_mm)),],
#                   test[(grepl("e2f1\\+cebpe$", test$peak_class) & grepl("up", test$de_mm)),])
# freqPlot(melt(table(test.sub[,c("de_mm", "cebp_class")])), "density")
# df.melt <- (melt(test.sub[,c("de_mm", "cebpa_gmp", "cebpe_lpmmy", "e2f", "numTagsNfyb")]))
# ggplot(df.melt, aes(de_mm, (value))) + geom_boxplot() + facet_wrap(~variable)

# ggarrange(freqPlot(melt(table(genes_cebpe[grep("-de", genes_cebpe$cluster_deRefined),c("cluster_deRefined", "nfyb_bound")])), "density", "") + theme_bw() + theme(legend.position = "top"), freqPlot(melt(table(genes_cebpe[grep("-neutral", genes_cebpe$cluster_deRefined),c("cluster_deRefined", "nfyb_bound")])), "density", "") + theme_bw() + theme(legend.position = "top"), nrow=2)
p3 <- freqPlot(melt(table(test[,c("peak_class", "nfyb_bound")])), "density", "") + 
  scale_fill_manual(values=c("#bababa", "#4d4d4d")) +
  theme_bw() + theme(legend.position = "top")

p4 <- ggplot(melt(test[,c("log2FC_mm", "peak_class")]), aes(value, color=peak_class)) + stat_ecdf() + theme_classic2() + xlim(c(-4,4)) +
  scale_color_manual(values=c("#8da0cb", "#66c2a5", "#fc8d62")) +
  theme(legend.position = "top")

test.melt <- melt(test[,c("peak_class", "mm_wt_norm", "mm_ko_norm")])
p5 <- ggboxplot(test.melt, x="variable", y="value", notch=T, notchwdth=0.6, facet.by="peak_class", fill="peak_class", outlier.shape = NA) +
  xlab("") + ylab("Gene expr. (norm)") +
  scale_fill_manual(values=c("#8da0cb", "#66c2a5", "#fc8d62")) + 
  ylim(c(-200,2100)) +
  stat_compare_means(comparisons = list(c("mm_ko_norm", "mm_wt_norm")), method.args = list(alternative = "two.sided"), label.y=2050) +
  theme(legend.position = "top")

# p <- lapply(unique(test$peak_class), function(x) {
#   t.melt <- (melt(test[which(test$peak_class==x),c(31,32,33,35,43,44,45,46)]))
#   #t.melt <- (melt(log(x[,c(11,12,13,15,16,24,25,26,27,28)]+1)))
#   t.melt$class <- gsub("_.*", "", (t.melt$variable))
#   t.melt$variable <- gsub("^[^\\_]*_", "", t.melt$variable)
#   t.melt <- summarySE(t.melt, measurevar = "value", groupvars = c("class", "variable"))
#   t.melt$class <- factor(t.melt$class, levels=c("gmpG", "epm", "lpm", "mm"))
#   ggplot(t.melt, aes(class, value, color=variable)) +
#     geom_errorbar(aes(ymin=value-se, ymax=value+se), width=.1, alpha=1) +
#     geom_line(aes(group=factor(variable), color=factor(variable)), alpha=1) +
#     geom_point(shape = 21, size = 2, fill="white") +
#     scale_color_manual(values=c("#313695", "#a50026")) +
#     theme_bw() + theme(legend.position="top")
# })
# names(p) <- unique(test$peak_class)
# ggarrange(plotlist = p, ncol=3)

test.melt <- melt(test[,c("cebpe_lpmmy", "peak_class")])
test.melt$value <- log(test.melt$value)
p6 <- ggboxplot(test.melt, x="peak_class", y="value", notch=1, notchwdth=0.6, facet.by="variable", fill="peak_class") +
  xlab("") + ylab("CEBPE Signal (log)") +
  scale_fill_manual(values=c("#8da0cb", "#66c2a5", "#fc8d62")) +
  stat_compare_means(comparisons = list(c("e2f1+cebpe+nfyb+lin54", "e2f1+cebpe"), c("e2f1+cebpe", "bkg")), method.args = list(alternative = "g")) +
  theme(legend.position = "top")

#p7 <- freqPlot(melt(table(test[,c("peak_class", "e2f_bound")])), "density") + theme_bw()

test.melt <- melt(test[,c("e2f_chip", "peak_class")])
test.melt$value <- log(test.melt$value)
p7 <- ggboxplot(test.melt, x="peak_class", y="value", notch=1, notchwdth=0.6, facet.by="variable", fill="peak_class") +
  xlab("") + ylab("E2f1 Signal (log)") +
  scale_fill_manual(values=c("#8da0cb", "#66c2a5", "#fc8d62")) +
  stat_compare_means(comparisons = list(c("e2f1+cebpe+nfyb+lin54", "e2f1+cebpe"), c("e2f1+cebpe", "bkg")), method.args = list(alternative = "g")) +
  theme(legend.position = "top")

test.melt <- melt(test[,c("nfyb_chip", "peak_class")])
test.melt$value <- log(test.melt$value)
p8 <- ggboxplot(test.melt, x="peak_class", y="value", notch=1, notchwdth=0.6, facet.by="variable", fill="peak_class") +
  xlab("") + ylab("Nfyb Signal (log)") +
  scale_fill_manual(values=c("#8da0cb", "#66c2a5", "#fc8d62")) +
  stat_compare_means(comparisons = list(c("e2f1+cebpe+nfyb+lin54", "e2f1+cebpe"), c("e2f1+cebpe", "bkg")), method.args = list(alternative = "g")) +
  theme(legend.position = "top")

ggarrange(p1, p2, p3, p4, p5, p6, p7, p8, nrow=2, ncol=4)

mat <- merge(res_mm, test[grep("G2_M", test$funcClass),c("gene", "peak_class", "numTagsNfyb", "funcClass", "cell_cycle")], by.x="gene", by.y="gene")
#mat <- merge(res_mm, test[(test$cell_cycle=="cell_cycle" | grepl("G1_S|G2_M", test$funcClass)),c("gene", "peak_class", "numTagsNfyb", "funcClass")], by.x="gene", by.y="gene")
#mat <- merge(res_mm, test[,c("gene", "peak_class", "numTagsNfyb", "funcClass")], by.x="gene", by.y="gene")
#mat <- mat[grep("e2f1\\+cebpe\\+nfyb", mat$peak_class),]
colVals <- rep("#8da0cb", nrow(mat))
names(colVals) <- rep("bkg", nrow(mat))
colVals[grep("e2f1\\+cebpe", mat$peak_class)] <- "#66c2a5"
names(colVals)[grep("e2f1\\+cebpe", mat$peak_class)] <- "e2f1+cebpe"
colVals[grep("e2f1\\+cebpe\\+nfyb", mat$peak_class)] <- "#fc8d62"
names(colVals)[grep("e2f1\\+cebpe\\+nfyb", mat$peak_class)] <- "e2f1+cebpe+nfyb"
table(names(colVals))
table(mat$peak_class)

# ggplot(mat, aes(log2FoldChange, -log10(padj), color=peak_class)) +
#   geom_point(shape = 21, size = log(mat$numTagsNfyb+2), fill="white") +
#   scale_color_manual(values=c("#313695", "#a50026", "#4daf4a")) +
#   theme(legend.position="top") + theme_bw()

EnhancedVolcano::EnhancedVolcano(mat, lab=mat$gene, x='log2FoldChange', y='padj', 
                                 pCutoff = 1,
                                 FCcutoff = 0,
                                 selectLab = mat[grep("G2_M", mat$funcClass),]$gene,
                                 drawConnectors = TRUE,
                                 widthConnectors = 0.2,
                                 colConnectors = '#000000',
                                 colCustom = colVals,
                                 transcriptPointSize=10,
                                 title="",
                                 subtitle = "G2/M genes")
dev.off()

# test.melt <- melt(test[,c(105:111)])
# test.melt$value <- log(test.melt$value)
# ggboxplot(test.melt, x="cluster_deRefined", y="value",
#           fill = "variable", notch=1, notchwdth=0.6, facet.by="variable") +
#   xlab("") + ylab("log2FC") +
#   stat_compare_means(comparisons = list(c("B_lpm-de", "B_lpm-neutral")), method.args = list(alternative = "g"))
  
# barplot(table(genes_cebpe[which(genes_cebpe$e2f_bound=="Y" & genes_cebpe$cebpe_bound=="Y"),]$cluster_deRefined)[c(1,3)]/table(genes_cebpe$cluster_deRefined)[c(1,3)], las=2)

# barplot(table(genes_cebpe[which(genes_cebpe$e2f_bound=="Y" & genes_cebpe$myc_bound=="Y"),]$cluster_deRefined)[c(1,3)]/table(genes_cebpe$cluster_deRefined)[c(1,3)], las=2)

# barplot(table(genes_cebpe[which(genes_cebpe$e2f_bound=="Y" & !is.na(genes_cebpe$motifScore_Lin54)),]$cluster_deRefined)[1:3]/table(genes_cebpe$cluster_deRefined)[1:3], las=2)

df <- read.table("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.hist", header=T, sep="\t")
colnames(df) <- gsub("[\\.\\_]+.*", "", colnames(df))

pdf("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/pdf/test.pdf")
p <- lapply(colnames(df)[2:7], function(x) {
  plot(df[,c(x)] ~ as.numeric(Distance), data = df[,c("Distance", x)], main=x, ylab="Sites per bp per peak")
  cars.spl <- with(df[,c("Distance", x)], smooth.spline(as.numeric(Distance), df[,c(x)]))
  lines(cars.spl, col = "blue")
})
dev.off()

df <- read.table("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/t")
colnames(df) <- c("chr_e2f1", "start_e2f1", "end_e2f1", "score_e2f1", "strand_e2f1", "dist_to_cebpe", colnames(genes_cebpe), "dist_to_gene")
df <- df[grep("B_lpm", df$cluster_deRefined),]
#df <- df[abs(df$dist_to_cebpe)<1000,]
df <- df[order(df$dist_to_cebpe),]
barplot(df$dist_to_cebpe)
barplot(df$dist_to_gene)
image(as.matrix(ifelse(df$peak_class=="bkg", 0, 1)))
image(as.matrix(df$log2FC_mm))

df$cebp_class <- "on_e2f"
df[abs(df$dist_to_cebpe)>1000,]$cebp_class <- "away_e2f"
ggplot(melt(df[,c("cebp_class", "log2FC_mm")]), aes(cebp_class, value)) + geom_boxplot()

#---------------------------------------------------------------------#

# plot(Venn(list(peak_e2f=genes_cebpe[which(genes_cebpe$e2f>0),]$gene, motif_e2f=genes_cebpe[which(!is.na(genes_cebpe$motifScore_E2f1)),]$gene)))
# plot(Venn(list(peak_myc=genes_cebpe[which(genes_cebpe$myc>0),]$gene, motif_myc=genes_cebpe[which(!is.na(genes_cebpe$motifScore_Myc)),]$gene)))
# par(mfrow=c(1,3))
# barplot(table(genes_cebpe[which(genes_cebpe$e2f>0 & !is.na(genes_cebpe$motifScore_E2f1)),]$cluster_de)/table(genes_cebpe$cluster_de), las=2, main="peak+motif")
# barplot(table(genes_cebpe[which(genes_cebpe$e2f==0 & !is.na(genes_cebpe$motifScore_E2f1)),]$cluster_de)/table(genes_cebpe$cluster_de), las=2, main="motif")
# barplot(table(genes_cebpe[which(genes_cebpe$e2f>0 & is.na(genes_cebpe$motifScore_E2f1)),]$cluster_de)/table(genes_cebpe$cluster_de), las=2, main="peak")

# barplot(table(genes_cebpe[which(genes_cebpe$cebpe_bound=="Y" & genes_cebpe$e2f_bound=="Y"),]$cluster_deRefined)/table(genes_cebpe$cluster_deRefined), las=2, ylab="% genes bound by CEBPE and E2F1", ylim=c(0,0.2))
# 
# barplot(table(genes_cebpe[which(genes_cebpe$cebpe_bound=="Y" & genes_cebpe$myc_bound=="Y"),]$cluster_deRefined)/table(genes_cebpe$cluster_deRefined), las=2, ylab="% genes bound by CEBPE and Myc", ylim=c(0,0.1))
```

#### enhancer analysis
```{r}
test <- genes_cebpe[grepl("C_gran", genes_cebpe$cluster_deRefined),]
test$cebpe_lpmmy_dis <- test$cebpe_lpmmy_dis/test$enhancer_count
test$cluster_deRefined <- factor(test$cluster_deRefined, levels=c("C_gran-de", "C_gran-neutral"), ordered = T)
table(test$cluster_deRefined)

test.melt <- melt(test[,c("cluster_deRefined", "cebpe_lpmmy", "cebpe_lpmmy_dis")])
test.melt$value <- log(test.melt$value)
p1 <- ggboxplot(test.melt, x="cluster_deRefined", y="value", notch=1, notchwdth=0.6, facet.by="variable", fill="cluster_deRefined") +
  xlab("") + ylab("CEBPE Signal (log)") +
  scale_fill_manual(values=c("#8da0cb", "#66c2a5")) +
  stat_compare_means(comparisons = list(c("C_gran-de", "C_gran-neutral")), method.args = list(alternative = "g")) +
  theme(legend.position = "top")

mat <- merge(res_mm, test[,c("gene", "cluster_deRefined", "cebpe_lpmmy", "cebpe_lpmmy_dis")], by.x="gene", by.y="gene")
colVals <- rep("#8da0cb", nrow(mat))
names(colVals) <- rep("bkg", nrow(mat))
colVals[grep("C_gran-de", mat$cluster_deRefined)] <- "#313695"
names(colVals)[grep("C_gran-de", mat$cluster_deRefined)] <- "C_gran-de"
colVals[grep("C_gran-neutral", mat$cluster_deRefined)] <- "#4daf4a"
names(colVals)[grep("C_gran-neutral", mat$cluster_deRefined)] <- "C_gran-neutral"
table(names(colVals))
table(mat$peak_class)

p2 <- ggplot(mat, aes(log2FoldChange, -log10(padj), color=cluster_deRefined)) +
  geom_point(shape = 21, size = log(mat$cebpe_lpmmy), fill="white") +
  scale_color_manual(values=c("#313695", "#4daf4a")) +
  theme_bw() + theme(legend.position="top")

p3 <- ggplot(mat, aes(log2FoldChange, -log10(padj), color=cluster_deRefined)) +
  geom_point(shape = 21, size = log(mat$cebpe_lpmmy_dis), fill="white") +
  scale_color_manual(values=c("#313695", "#4daf4a")) +
  theme_bw() + theme(legend.position="top")

gXGranClass <- ggarrange(p1, p2, p3, ncol=3)

gXGranClass
```

#### plot results showing analysis on gran class
```{r}
pdf("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/pdf/analysis_gran_class.pdf")
gXGranClass
dev.off()
```

#### cebpa and cebpe expression analysis
```{r}
#---------------------------------------------------------------------#
# Compare Cebpa and Cebpe expression profile during myelopoiesis

cebp_granulopoiesis <- melt(expr_data_norm_with_replicates[grep("^Cebpa$|^Cebpe$|^E2f1$|^Cebpb$|^Myc$|^Mxd1$|^Max$|^Cd81$|^Vcam1$|^Jag1$",
                                                                expr_data_norm_with_replicates$GeneID),
                               c("GeneID",
                                 grep("lsk_.*wt", colnames(expr_data_norm_with_replicates), value = T),
                                 grep("pregm_.*wt", colnames(expr_data_norm_with_replicates), value = T),
                                 grep("gmpG.*wt", colnames(expr_data_norm_with_replicates), value = T),
                                 grep("epm_.*wt", colnames(expr_data_norm_with_replicates), value = T),
                                 grep("lpm_.*wt", colnames(expr_data_norm_with_replicates), value = T),
                                 grep("my_.*wt", colnames(expr_data_norm_with_replicates), value = T),
                                 grep("mm_.*wt", colnames(expr_data_norm_with_replicates), value = T),
                                 grep("bc_.*wt", colnames(expr_data_norm_with_replicates), value = T),
                                 grep("pmn_.*wt", colnames(expr_data_norm_with_replicates), value = T)
                                 )])
LEVELS <- unique(gsub("_[Rr]ep.*|_cebpa", "", cebp_granulopoiesis$variable))
cebp_granulopoiesis$variable <- gsub("_[Rr]ep.*|_cebpa", "", cebp_granulopoiesis$variable)
cebp_granulopoiesis$value <- cebp_granulopoiesis$value
cebp_granulopoiesis <- summarySE(cebp_granulopoiesis, measurevar = "value", groupvars = c("GeneID", "variable"))
cebp_granulopoiesis$variable <- factor(cebp_granulopoiesis$variable, levels=LEVELS, ordered = T)
cebp_granulopoiesis$lineage <- "granulopoiesis"

cebp_monopoiesis <- melt(expr_data_norm_with_replicates[grep("^Cebpa$|^Cebpe$|^E2f1$|^Cebpb$|^Myc$|^Mxd1$|^Max$|^Cd81$|^Vcam1$|^Jag1$", 
                                                             expr_data_norm_with_replicates$GeneID),
                               c("GeneID",
                                 grep("lsk_.*wt", colnames(expr_data_norm_with_replicates), value = T),
                                 grep("pregm_.*wt", colnames(expr_data_norm_with_replicates), value = T),
                                 grep("gmpM.*wt", colnames(expr_data_norm_with_replicates), value = T),
                                 grep("pmo1_.*wt", colnames(expr_data_norm_with_replicates), value = T),
                                 grep("pmo2_.*wt", colnames(expr_data_norm_with_replicates), value = T),
                                 grep("mo1_.*wt", colnames(expr_data_norm_with_replicates), value = T),
                                 grep("mo2_.*wt", colnames(expr_data_norm_with_replicates), value = T)
                                 )])

LEVELS <- unique(gsub("_[Rr]ep.*|_cebpa", "", cebp_monopoiesis$variable))
cebp_monopoiesis$variable <- gsub("_[Rr]ep.*|_cebpa", "", cebp_monopoiesis$variable)
cebp_monopoiesis$value <- cebp_monopoiesis$value
cebp_monopoiesis <- summarySE(cebp_monopoiesis, measurevar = "value", groupvars = c("GeneID", "variable"))
cebp_monopoiesis$variable <- factor(cebp_monopoiesis$variable, levels=LEVELS, ordered = T)
cebp_monopoiesis$lineage <- "monopoiesis"

p1 <- ggplot(cebp_granulopoiesis[grep("Cebpa", cebp_granulopoiesis$GeneID),], aes(variable, value)) +
  geom_errorbar(aes(ymin=value-se, ymax=value+se), width=.1, alpha=1) +
  geom_point(shape = 21, size = 2, aes(fill=lineage)) +
  #scale_fill_manual(values = c(rep("#000000",12))) +
  geom_line(aes(group=1, color=lineage), alpha=1) +
  theme_classic() + theme(legend.position="top", axis.text.x = element_text(angle = 45, hjust = 1)) + ylab("Expr.") +
  facet_wrap(~lineage) + ylim(c(0,1250))
p2 <- ggplot(cebp_monopoiesis[grep("Cebpa", cebp_monopoiesis$GeneID),], aes(variable, value)) +
  geom_errorbar(aes(ymin=value-se, ymax=value+se), width=.1, alpha=1) +
  geom_point(shape = 21, size = 2, aes(fill=lineage)) +
  #scale_fill_manual(values = c(rep("#000000",12))) +
  geom_line(aes(group=1, color=lineage), alpha=1) +
  theme_classic() + theme(legend.position="top", axis.text.x = element_text(angle = 45, hjust = 1)) + ylab("Expr.") +
  facet_wrap(~lineage) + ylim(c(0,1250))
p3 <- ggplot(cebp_granulopoiesis[grep("Cebpe", cebp_granulopoiesis$GeneID),], aes(variable, value)) +
  geom_errorbar(aes(ymin=value-se, ymax=value+se), width=.1, alpha=1) +
  geom_point(shape = 21, size = 2, aes(fill=lineage)) +
  #scale_fill_manual(values = c(rep("#000000",12))) +
  geom_line(aes(group=1, color=lineage), alpha=1) +
  theme_classic() + theme(legend.position="top", axis.text.x = element_text(angle = 45, hjust = 1)) + ylab("Expr.") +
  facet_wrap(~lineage) + ylim(c(-30,7000))
p4 <- ggplot(cebp_monopoiesis[grep("Cebpe", cebp_monopoiesis$GeneID),], aes(variable, value)) +
  geom_errorbar(aes(ymin=value-se, ymax=value+se), width=.1, alpha=1) +
  geom_point(shape = 21, size = 2, aes(fill=lineage)) +
  #scale_fill_manual(values = c(rep("#000000",12))) +
  geom_line(aes(group=1, color=lineage), alpha=1) +
  theme_classic() + theme(legend.position="top", axis.text.x = element_text(angle = 45, hjust = 1)) + ylab("Expr.") +
  facet_wrap(~lineage) + ylim(c(-30,7000))
# p5 <- ggplot(cebp_granulopoiesis[grep("E2f1", cebp_granulopoiesis$GeneID),], aes(variable, value)) +
#   geom_errorbar(aes(ymin=value-se, ymax=value+se), width=.1, alpha=1) +
#   geom_point(shape = 21, size = 2, aes(fill=lineage)) +
#   #scale_fill_manual(values = c(rep("#000000",12))) +
#   geom_line(aes(group=1, color=lineage), alpha=1) +
#   theme_classic() + theme(legend.position="top", axis.text.x = element_text(angle = 45, hjust = 1)) + ylab("Expr.") +
#   facet_wrap(~lineage) +  ylim(c(100,800))
# p6 <- ggplot(cebp_monopoiesis[grep("E2f1", cebp_monopoiesis$GeneID),], aes(variable, value)) +
#   geom_errorbar(aes(ymin=value-se, ymax=value+se), width=.1, alpha=1) +
#   geom_point(shape = 21, size = 2, aes(fill=lineage)) +
#   #scale_fill_manual(values = c(rep("#000000",12))) +
#   geom_line(aes(group=1, color=lineage), alpha=1) +
#   theme_classic() + theme(legend.position="top", axis.text.x = element_text(angle = 45, hjust = 1)) + ylab("Expr.") +
#   facet_wrap(~lineage) +  ylim(c(100,800))
p5 <- ggplot(cebp_granulopoiesis[grep("Myc", cebp_granulopoiesis$GeneID),], aes(variable, value)) +
  geom_errorbar(aes(ymin=value-se, ymax=value+se), width=.1, alpha=1) +
  geom_point(shape = 21, size = 2, aes(fill=lineage)) +
  #scale_fill_manual(values = c(rep("#000000",12))) +
  geom_line(aes(group=1, color=lineage), alpha=1) +
  theme_classic() + theme(legend.position="top", axis.text.x = element_text(angle = 45, hjust = 1)) + ylab("Expr.") +
  facet_wrap(~lineage)
p6 <- ggplot(cebp_monopoiesis[grep("Myc", cebp_monopoiesis$GeneID),], aes(variable, value)) +
  geom_errorbar(aes(ymin=value-se, ymax=value+se), width=.1, alpha=1) +
  geom_point(shape = 21, size = 2, aes(fill=lineage)) +
  #scale_fill_manual(values = c(rep("#000000",12))) +
  geom_line(aes(group=1, color=lineage), alpha=1) +
  theme_classic() + theme(legend.position="top", axis.text.x = element_text(angle = 45, hjust = 1)) + ylab("Expr.") +
  facet_wrap(~lineage)
p7 <- ggplot(cebp_granulopoiesis[grep("Mxd1", cebp_granulopoiesis$GeneID),], aes(variable, value)) +
  geom_errorbar(aes(ymin=value-se, ymax=value+se), width=.1, alpha=1) +
  geom_point(shape = 21, size = 2, aes(fill=lineage)) +
  #scale_fill_manual(values = c(rep("#000000",12))) +
  geom_line(aes(group=1, color=lineage), alpha=1) +
  theme_classic() + theme(legend.position="top", axis.text.x = element_text(angle = 45, hjust = 1)) + ylab("Expr.") +
  facet_wrap(~lineage) + ylim(c(0,40000))
p8 <- ggplot(cebp_monopoiesis[grep("Mxd1", cebp_monopoiesis$GeneID),], aes(variable, value)) +
  geom_errorbar(aes(ymin=value-se, ymax=value+se), width=.1, alpha=1) +
  geom_point(shape = 21, size = 2, aes(fill=lineage)) +
  #scale_fill_manual(values = c(rep("#000000",12))) +
  geom_line(aes(group=1, color=lineage), alpha=1) +
  theme_classic() + theme(legend.position="top", axis.text.x = element_text(angle = 45, hjust = 1)) + ylab("Expr.") +
  facet_wrap(~lineage) + ylim(c(0,40000))
p9 <- ggplot(cebp_granulopoiesis[grep("Max", cebp_granulopoiesis$GeneID),], aes(variable, value)) +
  geom_errorbar(aes(ymin=value-se, ymax=value+se), width=.1, alpha=1) +
  geom_point(shape = 21, size = 2, aes(fill=lineage)) +
  #scale_fill_manual(values = c(rep("#000000",12))) +
  geom_line(aes(group=1, color=lineage), alpha=1) +
  theme_classic() + theme(legend.position="top", axis.text.x = element_text(angle = 45, hjust = 1)) + ylab("Expr.") +
  facet_wrap(~lineage) +  ylim(c(1000,16000))
p10 <- ggplot(cebp_monopoiesis[grep("Max", cebp_monopoiesis$GeneID),], aes(variable, value)) +
  geom_errorbar(aes(ymin=value-se, ymax=value+se), width=.1, alpha=1) +
  geom_point(shape = 21, size = 2, aes(fill=lineage)) +
  #scale_fill_manual(values = c(rep("#000000",12))) +
  geom_line(aes(group=1, color=lineage), alpha=1) +
  theme_classic() + theme(legend.position="top", axis.text.x = element_text(angle = 45, hjust = 1)) + ylab("Expr.") +
  facet_wrap(~lineage) +  ylim(c(1000,16000))
ggplot(cebp_granulopoiesis[grep("Cd81", cebp_granulopoiesis$GeneID),], aes(variable, value)) +
  geom_errorbar(aes(ymin=value-se, ymax=value+se), width=.1, alpha=1) +
  geom_point(shape = 21, size = 2, aes(fill=lineage)) +
  #scale_fill_manual(values = c(rep("#000000",12))) +
  geom_line(aes(group=1, color=lineage), alpha=1) +
  theme_classic() + theme(legend.position="top", axis.text.x = element_text(angle = 45, hjust = 1)) + ylab("Expr.") +
  facet_wrap(~lineage) + ylim(c(-200,5000))
ggplot(cebp_monopoiesis[grep("Cd81", cebp_monopoiesis$GeneID),], aes(variable, value)) +
  geom_errorbar(aes(ymin=value-se, ymax=value+se), width=.1, alpha=1) +
  geom_point(shape = 21, size = 2, aes(fill=lineage)) +
  #scale_fill_manual(values = c(rep("#000000",12))) +
  geom_line(aes(group=1, color=lineage), alpha=1) +
  theme_classic() + theme(legend.position="top", axis.text.x = element_text(angle = 45, hjust = 1)) + ylab("Expr.") +
  facet_wrap(~lineage) + ylim(c(-200,5000))
ggplot(cebp_granulopoiesis[grep("Vcam1", cebp_granulopoiesis$GeneID),], aes(variable, value)) +
  geom_errorbar(aes(ymin=value-se, ymax=value+se), width=.1, alpha=1) +
  geom_point(shape = 21, size = 2, aes(fill=lineage)) +
  #scale_fill_manual(values = c(rep("#000000",12))) +
  geom_line(aes(group=1, color=lineage), alpha=1) +
  theme_classic() + theme(legend.position="top", axis.text.x = element_text(angle = 45, hjust = 1)) + ylab("Expr.") +
  facet_wrap(~lineage) + ylim(c(-200,3000))
ggplot(cebp_monopoiesis[grep("Vcam1", cebp_monopoiesis$GeneID),], aes(variable, value)) +
  geom_errorbar(aes(ymin=value-se, ymax=value+se), width=.1, alpha=1) +
  geom_point(shape = 21, size = 2, aes(fill=lineage)) +
  #scale_fill_manual(values = c(rep("#000000",12))) +
  geom_line(aes(group=1, color=lineage), alpha=1) +
  theme_classic() + theme(legend.position="top", axis.text.x = element_text(angle = 45, hjust = 1)) + ylab("Expr.") +
  facet_wrap(~lineage) + ylim(c(-200,3000))
ggplot(cebp_granulopoiesis[grep("Jag1", cebp_granulopoiesis$GeneID),], aes(variable, value)) +
  geom_errorbar(aes(ymin=value-se, ymax=value+se), width=.1, alpha=1) +
  geom_point(shape = 21, size = 2, aes(fill=lineage)) +
  #scale_fill_manual(values = c(rep("#000000",12))) +
  geom_line(aes(group=1, color=lineage), alpha=1) +
  theme_classic() + theme(legend.position="top", axis.text.x = element_text(angle = 45, hjust = 1)) + ylab("Expr.") +
  facet_wrap(~lineage) + ylim(c(-200,1000))
ggplot(cebp_monopoiesis[grep("Jag1", cebp_monopoiesis$GeneID),], aes(variable, value)) +
  geom_errorbar(aes(ymin=value-se, ymax=value+se), width=.1, alpha=1) +
  geom_point(shape = 21, size = 2, aes(fill=lineage)) +
  #scale_fill_manual(values = c(rep("#000000",12))) +
  geom_line(aes(group=1, color=lineage), alpha=1) +
  theme_classic() + theme(legend.position="top", axis.text.x = element_text(angle = 45, hjust = 1)) + ylab("Expr.") +
  facet_wrap(~lineage) + ylim(c(-200,1000))

gXExprProfileCebpAE <- ggarrange(p1, p3, p5, p7, p9, p2, p4, p6, p8, p10, nrow=2, ncol=5)

## differential expression between pregm and gmpG vs pregm and gmpM
mat <- as.matrix(expr_data[,c(grep("pregm_.*wt", colnames(expr_data_norm_with_replicates), value = T),
              grep("gmpG.*wt", colnames(expr_data_norm_with_replicates), value = T))])
rownames(mat) <- expr_data$GeneID
mat <- mat[apply(mat, 1, function(x) length(x[x>5]) >= 2),]
colnames(mat)
plotCustomPCA(mat, ncol(mat))
res_pregm_gmpG <- matrix2deseq2(mat, "wt,wt,wt,ko,ko,ko,ko,ko,ko")

mat <- as.matrix(expr_data[,c(grep("pregm_.*wt", colnames(expr_data_norm_with_replicates), value = T),
              grep("gmpM.*wt", colnames(expr_data_norm_with_replicates), value = T))])
rownames(mat) <- expr_data$GeneID
mat <- mat[apply(mat, 1, function(x) length(x[x>5]) >= 2),]
colnames(mat)
plotCustomPCA(mat, ncol(mat))
res_pregm_gmpM <- matrix2deseq2(mat, "wt,wt,wt,ko,ko,ko,ko,ko")

mat <- as.matrix(expr_data[,c(grep("gmpG_.*wt", colnames(expr_data_norm_with_replicates), value = T),
              grep("gmpM.*wt", colnames(expr_data_norm_with_replicates), value = T))])
rownames(mat) <- expr_data$GeneID
mat <- mat[apply(mat, 1, function(x) length(x[x>5]) >= 2),]
colnames(mat)
plotCustomPCA(mat, ncol(mat))
res_gmpG_gmpM <- matrix2deseq2(mat, "wt,wt,wt,wt,wt,wt,ko,ko,ko,ko,ko")

res_pregm_gmpG[grep("Cebpa", res_pregm_gmpG$gene),c(1:7)]
res_pregm_gmpM[grep("Cebpa", res_pregm_gmpM$gene),c(1:7)]
res_gmpG_gmpM[grep("Cebpa", res_gmpG_gmpM$gene),c(1:7)]

## Cebpa gene TF binding
df <- read.table("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/cebpa_promoter_overlap.bed.stat")
#df <- read.table("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/cebpa_overlap.bed.stat")
df$pVal <- apply(df, 1, function(x) fisher.test(matrix(c(as.numeric(x[2]), sum(df$V2), as.numeric(x[3]), sum(df$V3)), nrow=2, ncol=2), alternative = "g")$p.value)
df$qVal <- p.adjust(df$pVal, method = "BH")
df <- df[order(df$qVal),]
# df$cor <- unlist(lapply(df$V1, function(x) {
#   if(any(grepl(sprintf("^%s$", x), genes_cebpe$gene))) {
#     cor(unlist(genes_cebpe[grep("^Cebpa$", genes_cebpe$gene),c(28:42)]), 
#                                                     unlist(genes_cebpe[grep(sprintf("^%s$", x), genes_cebpe$gene),c(28:42)]), method="spearman")
#   } else {
#       0
#     }
#   }))
# df$corPVal <- unlist(lapply(df$V1, function(x) {
#   if(any(grepl(sprintf("^%s$", x), genes_cebpe$gene))) {
#     cor.test(unlist(genes_cebpe[grep("^Cebpa$", genes_cebpe$gene),c(28:42)]), 
#                                                     unlist(genes_cebpe[grep(sprintf("^%s$", x), genes_cebpe$gene),c(28:42)]), method="spearman")$p.value
#   } else {
#       1
#     }
#   }))
# df$corQVal <- p.adjust(df$corPVal, method = "BH")

barplot(df[which(df$qVal<0.001),]$V2, names=df[which(df$qVal<0.001),]$V1, las=2)
# barplot(df[which(df$qVal<0.001),]$cor, names=df[which(df$qVal<0.001),]$V1, las=2)
```

## cebpe and cebpa regulation analysis using Shyamsunder et al. Blood 2019 data
```{r}
df <- read.table("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/shyamsunder_2019/cebpe_enhancer_sg2_5.unique", header=T)
mat <- round(df[,c(2:ncol(df))],0)
row.names(mat) <- df$MGIS
mat <- mat[,c(1,2,5,6,7,8)]

x <- as.factor(c(rep("1_WT",2), rep("1_KD",2), rep("2_KD",2)))
x <- relevel(x, ref="1_WT")
set <- as.matrix(mat)
row.names(set) <- row.names(mat)
set <- newSeqExpressionSet(set, phenoData = data.frame(x, row.names = colnames(mat)))
colors <- brewer.pal(3, "Set2")
set <- betweenLaneNormalization(set, which="upper")

design <- model.matrix(~x, data=pData(set))
y <- DGEList(counts=counts(set), group=x)
y <- calcNormFactors(y, method="TMM")  
y <- estimateDisp(y, design)
fit <- glmQLFit(y, design)
res_shrna1 <- glmQLFTest(fit, coef=2)
res_shrna2 <- glmQLFTest(fit, coef=3)
# plotCustomPCA(fit$fitted.values, 10)
df_shrna1 <- as.data.frame(topTags(res_shrna1, n=Inf, adjust.method="BH", p.value = 1, sort.by = "none"))  
df_shrna1 <- cbind(row.names(df_shrna1), df_shrna1)  
colnames(df_shrna1)[1] <- "gene"  
df_shrna2 <- as.data.frame(topTags(res_shrna2, n=Inf, adjust.method="BH", p.value = 1, sort.by = "none"))
df_shrna2 <- cbind(row.names(df_shrna2), df_shrna2)
colnames(df_shrna2)[1] <- "gene"
df_shrna12 <- merge(df_shrna1, df_shrna2, by.x="gene", by.y="gene")

empirical <- head(as.character(df_shrna12[which(df_shrna12$FDR.x>0.99 & df_shrna12$FDR.y>0.99),]$gene), 100)
length(empirical)
set2 <- RUVg(set, empirical, k=1)
design <- model.matrix(~x + W_1, data=pData(set2))
y <- DGEList(counts=counts(set), group=x)
y <- calcNormFactors(y, method="TMM")
y <- estimateDisp(y, design)
fit <- glmFit(y, design)
res_shrna1 <- glmLRT(fit, coef=2)
res_shrna2 <- glmLRT(fit, coef=3)

boxplot(log(counts(set)[rownames(counts(set)) %in% empirical,]+1), las=2, main="A", ylab="Empirical read counts (log)")
boxplot(log(normCounts(set2)[rownames(normCounts(set2)) %in% empirical,]+1), las=2, main="B", ylab="Empirical read counts (log)")
barplot(pData(set2)$W_1)
plotRLE(set, outline=FALSE, ylim=c(-4, 4), col=colors[x], las=2, main="C", ylab="RLE")
plotRLE(set2, outline=FALSE, ylim=c(-4, 4), col=colors[x], las=2, main="D", ylab="RLE")
EDASeq::plotPCA(set, col=colors[x], cex=1.2, main="H")
EDASeq::plotPCA(set2, col=colors[x], cex=1.2, main="H")

df_shrna1 <- as.data.frame(topTags(res_shrna1, n=Inf, adjust.method="BH", p.value = 1, sort.by = "none"))
df_shrna1 <- cbind(df_shrna1, res_shrna1$fitted.values)
# df_shrna1 <- cbind(df_shrna1, counts(set))
df_shrna1 <- cbind(row.names(df_shrna1), df_shrna1)
colnames(df_shrna1)[1] <- "gene"

df_shrna2 <- as.data.frame(topTags(res_shrna2, n=Inf, adjust.method="BH", p.value = 1, sort.by = "none"))
df_shrna2 <- cbind(df_shrna2, res_shrna2$fitted.values)
# df_shrna2 <- cbind(df_shrna2, counts(set))
df_shrna2 <- cbind(row.names(df_shrna2), df_shrna2)
colnames(df_shrna2)[1] <- "gene"

df_shrna12 <- merge(df_shrna1, df_shrna2, by.x="gene", by.y="gene")
row.names(df_shrna12) <- df_shrna12$gene
df_shrna12$class <- "neutral"

published <- c("Lpar6", "Aqp9", "Gapt", "Plch1", "Ngp", "Cebpe", "S100a8", "Ltf", "Ahcy", "Lyz2", "Slc7a8", "Ablim1", "Pecr", "Ifitm6", "Ltb4r1", 
               "S100a9", "Alas1", "Gm9797", "Car12", "Gm5537", "Gm10241", "Hist1h2be", "Ero1l", "Grin1", "Ankrd37", "Fxyd4", "Jund", "Hist1h2br",
               "Gipr", "Junb", "Trpc2", "Lgals3", "Prss34", "Ppbp", "Ndrg1", "Gm10171", "Tnfrsf9", "Hist1h1b", "Prss29", "Gm853", "Slc2a1", "Hist1h1c",
               "Hist1h1d", "Hilpda", "Gm20431", "Nmral1", "Hcar2", "Hist1h1e", "Zfp750", "Nos2", "Atg9b", "Hist3h2a")

plot(Venn(list(
  cebpe=genes_cebpe[which(genes_cebpe$cluster_de!="F_rest"),]$gene, 
  #cebpe=res_mm[which(res_mm$class!="neutral"),]$gene,
  cebpe_enhancer=df_shrna12[which(df_shrna12$FDR.x<0.05 | df_shrna12$FDR.y<0.05),]$gene
  #published = published
  )))
df_shrna12[grep("^Cebpe$", df_shrna12$gene),]
df_shrna12[grep("^Cebpa$", df_shrna12$gene),]
#df[grep("^Cebpe$", df$MGIS),]
#df[grep("^Cebpa$", df$MGIS),]

df.melt <- melt(df_shrna12[grep("Cebpa|Cebpe", df_shrna12$gene),c("gene", "EV_1.x", "EV_2.x", "sg2_1.x", "sg2_2.x", "sg5_1.x", "sg5_2.x")])
df.melt$variable <- gsub("\\_.*", "", df.melt$variable)
df.melt <- summarySE(df.melt, measurevar = "value", groupvars = c("gene", "variable"))

gXCebpeEnhancer <- ggplot(df.melt, aes(variable, value)) +
  geom_errorbar(aes(ymin=value-se, ymax=value+se), width=.1, alpha=1) +
  geom_point(shape = 21, size = 2, aes(fill=gene)) +
  #scale_fill_manual(values = c(rep("#000000",12))) +
  geom_line(aes(group=1, color=gene), alpha=1) +
  theme_classic() + theme(legend.position="top", axis.text.x = element_text(angle = 45, hjust = 1)) + ylab("Expr.") +
  facet_wrap(~gene, scales="free")
```

## cebpa, cebpe and e2f1 peak analysis
```{r}

#-----------------------------------------------------------------------#
## cebpe and e2f1 peak distance analysis

test <- genes_cebpe[which(grepl("cebpe", genes_cebpe$distance_cebpe_e2f1_class)),]
dim(test)
test$test_class <- cut2(test$distance_cebpe_e2f1, g=20)
table(test$test_class)

p1 <- ggline(test, x = "test_class", y = "motifScore_Cebpe", add = c("mean_se")) +
  stat_compare_means(label.y = 2.5)
  
#p1 <- ggplot(test, aes(motifScore_Cebpe, color=test_class)) + stat_ecdf() +
#  scale_color_manual(values = rev(colorRampPalette(brewer.pal(8, "RdBu"), bias=1)(8))) + theme_bw()
  #ylim(c(0.7,1))

p2 <- ggplot(test, aes(log(cebpe_lpmmy), color=test_class)) + stat_ecdf() +
  scale_color_manual(values = rev(colorRampPalette(brewer.pal(8, "RdBu"), bias=1)(8))) + theme_bw()

p3 <- ggplot(test, aes(log(e2f_chip), color=test_class)) + stat_ecdf() +
  scale_color_manual(values = rev(colorRampPalette(brewer.pal(8, "RdBu"), bias=1)(8))) + theme_bw()

p4 <- ggplot(test, aes(log(myc_chip), color=test_class)) + stat_ecdf() +
  scale_color_manual(values = rev(colorRampPalette(brewer.pal(8, "RdBu"), bias=1)(8))) + theme_bw()

# test.melt <- test[,c("test_class", "cebpa_gmp", "cebpe_lpmmy")]
# # test.melt$cebpa_gmp <- rank(test.melt$cebpa_gmp)
# # test.melt$cebpe_lpmmy <- rank(test.melt$cebpe_lpmmy)
# ggplot(melt(test.melt), aes(test_class, log(value), fill=variable)) + geom_boxplot()

p5 <- freqPlot(melt(table(genes_cebpe[grepl("B_lpm", genes_cebpe$cluster_deRefined),c("peak_class", "distance_cebpe_e2f1_class")])), "density") + theme_bw()

p6 <- ggplot(melt(table(test[which(grepl("B_lpm", test$cluster_deRefined) & test$de_mm=="up"),c("test_class", "de_mm")])), aes(test_class, value, label=value)) + geom_bar(stat="identity") + geom_text(size = 3, position = position_stack(vjust = 0.5))

p7 <- freqPlot(melt(table(test[,c("cluster_de", "test_class")])), "density") + 
  scale_fill_manual(values = rev(colorRampPalette(brewer.pal(8, "RdBu"), bias=1)(8))) + theme_bw()

# p6 <- freqPlot(melt(table(test[grepl("B_lpm", test$cluster_deRefined),c("test_class", "de_mm")])), "density") + theme_bw()

# freqPlot(melt(table(test[,c("distance_cebpe_e2f1_class", "cebp_class")])), "density") + theme_bw()
# df <- test[grepl("B_lpm", test$cluster_deRefined),c("distance_cebpe_e2f1_class", "cluster_deRefined",
#                                                     "cebpa_lsk", "cebpa_pregm", "cebpa_gmp",
#                                                     "cebpe_lpmmy", "cebpe_mm", "cebpe_bc", "cebpe_pmn")]
# table(df$distance_cebpe_e2f1_class)
# df.melt <- summarySE(melt(df), measurevar = "value", groupvars = c("distance_cebpe_e2f1_class", "variable"))
# ggplot(df.melt, aes(variable, value)) +
#   geom_errorbar(aes(ymin=value-se, ymax=value+se), width=.1, alpha=1) +
#   geom_point(shape = 21, size = 2, aes(fill=distance_cebpe_e2f1_class)) +
#   #scale_fill_manual(values = c(rep("#000000",12))) +
#   geom_line(aes(group=1, color=distance_cebpe_e2f1_class), alpha=1) +
#   theme_classic() + theme(legend.position="top", axis.text.x = element_text(angle = 45, hjust = 1)) + ylab("Expr.") +
#   facet_wrap(~distance_cebpe_e2f1_class)

df <- genes_cebpe[which(grepl("B_lpm", genes_cebpe$cluster_deRefined) & grepl("cebpe", genes_cebpe$distance_cebpe_e2f1_class)),]
dim(df)
attributes(Venn(list(cenl=df[which(df$peak_class=="e2f1+cebpe+nfyb+lin54"),]$gene, up=df[which(df$de_mm=="up"),]$gene,
          overlap=df[which(df$distance_cebpe_e2f1_class=="cebpe_e2f1"),]$gene)))
plot(Venn(list(cenl=df[which(df$peak_class=="e2f1+cebpe+nfyb+lin54"),]$gene, up=df[which(df$de_mm=="up"),]$gene,
          overlap=df[which(df$distance_cebpe_e2f1_class=="cebpe_e2f1"),]$gene)))

gXCebpeE2f1DistAll <- ggarrange(ggarrange(p1, p2, p3, p4, nrow=2, ncol=2),
                                ggarrange(p5, p6, p7, nrow=1, ncol=3), nrow=2)
gXCebpeE2f1DistAll

ggboxplot(genes_cebpe[grepl("A_epm|B_lpm", genes_cebpe$cluster_deRefined),], 
          x="distance_cebpe_e2f1_class", y="motifScore_Cebpe", notch = T, notchwidth = 0.6) +
  theme_bw() +
  stat_compare_means(comparisons = list(c("cebpe", "cebpe_e2f1"), c("cebpe_e2f1", "e2f1")), method.args = list(alternative = "g")) +
  ylab("CEBPE motif score")

p1 <- ggplot(genes_cebpe[grepl("A_epm|B_lpm", genes_cebpe$cluster_deRefined),], aes(motifScore_Cebpe, color=distance_cebpe_e2f1_class)) + stat_ecdf() +
  scale_color_manual(values = rev(colorRampPalette(brewer.pal(4, "Set1"), bias=1)(4))) + theme_bw()

p2 <- ggplot(genes_cebpe[grepl("A_epm|B_lpm", genes_cebpe$cluster_deRefined),], aes(log(cebpe_lpmmy), color=distance_cebpe_e2f1_class)) + stat_ecdf() +
  scale_color_manual(values = rev(colorRampPalette(brewer.pal(4, "Set1"), bias=1)(4))) + theme_bw()

p3 <- ggplot(genes_cebpe[grepl("A_epm|B_lpm", genes_cebpe$cluster_deRefined),], aes(log(e2f_chip), color=distance_cebpe_e2f1_class)) + stat_ecdf() +
  scale_color_manual(values = rev(colorRampPalette(brewer.pal(4, "Set1"), bias=1)(4))) + theme_bw()

p4 <- ggplot(genes_cebpe[grepl("A_epm|B_lpm", genes_cebpe$cluster_deRefined),], aes(log(myc_chip), color=distance_cebpe_e2f1_class)) + stat_ecdf() +
  scale_color_manual(values = rev(colorRampPalette(brewer.pal(4, "Set1"), bias=1)(4))) + theme_bw()

p5 <- ggplot(genes_cebpe[which(grepl("A_epm|B_lpm", genes_cebpe$cluster_deRefined) & genes_cebpe$cell_cycle=="cell_cycle"),],
       aes(log2FC_mm, color=distance_cebpe_e2f1_class)) + stat_ecdf() +
  scale_color_manual(values = rev(colorRampPalette(brewer.pal(4, "Set1"), bias=1)(4))) + theme_bw()

p6 <- freqPlot(melt(table(genes_cebpe[grepl("A_epm|B_lpm", genes_cebpe$cluster_deRefined),c("cluster_deRefined", "distance_cebpe_e2f1_class")])), "density") + theme_bw()

gXCebpeE2f1Dist <- ggarrange(p1, p2, p3, p4, p5, p6, nrow=3, ncol=2)

#-----------------------------------------------------------------------#
## percentage of TF peaks overlapping with promoters
mat <- as.data.frame(matrix(c(890, 2532, 4448, 16248, 2844, 25420, 3802, 41057, 1008, 21321), ncol=2, byrow = T))
colnames(mat) <- c("promoter", "distal")
mat$tf <- c("myc", "e2f1", "cebpe", "pu1", "cebpa")
mat$tf <- factor(mat$tf, levels=c("myc", "e2f1", "cebpe", "pu1", "cebpa"), ordered=T)
mat$percentage <- (mat$promoter*100)/mat$distal
ggplot(mat, aes(tf, percentage)) + geom_bar(stat="identity") + theme_bw()

#-----------------------------------------------------------------------#
## analysis to characterize cebpa and cebpe peaks

df <- read.table("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/cebpe_e2f1_distanceAll")
colnames(df) <- c("chr", "start", "end", "gene", "score", "strand", "cebp_classPeaks", "dist_to_tss", "dist_to_e2f1", 
                  "cebpa_gmp_rep1", "cebpa_gmp_rep2", "cebpe_lpmmy_rep1", "cebpe_lpmmy_rep2", "cebpa_p30_rep1", "cebpa_p30_rep2", "p30_bound")
df$cebp_classPeaksLocation <- "promoter"
df[which(df$dist_to_tss>=1000),]$cebp_classPeaksLocation <- "enhancer"
df$cebp_classPeaks <- gsub(",cebpa", "", df$cebp_classPeaks)
df[which(df$cebp_classPeaks=="cebpe"),]$cebp_classPeaks <- "cebpe_cebpa"
df[which(df$cebp_classPeaks=="cebpe_cebpa" & df$cebp_classPeaksLocation=="enhancer" & 
           df$cebpa_gmp_rep1<1 & df$cebpa_gmp_rep2<1),]$cebp_classPeaks <- "cebpe"
table(df$cebp_classPeaks)
df$cebp_classPeaksBroad <- sprintf("%s_%s", df$cebp_classPeaks, df$cebp_classPeaksLocation)
table(df$cebp_classPeaksBroad)
df$id <- sprintf("REGION%d", seq(1, nrow(df), by=1))
df$p30_bound <- ifelse(df$p30_bound>0, "Y", "N")

# hist(log(df[which(df$cebp_classPeaksLocation=="enhancer"),]$dist_to_tss), xlab="distance to tss (log)", main="distribution of distance between enhancer and target promoter defined using closest-gene approach")
# nrow(df[which(df$cebp_classPeaksLocation=="enhancer" & log(df$dist_to_tss)<12),])/nrow(df[which(df$cebp_classPeaksLocation=="enhancer"),])

write.table(df[,c(2:4,1,6,7,5,9,10,19,20)], "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/xls/data_cebpAE.xls", sep="\t", quote = F, row.names = F, col.names = T)

motifAna <- read.table("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/cebpa_cebpe/refined/motifAna/REGIONS_INTEREST.find.unique", header=T, sep="\t")
df <- merge(df, motifAna[grep("CEBPE/jaspar", motifAna$Motif.Name),c("PositionID", "MotifScore")], by.x="id", by.y="PositionID", all=T)
colnames(df)[ncol(df)] <- "motifScore_CEBPE"
df[is.na(df$motifScore_CEBPE),]$motifScore_CEBPE <- 0

df <- merge(df[,c("gene", "cebp_classPeaks", "cebp_classPeaksLocation", "cebp_classPeaksBroad", "dist_to_tss", "dist_to_e2f1", "cebpa_gmp_rep1", "cebpa_gmp_rep2", "cebpe_lpmmy_rep1", "cebpe_lpmmy_rep2", "cebpa_p30_rep1", "cebpa_p30_rep2", "p30_bound", "motifScore_CEBPE")], genes_cebpe, by.x="gene", by.y="gene", all.x=T)
df$cebp_classPeaks <- factor(df$cebp_classPeaks, levels=c("cebpa", "cebpe_cebpa", "cebpe"), ordered = T)
df$dist_to_e2f1 <- df$dist_to_e2f1+1
df$dist_to_tss <- df$dist_to_tss+1
table(df$cebp_classPeaks)
table(df$cebp_classPeaksLocation)
table(df$cebp_classPeaksBroad)
table(df$p30_bound)

p1 <- ggboxplot(df, x="cebp_classPeaks", y="dist_to_e2f1", notch=T, notchwdth=0.6, yscale="log2") + 
  stat_compare_means(comparisons = list(c("cebpe", "cebpe_cebpa"), c("cebpe", "cebpa")), method.args = list(alternative = "l"))

p2 <- ggboxplot(df, x="cebp_classPeaks", y="dist_to_tss", notch=T, notchwdth=0.6, yscale="log2") + 
  stat_compare_means(comparisons = list(c("cebpe", "cebpe_cebpa"), c("cebpe", "cebpa")), method.args = list(alternative = "l"))

p3 <- freqPlot(melt(table(genes_cebpe[which(genes_cebpe$cluster_de!="F_rest"),c("cluster_de", "cebp_closestPeakClass")])), "density") + theme_bw() + ggtitle("") +
  scale_fill_manual(values=c("#fbb4ae", "#decbe4", "#b3cde3", "#ccebc5")) +
  theme(legend.position="top", axis.text.x = element_text(angle = 45, hjust = 1))

p4 <- ggplot(melt(genes_cebpe[,c("cebpa_gmp", "cebpe_lpmmy", "cebp_class")]), aes(cebp_class, log(value))) + geom_boxplot() + facet_grid(.~variable) + theme_bw()

p5 <- freqPlot(melt(table(df[,c("cluster_de", "cebp_classPeaksBroad")])), "desnity", title="") + theme_bw() +
  scale_fill_manual(values=c("#fbb4ae", "#decbe4", "#b3cde3", "#ccebc5", "#ffffb3"))

df.melt <- melt(df[,c("cebp_classPeaksLocation", "cebpa_gmp_rep1", "cebpa_gmp_rep2", "cebpe_lpmmy_rep1", "cebpe_lpmmy_rep2"),])
p6 <- ggboxplot(df.melt, x="cebp_classPeaksLocation", y="value", notch=T, notchwdth=0.6, yscale="log2") +
  stat_compare_means(comparisons = list(c("enhancer", "promoter"))) +
  facet_grid(.~variable)

p7 <- freqPlot(melt(table(df[,c("cebp_classPeaksBroad", "p30_bound")])), "density", title="p30_bound")
p8 <- freqPlot(melt(table(df[which(df$cluster_de!="F_rest"),c("p30_bound", "cluster_de")])), "density", title="p30_bound")
# freqPlot(melt(table(df[,c("cebp_classPeaks", "p30_bound")])), "density", title="p30_bound") + theme_bw()
# freqPlot(melt(table(df[which(df$cebp_classPeaks=="cebpe" & df$cluster_de!="F_rest"),c("p30_bound", "cluster_de")])), "density", title="p30_bound") + theme_bw()

gXCebpAEPeaks <- ggarrange(ggarrange(p1, p2, p3, nrow=1, ncol=3),
                           ggarrange(p4, p5, p6, nrow=1, ncol=3),
                           ggarrange(p7, p8, nrow=1, ncol=3), nrow=3)
gXCebpAEPeaks

# test <- unique(df[which(df$cluster_de!="F_rest"),c("gene", "cebp_classPeaksBroad", "cluster_de")])
# test <- test[order(test$gene, test$cebp_classPeaksBroad, decreasing = F),]
# test <- test[match(unique(test$gene), test$gene),]
# table(test$cebp_classPeaksBroad)
# table(genes_cebpe$cluster_de)
# table(test$cluster_de)
# piePlot(melt(table(test[,c("cebp_classPeaksBroad", "cluster_de")])), title = "") + scale_fill_manual(values=c("#a6611a", "#dfc27d", "#7570b3", "#80cdc1", "#018571"))

# df.melt <- melt(df[which(df$cluster_de!="F_rest"),c("cebp_classPeaksBroad", colnames(df[,c(41:55)]))])
# ggplot(df.melt, aes(variable, log(value))) + geom_boxplot() + facet_grid(.~cebp_classPeaksBroad) + theme_bw()

# test <- (df[which(df$cluster_de!="F_rest"),c("gene", "cebp_classPeaksBroad", "cluster")])
# piePlot(melt(table(test[which(test$cebp_classPeaksBroad=="cebpa_enhancer"),c("cebp_classPeaksBroad", "cluster")])), title = "") +
#   scale_fill_manual(values=colorBrewer2palette(name="Spectral", count = 15))

test <- (df[which(df$cluster_de!="F_rest"),c("gene", "cebp_classPeaksBroad", "cluster_de")])
gXCebpAEPeaksGeneClass <- ggarrange(piePlot(melt(table(test[which(test$cebp_classPeaksBroad=="cebpa_enhancer"),c("cebp_classPeaksBroad", "cluster_de")])), title = "") + scale_fill_manual(values=c("#a6611a", "#dfc27d", "#7570b3", "#80cdc1", "#018571")),
          piePlot(melt(table(test[which(test$cebp_classPeaksBroad=="cebpa_promoter"),c("cebp_classPeaksBroad", "cluster_de")])), title = "") + scale_fill_manual(values=c("#a6611a", "#dfc27d", "#7570b3", "#80cdc1", "#018571")),
          piePlot(melt(table(test[which(test$cebp_classPeaksBroad=="cebpe_cebpa_enhancer"),c("cebp_classPeaksBroad", "cluster_de")])), title = "") + scale_fill_manual(values=c("#a6611a", "#dfc27d", "#7570b3", "#80cdc1", "#018571")),
          piePlot(melt(table(test[which(test$cebp_classPeaksBroad=="cebpe_cebpa_promoter"),c("cebp_classPeaksBroad", "cluster_de")])), title = "") + scale_fill_manual(values=c("#a6611a", "#dfc27d", "#7570b3", "#80cdc1", "#018571")), 
          piePlot(melt(table(test[which(test$cebp_classPeaksBroad=="cebpe_enhancer"),c("cebp_classPeaksBroad", "cluster_de")])), title = "") + scale_fill_manual(values=c("#a6611a", "#dfc27d", "#7570b3", "#80cdc1", "#018571")), nrow=3, ncol=2)

gXCebpAEPeaksGeneClass

# test <- (df[,c("gene", "cebp_classPeaksBroad", "cluster_deRefined")])
# test$cluster_deRefined <- gsub("-de", "", gsub("-neutral", "", test$cluster_deRefined))
# ggarrange(piePlot(melt(table(test[which(test$cebp_classPeaksBroad=="cebpa_enhancer" & test$cluster_deRefined!="F_rest"),c("cebp_classPeaksBroad", "cluster_deRefined")])), title = "") + scale_fill_manual(values=c("#a6611a", "#dfc27d", "#7570b3", "#80cdc1", "#018571")),
#           piePlot(melt(table(test[which(test$cebp_classPeaksBroad=="cebpa_promoter" & test$cluster_deRefined!="F_rest"),c("cebp_classPeaksBroad", "cluster_deRefined")])), title = "") + scale_fill_manual(values=c("#a6611a", "#dfc27d", "#7570b3", "#80cdc1", "#018571")),
#           piePlot(melt(table(test[which(test$cebp_classPeaksBroad=="cebpe_cebpa_enhancer" & test$cluster_deRefined!="F_rest"),c("cebp_classPeaksBroad", "cluster_deRefined")])), title = "") + scale_fill_manual(values=c("#a6611a", "#dfc27d", "#7570b3", "#80cdc1", "#018571")),
#           piePlot(melt(table(test[which(test$cebp_classPeaksBroad=="cebpe_cebpa_promoter" & test$cluster_deRefined!="F_rest"),c("cebp_classPeaksBroad", "cluster_deRefined")])), title = "") + scale_fill_manual(values=c("#a6611a", "#dfc27d", "#7570b3", "#80cdc1", "#018571")), 
#           piePlot(melt(table(test[which(test$cebp_classPeaksBroad=="cebpe_enhancer" & test$cluster_deRefined!="F_rest"),c("cebp_classPeaksBroad", "cluster_deRefined")])), title = "") + scale_fill_manual(values=c("#a6611a", "#dfc27d", "#7570b3", "#80cdc1", "#018571")), nrow=3, ncol=2)

gXCebpAEPeaksGeneClass1 <- ggarrange(piePlot(melt(table(test[which(test$cluster_de=="E_mono" & grepl("enhancer", test$cebp_classPeaksBroad)),
                                                             c("cluster_de", "cebp_classPeaksBroad")]))) +
                                       scale_fill_manual(values=c("#b3de69", "#bc80bd", "#fb8072")),
          piePlot(melt(table(test[which(test$cluster_de=="C_gran" & grepl("enhancer", test$cebp_classPeaksBroad)),
                                  c("cluster_de", "cebp_classPeaksBroad")]))) +
            scale_fill_manual(values=c("#b3de69", "#bc80bd", "#fb8072")), nrow=1, ncol=2)

gXCebpAEPeaksGeneClass1

#-----------------------------------------------------------------------#
## analysis to compare cebpa and cebpe binding at four classes defined based on their binding dynamics during hematopoiesis

df$cebp_classPeaksBroad <- factor(df$cebp_classPeaksBroad, levels=c("cebpa_enhancer", "cebpa_promoter", 
                                                                    "cebpe_cebpa_enhancer", "cebpe_cebpa_promoter",
                                                                    "cebpe_enhancer"),
                                  ordered = T)
table(df$cebp_classPeaksBroad)
p1 <- ggboxplot(df, x="cebp_classPeaksBroad", y="cebpa_gmp_rep1", notch=T, notchwdth=0.6, yscale="log2") +
  stat_compare_means(comparisons = list(c("cebpa_enhancer", "cebpe_cebpa_enhancer"), 
                                        c("cebpa_promoter", "cebpe_cebpa_promoter"),
                                        c("cebpa_enhancer", "cebpe_enhancer")), method.args = list(alternative = "g"))

p2 <- ggboxplot(df, x="cebp_classPeaksBroad", y="cebpe_lpmmy_rep1", notch=T, notchwdth=0.6, yscale="log2") +
  stat_compare_means(comparisons = list(c("cebpa_enhancer", "cebpe_cebpa_enhancer"), 
                                        c("cebpa_promoter", "cebpe_cebpa_promoter"),
                                        c("cebpa_enhancer", "cebpe_enhancer")), method.args = list(alternative = "l"))

gXCebpAEPeaksSignal <- ggarrange(p1, p2, ncol=2, nrow=1)

#-----------------------------------------------------------------------#
## analysis to understand relationship between cebpa and cebpe peaks and the cebpe motif, and gene cluster classes

test <- df
test$test_class <- cut2(test$cebpa_gmp_rep1, m=30)
test$p30_cebpa_fc <- rowMeans(test[,c("cebpa_p30_rep1", "cebpa_p30_rep2")])/rowMeans(test[,c("cebpa_gmp_rep1", "cebpa_gmp_rep2")])
test.melt <- melt(summaryBy(motifScore_CEBPE ~ test_class + cebp_classPeaksBroad, data=test, FUN=list(mean)))

# test$test_class <- 0
# p <- lapply(unique(test$cebp_classPeaksBroad), function(x) {
#   test[which(test$cluster_de!="F_rest" & test$cebp_classPeaksBroad==x),]$test_class <- cut2(test[which(test$cluster_de!="F_rest" & test$cebp_classPeaksBroad==x),]$cebpa_gmp_rep1, m=30);
#   freqPlot(melt(table(test[which(test$cluster_de!="F_rest" & test$cebp_classPeaksBroad==x),c("test_class", "cluster_de")])), "density", title=x)
# })
# ggarrange(arrangeGrob(grobs=p, nrow=2, ncol=2))

p1 <- ggplot(test.melt, aes(test_class, variable, fill=(value))) + geom_tile() + 
  scale_fill_gradientn(colours = rev(colorRampPalette(brewer.pal(11, "RdBu"), bias=0.5)(256))) +
  theme(axis.text.x = element_text(size=10, angle=45)) + 
  facet_grid(.~cebp_classPeaksBroad)

p2 <- ggline(df, x = "cebp_classPeaksBroad", y = "motifScore_CEBPE", add = c("mean_se")) +
  stat_compare_means(label.y = 2.5)
#p2 <- ggplot(df, aes(motifScore_CEBPE, color=cebp_classPeaksBroad)) + stat_ecdf(geom = "step", pad = T) + theme_bw()

# test$motifScore_CEBPE <- cut(df$motifScore_CEBPE, breaks=c(-1,6,7,8,9,10,11))
# df.melt <- as.data.frame(table(test[,c("motifScore_CEBPE", "cebp_classPeaksBroad")]))
# df.melt$total <- c(rep(7461,6), rep(553,6), rep(22613,6), rep(5643,6))
# df.melt$density <- df.melt$Freq/df.melt$total
# df.melt$cumsum <- as.vector(t(summaryBy(density ~ cebp_classPeaksBroad, data = df.melt, FUN=list(cumsum))[,c(2:7)]))
# ggplot(df.melt, aes(motifScore_CEBPE, cumsum, color=cebp_classPeaksBroad)) + geom_line(aes(group=cebp_classPeaksBroad)) + theme_bw() + ylim(c(0,1))
# 
# dat.binned = df %>% count(score=cut(motifScore_CEBPE,c(-1,seq(6,11,by=0.05)))) %>%
#          mutate(pct = n/sum(n))
# ggplot(dat.binned, aes(score, cumsum(pct))) + geom_step(aes(group=1)) + scale_y_continuous()  + ylim(c(0,1))

test.melt <- melt(summaryBy(p30_cebpa_fc ~ test_class + cebp_classPeaksBroad, data=test, FUN=list(mean)))
p3 <- ggplot(test.melt, aes(test_class, variable, fill=log2(value))) + geom_tile() + 
  scale_fill_gradientn(colours = rev(colorRampPalette(brewer.pal(11, "RdBu"), bias=1.8)(256))) +
  theme(axis.text.x = element_blank()) + 
  facet_grid(.~cebp_classPeaksBroad)

gXCebpAEPeakMotifs <- ggarrange(p1, p2, p3, nrow=3, ncol=1)
gXCebpAEPeakMotifs

#---------------------------------------------------------------------#
# Compare Cebpa and Cebpe binding profile for the five gene clusters
df <- genes_cebpe[which(genes_cebpe$cluster_de!="F_rest"),c("cluster_deRefined", "cebpa_lsk", "cebpa_pregm", "cebpa_gmp",
                                                            "cebpe_lpmmy", "cebpe_mm", "cebpe_bc", "cebpe_pmn")]
df.melt <- summarySE(melt(df), measurevar = "value", groupvars = c("cluster_deRefined", "variable"))
gXCebpAEPeaksPromoter <- ggplot(df.melt, aes(variable, value)) +
  geom_errorbar(aes(ymin=value-se, ymax=value+se), width=.1, alpha=1) +
  geom_point(shape = 21, size = 2, aes(fill=cluster_deRefined)) +
  #scale_fill_manual(values = c(rep("#000000",12))) +
  geom_line(aes(group=1, color=cluster_deRefined), alpha=1) +
  theme_classic() + theme(legend.position="top", axis.text.x = element_text(angle = 45, hjust = 1)) + ylab("Expr.") +
  facet_wrap(~cluster_deRefined)

df <- genes_cebpe[which(genes_cebpe$cluster_de!="F_rest"),c("cluster_deRefined", "cebpa_lsk_dis", "cebpa_pregm_dis", "cebpa_gmp_dis",
                                                            "cebpe_lpmmy_dis", "cebpe_mm_dis", "cebpe_bc_dis", "cebpe_pmn_dis")]
df.melt <- summarySE(melt(df), measurevar = "value", groupvars = c("cluster_deRefined", "variable"))
gXCebpAEPeaksEnhancer <- ggplot(df.melt, aes(variable, value)) +
  geom_errorbar(aes(ymin=value-se, ymax=value+se), width=.1, alpha=1) +
  geom_point(shape = 21, size = 2, aes(fill=cluster_deRefined)) +
  #scale_fill_manual(values = c(rep("#000000",12))) +
  geom_line(aes(group=1, color=cluster_deRefined), alpha=1) +
  theme_classic() + theme(legend.position="top", axis.text.x = element_text(angle = 45, hjust = 1)) + ylab("Expr.") +
  facet_wrap(~cluster_deRefined)

df <- genes_cebpe[which(grepl("A_epm|B_lpm", genes_cebpe$cluster_deRefined)),c("funcClass", "cluster_deRefined",
                                                                               "cebpa_lsk", "cebpa_pregm", 
                                                                               "cebpa_gmp", "cebpe_lpmmy", "cebpe_mm", "cebpe_bc", "cebpe_pmn")]
df$funcClass <- as.character(df$funcClass)
df[is.na(df$funcClass),]$funcClass <- "bkg"
df[which(df$funcClass!="G1_S" & df$funcClass!="G2_M"),]$funcClass <- "bkg"
table(df$funcClass)
df.melt <- summarySE(melt(df), measurevar = "value", groupvars = c("funcClass", "variable"))
gXCebpAEPeaksCellCycle <- ggplot(df.melt, aes(variable, value)) +
  geom_errorbar(aes(ymin=value-se, ymax=value+se), width=.1, alpha=1) +
  geom_point(shape = 21, size = 2, aes(fill=funcClass)) +
  #scale_fill_manual(values = c(rep("#000000",12))) +
  geom_line(aes(group=1, color=funcClass), alpha=1) +
  theme_classic() + theme(legend.position="top", axis.text.x = element_text(angle = 45, hjust = 1)) + ylab("Expr.") +
  facet_wrap(~funcClass)
```

## plot results showing analysis on cebpa and cebpe regulation, expression and binding dynamics
```{r}
pdf("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/pdf/analysis_cebpAE.pdf")
gXCebpeEnhancer
gXCebpAEPeaks
gXCebpAEPeaksGeneClass
gXCebpAEPeaksGeneClass1
gXCebpAEPeaksSignal
gXCebpAEPeakMotifs
gXCebpAEPeaksPromoter
gXCebpAEPeaksEnhancer
gXCebpAEPeaksCellCycle
gXExprProfileCebpAE
gXCebpeE2f1DistAll
gXCebpeE2f1Dist
dev.off()
```

#### gviz tracks for regions around CEBPA and CEBPE genes
```{r}
pdf("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/pdf/gviz.pdf")
browser_files <- read.table(pipe("grep -v '#\\|gmpG' /home/pundhir/project/chip-seq-analysis/analysis_cebpe/FILES_FOR_GENOME_BROWSER | grep -v 'cebpa_my12' | grep -v 'cebpa_mm'"))
colnames(browser_files) <- c("id", "bam_files")

# CEBPE (+6 kb enhancer)
cebpe <- plot_gviz_track(GEN = "mm9", COOR = "chr14:55323682-55332011",
                        BAMFILE = browser_files[grep("cebp.*Rep1|myc_esc|max_esc|e2f1_esc_Rep1", browser_files$id),]$bam_files, 
                        type = "a", maxlimit = "500,500,500,500,500,500,500,500,100,20,20",
                        BEDFILE=c("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/E2f1_mm9_lifted.bed",
                                  "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/cMyc_mm9_lifted.bed",
                                  "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.E2F1"),
                        cex = 0.5)

cebpe_h3k27ac <- plot_gviz_track(GEN = "mm9", COOR = "chr14:55323682-55332011",
                        BAMFILE = browser_files[grep("h3k27ac.*Rep1", browser_files$id),]$bam_files,
                        type = "a", maxlimit = 60)

# CEBPA (+37 kb enhancer)
cebpa <- plot_gviz_track(GEN = "mm9", COOR = "chr7:35891494-35950004",
                        BAMFILE = browser_files[grep("cebp.*Rep1|myc_esc|max_esc|e2f1_esc_Rep1", browser_files$id),]$bam_files, 
                        type = "a", maxlimit = "150,150,150,150,150,150,150,150,20,10,20",
                        BEDFILE=c("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/E2f1_mm9_lifted.bed",
                                  "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/cMyc_mm9_lifted.bed",
                                  "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.E2F1",
                                  "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.CEBPE",
                                  "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/E2f1_mm9_source2.bed.ucsc"),
                       cex = 0.5)

cebpa_h3k27ac <- plot_gviz_track(GEN = "mm9", COOR = "chr7:35891494-35950004",
                        BAMFILE = browser_files[grep("h3k27ac.*Rep1", browser_files$id),]$bam_files,
                        type = "a", maxlimit = 60)

## Myc (chr15:61811875-61826937 or chr15:61815896-61822916)
myc <- plot_gviz_track(GEN = "mm9", COOR = "chr15:61811875-61826937",
                       BAMFILE = browser_files[grep("cebp.*Rep2|control|myc_esc|e2f1_esc_Rep1", browser_files$id),]$bam_files,
                       type = "a", maxlimit = "150,150,150,150,150,150,150,150,150,150",
                       BEDFILE=c("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/E2f1_mm9_lifted.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/cMyc_mm9_lifted.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/nfyb_esc_ucsc.bed"),
                       cex = 0.5
                       )

myc1 <- plot_gviz_track(GEN = "mm9", COOR = "chr15:61815896-61822916",
                       BAMFILE = browser_files[grep("cebp.*Rep2|control|myc_esc|e2f1_esc_Rep1", browser_files$id),]$bam_files,
                       type = "a", maxlimit = "30,30,30,30,30,30,30,30,100,100",
                       BEDFILE=c("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/E2f1_mm9_lifted.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/cMyc_mm9_lifted.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/nfyb_esc_ucsc.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.NFYB",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.LIN54",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/motifAna_selGenes/motifAna/myc/REGIONS_INTEREST.annoBed"),
                       cex = 0.5
                       )

myc_expr <- plot_gviz_track(GEN = "mm9", COOR = "chr15:61811875-61826937",
                       BAMFILE = browser_files[grep("epm_|lpm_", browser_files$id),]$bam_files,
                       type = "a", maxlimit = 700,
                       cex = 0.5
                       )

## Mybl2 (BMYB)
myb <- plot_gviz_track(GEN = "mm9", COOR = "chr2:162872857-162917936",
                       BAMFILE = browser_files[grep("cebp.*Rep2|control", browser_files$id),]$bam_files,
                       type = "a", maxlimit = 50,
                       BEDFILE=c("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/E2f1_mm9_lifted.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/cMyc_mm9_lifted.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/nfyb_esc_ucsc.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.NFYB",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.LIN54"),
                       cex = 0.5
                       )

## Mxd1 (chr5:140465299-140802079)
mad1 <- plot_gviz_track(GEN = "mm9", COOR = "chr6:86617639-86620223",
                        BAMFILE = browser_files[grep("cebp.*Rep1|myc_esc|e2f1_esc_Rep1", browser_files$id),]$bam_files, 
                        type = "a", maxlimit = "400,400,400,400,400,400,400,400,40,150",
                        BEDFILE=c("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/E2f1_mm9_lifted.bed",
                                  "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/cMyc_mm9_lifted.bed",
                                  "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.E2F1",
                                  "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.CEBPE",
                                  "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/E2f1_mm9_source2.bed.ucsc"),
                       cex = 0.5)

## Max (chr12:78032011-78069480)
max <- plot_gviz_track(GEN = "mm9", COOR = "chr12:78062295-78064304",
                        BAMFILE = browser_files[grep("cebp.*Rep1|myc_esc|e2f1_esc_Rep1", browser_files$id),]$bam_files, 
                        type = "a", maxlimit = "40,40,40,40,40,40,40,40,40,100",
                        BEDFILE=c("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/E2f1_mm9_lifted.bed",
                                  "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/cMyc_mm9_lifted.bed",
                                  "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.E2F1",
                                  "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/E2f1_mm9_source2.bed.ucsc"),
                       cex = 0.5)

## Foxm1
foxm1 <- plot_gviz_track(GEN = "mm9", COOR = "chr6:128309788-128329127",
                       BAMFILE = browser_files[grep("cebp.*Rep2|control", browser_files$id),]$bam_files,
                       type = "a", maxlimit = 50,
                       BEDFILE=c("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/E2f1_mm9_lifted.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/cMyc_mm9_lifted.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/nfyb_esc_ucsc.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.NFYB",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.LIN54"),
                       cex = 0.5
                       )

## Lin54 (chr5:100856924-100941726)
lin54 <- plot_gviz_track(GEN = "mm9", COOR = "chr5:100925190-100930401",
                       BAMFILE = browser_files[grep("cebp.*Rep1", browser_files$id),]$bam_files,
                       type = "a", maxlimit = 200,
                       BEDFILE=c("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/E2f1_mm9_lifted.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/cMyc_mm9_lifted.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/nfyb_esc_ucsc.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.NFYB",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.LIN54")
                       )

## Nfyb (chr10:82205984-82235677)
nfyb <- plot_gviz_track(GEN = "mm9", COOR = "chr10:82222905-82227799",
                       BAMFILE = browser_files[grep("cebp.*Rep1", browser_files$id),]$bam_files,
                       type = "a", maxlimit = 70,
                       BEDFILE=c("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/E2f1_mm9_lifted.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/cMyc_mm9_lifted.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/nfyb_esc_ucsc.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.NFYB",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.LIN54")
                       )

## Mmp9
mmp9 <- plot_gviz_track(GEN = "mm9", COOR = "chr2:164761367-164783160",
                       BAMFILE = browser_files[grep("cebp.*Rep1", browser_files$id),]$bam_files,
                       type = "a", maxlimit = 400,
                       BEDFILE=c("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/E2f1_mm9_lifted.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/cMyc_mm9_lifted.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/nfyb_esc_ucsc.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.NFYB",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.LIN54"
                                 )
                       )

## Skp1a (Skp1a: chr11:52040989-52064301; Skp2: chr15:9040659-9086437)
skp2 <- plot_gviz_track(GEN = "mm9", COOR = "chr15:9034626-9077324",
                       BAMFILE = browser_files[grep("cebp.*Rep1", browser_files$id),]$bam_files,
                       type = "a", maxlimit = 50,
                       BEDFILE=c("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/E2f1_mm9_lifted.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/cMyc_mm9_lifted.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/nfyb_esc_ucsc.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.NFYB",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.LIN54"
                                 )
                       )


## Ltf
ltf <- plot_gviz_track(GEN = "mm9", COOR = "chr9:110915927-110951139",
                       BAMFILE = browser_files[grep("cebp.*Rep1", browser_files$id),]$bam_files,
                       type = "a", maxlimit = 200,
                       BEDFILE=c("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/E2f1_mm9_lifted.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/cMyc_mm9_lifted.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/nfyb_esc_ucsc.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.NFYB",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.LIN54"
                                 )
                       )

## E2f1
e2f1 <- plot_gviz_track(GEN = "mm9", COOR = "chr2:154382822-154398142",
                       BAMFILE = browser_files[grep("cebp.*Rep1", browser_files$id),]$bam_files,
                       type = "a", maxlimit = 20,
                       BEDFILE=c("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/E2f1_mm9_lifted.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/cMyc_mm9_lifted.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/nfyb_esc_ucsc.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.NFYB",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.LIN54"
                                 )
                       )

## Trp53 (chr11:69390982-69408254)
trp53 <- plot_gviz_track(GEN = "mm9", COOR = "chr11:69390982-69408254",
                       BAMFILE = browser_files[grep("cebp.*Rep1", browser_files$id),]$bam_files,
                       type = "a", maxlimit = 200,
                       BEDFILE=c("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/E2f1_mm9_lifted.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/cMyc_mm9_lifted.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/nfyb_esc_ucsc.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.NFYB",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.LIN54"
                                 )
                       )

## Prkdc (chr16:15637626-15652417) - Mcm4 gene (reverse strand)
prkdc <- plot_gviz_track(GEN = "mm9", COOR = "chr16:15637626-15652417",
                       BAMFILE = browser_files[grep("cebp.*Rep1", browser_files$id),]$bam_files,
                       type = "a", maxlimit = 200,
                       BEDFILE=c("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/E2f1_mm9_lifted.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/cMyc_mm9_lifted.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/nfyb_esc_ucsc.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.NFYB",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.LIN54"
                                 )
                       )

## Atm (chr9:53325580-53343260) - Npat gene (reverse strand)
atm <- plot_gviz_track(GEN = "mm9", COOR = "chr9:53341437-53344958",
                       BAMFILE = browser_files[grep("cebp.*Rep1", browser_files$id),]$bam_files,
                       type = "a", maxlimit = 200,
                       BEDFILE=c("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/E2f1_mm9_lifted.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/cMyc_mm9_lifted.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/nfyb_esc_ucsc.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.NFYB",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.LIN54"
                                 )
                       )

## Rb1 (chr14:73562737-73758171)
rb1 <- plot_gviz_track(GEN = "mm9", COOR = "chr14:73722371-73729787",
                        BAMFILE = browser_files[grep("cebp.*Rep1|myc_esc|e2f1_esc_Rep1", browser_files$id),]$bam_files, 
                        type = "a", maxlimit = "150,150,150,150,150,150,150,150,150,50",
                        BEDFILE=c("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/E2f1_mm9_lifted.bed",
                                  "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/cMyc_mm9_lifted.bed",
                                  "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.E2F1",
                                  "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.CEBPE",
                                  "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/E2f1_mm9_source2.bed.ucsc"),
                       cex = 0.5)

## E2f7 (chr10:110172041-110234920)
e2f7 <- plot_gviz_track(GEN = "mm9", COOR = "chr10:110172041-110234920",
                        BAMFILE = browser_files[grep("cebp.*Rep1|myc_esc|e2f1_esc_Rep1", browser_files$id),]$bam_files, 
                        type = "a", maxlimit = "50,50,50,50,50,50,50,50,50,20",
                        BEDFILE=c("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/E2f1_mm9_lifted.bed",
                                  "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/cMyc_mm9_lifted.bed",
                                  "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.E2F1",
                                  "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/E2f1_mm9_source2.bed.ucsc"),
                       cex = 0.5)

## E2f8 (chr7:56118145-56140064)
e2f8 <- plot_gviz_track(GEN = "mm9", COOR = "chr7:56118145-56140064",
                        BAMFILE = browser_files[grep("cebp.*Rep1|myc_esc|e2f1_esc_Rep1", browser_files$id),]$bam_files, 
                        type = "a", maxlimit = "50,50,50,50,50,50,50,50,50,20",
                        BEDFILE=c("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/E2f1_mm9_lifted.bed",
                                  "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/cMyc_mm9_lifted.bed",
                                  "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.E2F1",
                                  "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/E2f1_mm9_source2.bed.ucsc"),
                       cex = 0.5)
dev.off()
#primary_tracks <- create_primary_track("mm9", "chr14:55324682-55331011")
#data_tracks <- create_data_track("mm9", "chr14:55324682-55331011", browser_files[grep("cebpa", browser_files$bam_files),][1], "a", 200)
#plotTracks(c(primary_tracks$itrack, primary_tracks$gtrack, primary_tracks$txtr, data_tracks), from=55324682, to=55331011)

##--------------------------------------------------------------------------##
## Gviz plots for Cell cycle genes
##--------------------------------------------------------------------------##

pdf("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/pdf/gviz_cell_cycle.pdf")
browser_files <- read.table(pipe("grep -v '#\\|gmpG' /home/pundhir/project/chip-seq-analysis/analysis_cebpe/FILES_FOR_GENOME_BROWSER"))
colnames(browser_files) <- c("id", "bam_files")

## G1/S genes
## Ccnd2 (chr6:127069392-127107401) (up) (cluster 3)
ccnd2 <- plot_gviz_track(GEN = "mm9", COOR = "chr6:127069392-127107401",
                       BAMFILE = browser_files[grep("cebp.*Rep1", browser_files$id),]$bam_files,
                       type = "a", maxlimit = 100,
                       BEDFILE=c("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/E2f1_mm9_lifted.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/cMyc_mm9_lifted.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/nfyb_esc_ucsc.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.NFYB",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.LIN54")
                       )

## Cdk4 (chr10:126499725-126505396) (up) (cluster 4)
## Cdk6 (chr5:3299592-3566752) (down)
cdk4 <- plot_gviz_track(GEN = "mm9", COOR = "chr10:126499725-126505396",
                       BAMFILE = browser_files[grep("cebp.*Rep1|myc_esc|e2f1_esc_Rep1", browser_files$id),]$bam_files,
                       type = "a", maxlimit = "150,150,150,150,150,150,150,150,40,150",
                       BEDFILE=c("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/E2f1_mm9_lifted.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/cMyc_mm9_lifted.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/nfyb_esc_ucsc.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.NFYB",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.LIN54")
                       )

## Cdkn2d (chr9:123836173-123840294) (down) (cluster 10)
cdkn2d <- plot_gviz_track(GEN = "mm9", COOR = "chr9:123836173-123840294",
                       BAMFILE = browser_files[grep("cebp.*Rep1", browser_files$id),]$bam_files,
                       type = "a", maxlimit = 150,
                       BEDFILE=c("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/E2f1_mm9_lifted.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/cMyc_mm9_lifted.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/nfyb_esc_ucsc.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.NFYB",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.LIN54")
                       )

## Cdkn2c (chr4:109332356-109339101) (up) (cluster 9)
cdkn2c <- plot_gviz_track(GEN = "mm9", COOR = "chr4:109332356-109339101",
                       BAMFILE = browser_files[grep("cebp.*Rep1", browser_files$id),]$bam_files,
                       type = "a", maxlimit = 30,
                       BEDFILE=c("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/E2f1_mm9_lifted.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/cMyc_mm9_lifted.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/nfyb_esc_ucsc.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.NFYB",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.LIN54"
                                 )
                       )

## Cdk2 (chr10:128133216-12814388) (neutral)
## Ccne2 (neutral)
## Cdkn1b (chr6:134869137-134876824) (neutral)
## Cdkn1a (chr17:29226372-29242012) (up) (cluster 15)
cdkn1a <- plot_gviz_track(GEN = "mm9", COOR = "chr17:29226372-29242012",
                       BAMFILE = browser_files[grep("cebp.*Rep1", browser_files$id),]$bam_files,
                       type = "a", maxlimit = 30,
                       BEDFILE=c("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/E2f1_mm9_lifted.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/cMyc_mm9_lifted.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/nfyb_esc_ucsc.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.NFYB",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.LIN54"
                                 )
                       )

## Cdkn3 (chr14:47377469-47393946) (down) (cluster 9)
cdkn3 <- plot_gviz_track(GEN = "mm9", COOR = "chr14:47377469-47393946",
                       BAMFILE = browser_files[grep("cebp.*Rep1", browser_files$id),]$bam_files,
                       type = "a", maxlimit = 30,
                       BEDFILE=c("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/E2f1_mm9_lifted.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/cMyc_mm9_lifted.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/nfyb_esc_ucsc.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.NFYB",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.LIN54"
                                 )
                       )

## G2/M
## Ccna2 (chr3:36462004-3647270) (up) (cluster 6)
ccna2 <- plot_gviz_track(GEN = "mm9", COOR = "chr3:36462004-36472701",
                       BAMFILE = browser_files[grep("cebp.*Rep1", browser_files$id),]$bam_files,
                       type = "a", maxlimit = 30,
                       BEDFILE=c("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/E2f1_mm9_lifted.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/cMyc_mm9_lifted.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/nfyb_esc_ucsc.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.NFYB",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.LIN54"
                                 )
                       )

## Ccnb1 (chr13:101547075-101557803) (up) (cluster 6)
## Ccnb2 (chr9:70252030-70272828) (up) (cluster 6)
ccnb1 <- plot_gviz_track(GEN = "mm9", COOR = "chr13:101547075-101557803",
                       BAMFILE = browser_files[grep("cebp.*Rep1|^nfyb|^e2f1_esc_Rep1", browser_files$id),]$bam_files,
                       type = "a", maxlimit = "20,20,20,20,20,20,20,20,200,30",
                       BEDFILE=c("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/E2f1_mm9_lifted.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/cMyc_mm9_lifted.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/nfyb_esc_ucsc.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.NFYB",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.LIN54",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.E2F1"
                                 )
                       )

ccnb1_promoter <- plot_gviz_track(GEN = "mm9", COOR = "chr13:101556173-101557170",
                       BAMFILE = browser_files[grep("cebp.*Rep1|^nfyb|^e2f1_esc_Rep1", browser_files$id),]$bam_files,
                       type = "a", maxlimit = "20,20,20,20,20,20,20,20,200,30",
                       BEDFILE=c("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/E2f1_mm9_lifted.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/cMyc_mm9_lifted.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/nfyb_esc_ucsc.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.NFYB",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.LIN54",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.E2F1",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.CEBPE"
                                 )
                       )

## Cdk1 (chr10:68783105-68831938) (up) (cluster 6)
cdk1 <- plot_gviz_track(GEN = "mm9", COOR = "chr10:68783105-68831938",
                       BAMFILE = browser_files[grep("cebp.*Rep1|^nfyb|^e2f1_esc_Rep1", browser_files$id),]$bam_files,
                       type = "a", maxlimit = "40,40,40,40,40,40,40,40,150,30",
                       BEDFILE=c("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/E2f1_mm9_lifted.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/cMyc_mm9_lifted.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/nfyb_esc_ucsc.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.NFYB",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.LIN54",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.E2F1"
                                 )
                       )

cdk1_promoter <- plot_gviz_track(GEN = "mm9", COOR = "chr10:68815381-68816390",
                       BAMFILE = browser_files[grep("cebp.*Rep1|^nfyb|^e2f1_esc_Rep1", browser_files$id),]$bam_files,
                       type = "a", maxlimit = "80,80,80,80,80,80,80,80,250,60",
                       BEDFILE=c("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/E2f1_mm9_lifted.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/cMyc_mm9_lifted.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/nfyb_esc_ucsc.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.NFYB",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.LIN54",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.E2F1",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.CEBPE"
                                 )
                       )

## Gadd45a (chr6:66984512-66987979) (down) (cluster 11)
gadd45a <- plot_gviz_track(GEN = "mm9", COOR = "chr6:66986910-66988421",
                       BAMFILE = browser_files[grep("cebp.*Rep1", browser_files$id),]$bam_files,
                       type = "a", maxlimit = 200,
                       BEDFILE=c("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/E2f1_mm9_lifted.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/cMyc_mm9_lifted.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/nfyb_esc_ucsc.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.NFYB",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.LIN54")
                       )

## Ccnf (chr17:24350936-24398940) (up) (cluster 9)
ccnf <- plot_gviz_track(GEN = "mm9", COOR = "chr17:24350936-24398940",
                       BAMFILE = browser_files[grep("cebp.*Rep1", browser_files$id),]$bam_files,
                       type = "a", maxlimit = 500,
                       BEDFILE=c("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/E2f1_mm9_lifted.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/cMyc_mm9_lifted.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/nfyb_esc_ucsc.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.NFYB",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.LIN54")
                       )
dev.off()

pdf("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/pdf/gviz_cell_cycle_inhibitors.pdf")
## Cdkn2d (chr9:123836173-123840294; second copy, however GTF file we used for expression quantification has only one copy) (down) (cluster 10)
cdkn2d <- plot_gviz_track(GEN = "mm9", COOR = "chr9:21092422-21093964",
                       BAMFILE = browser_files[grep("cebp.*Rep1", browser_files$id),]$bam_files,
                       type = "a", maxlimit = 200,
                       BEDFILE=c("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/E2f1_mm9_lifted.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/cMyc_mm9_lifted.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/nfyb_esc_ucsc.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.NFYB",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.LIN54")
                       )

## Gadd45a (chr6:66984512-66987979) (down) (cluster 11)
gadd45a <- plot_gviz_track(GEN = "mm9", COOR = "chr6:66986910-66988421",
                       BAMFILE = browser_files[grep("cebp.*Rep1", browser_files$id),]$bam_files,
                       type = "a", maxlimit = 200,
                       BEDFILE=c("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/E2f1_mm9_lifted.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/cMyc_mm9_lifted.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/nfyb_esc_ucsc.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.NFYB",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.LIN54")
                       )
## Cdkn1a (chr17:29226372-29242012) (up) (cluster 15)
cdkn1a <- plot_gviz_track(GEN = "mm9", COOR = "chr17:29226372-29242012",
                       BAMFILE = browser_files[grep("cebp.*Rep1", browser_files$id),]$bam_files,
                       type = "a", maxlimit = 200,
                       BEDFILE=c("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/E2f1_mm9_lifted.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/cMyc_mm9_lifted.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/nfyb_esc_ucsc.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.NFYB",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.LIN54"
                                 )
                       )
dev.off()

##--------------------------------------------------------------------------##
## Gviz plots for Cebpe target genes
##--------------------------------------------------------------------------##

pdf("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/pdf/gviz_cebpe_cripr.pdf")

browser_files <- read.table(pipe("grep -v '#\\|gmpG' /home/pundhir/project/chip-seq-analysis/analysis_cebpe/FILES_FOR_GENOME_BROWSER"))
colnames(browser_files) <- c("id", "bam_files")

##  Card10
# card10 <- plot_gviz_track(GEN = "mm9", COOR = "chr15:78605566-78641472",
#                        BAMFILE = browser_files[grep("cebp.*Rep2|control|h3k27ac", browser_files$id),]$bam_files,
#                        type = "a", maxlimit = "400,400,400,400,400,400,400,400,40,40,40,40,40,40,40,40",
#                        BEDFILE=c("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/ALLPEAKS.BED.SIG.cebpe",
#                                  "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/revision/card10.bed"),
#                        cex = 0.5
#                        )

## hlx
hlx <- plot_gviz_track(GEN = "mm9", COOR = "chr1:186196483-186602253",
                       BAMFILE = browser_files[grep("cebp.*Rep2|control|h3k27ac.*Rep1", browser_files$id),]$bam_files,
                       type = "a", maxlimit = "60,60,60,60,60,60,60,60,20,20,20,20,300,300,300,300,150,150,150,150",
                       BEDFILE=c("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/ALLPEAKS.BED.SIG.cebpe",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/revision/hlx.bed"),
                       cex = 0.5
                       )

## igf1r
# igf1r <- plot_gviz_track(GEN = "mm9", COOR = "chr7:75026790-75448906",
#                        BAMFILE = browser_files[grep("cebp.*Rep2|control|h3k27ac.*Rep1", browser_files$id),]$bam_files,
#                        type = "a", maxlimit = "100,100,100,100,100,100,100,100,20,20,20,20",
#                        BEDFILE=c("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/ALLPEAKS.BED.SIG.cebpe",
#                                  "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/revision/igf1r.bed"),
#                        cex = 0.5
#                        )

## tgfbr2
# tgfbr2 <- plot_gviz_track(GEN = "mm9", COOR = "chr9:115986024-116090171",
#                        BAMFILE = browser_files[grep("cebp.*Rep2|control|h3k27ac.*Rep1", browser_files$id),]$bam_files,
#                        type = "a", maxlimit = "100,100,100,100,100,100,100,100,40,40,40,40",
#                        BEDFILE=c("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/ALLPEAKS.BED.SIG.cebpe",
#                                  "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/revision/tgfbr2.bed"),
#                        cex = 0.5
#                        )

## picalm
# picalm <- plot_gviz_track(GEN = "mm9", COOR = "chr7:97136056-97328847",
#                        BAMFILE = browser_files[grep("cebp.*Rep2|control|h3k27ac.*Rep1", browser_files$id),]$bam_files,
#                        type = "a", maxlimit = "60,60,60,60,60,60,60,60,50,50,50,50",
#                        BEDFILE=c("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/ALLPEAKS.BED.SIG.cebpe",
#                                  "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/revision/picalm.bed"),
#                        cex = 0.5
#                        )

## nt5e
# nt5e <- plot_gviz_track(GEN = "mm9", COOR = "chr9:88172447-88266927",
#                        BAMFILE = browser_files[grep("cebp.*Rep2|control|h3k27ac.*Rep1", browser_files$id),]$bam_files,
#                        type = "a", maxlimit = "60,60,60,60,60,60,60,60,30,30,30,30",
#                        BEDFILE=c("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/ALLPEAKS.BED.SIG.cebpe",
#                                  "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/revision/nt5e.bed"),
#                        cex = 0.5
#                        )

## mmp8
# mmp8 <- plot_gviz_track(GEN = "mm9", COOR = "chr9:7555915-7571001",
#                        BAMFILE = browser_files[grep("cebp.*Rep2|control|h3k27ac.*Rep1", browser_files$id),]$bam_files,
#                        type = "a", maxlimit = "500,500,500,500,500,500,500,500,30,30,30,30",
#                        BEDFILE=c("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/ALLPEAKS.BED.SIG.cebpe",
#                                  "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/revision/mmp8.bed"),
#                        cex = 0.5
#                        )

## ltf
# plot_gviz_track(GEN = "mm9", COOR = "chr9:110915927-110951139",
#                        BAMFILE = browser_files[grep("cebp.*Rep2|control|h3k27ac.*Rep1", browser_files$id),]$bam_files,
#                        type = "a", maxlimit = "400,400,400,400,400,400,400,400,30,30,30,30",
#                        BEDFILE=c("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/ALLPEAKS.BED.SIG.cebpe",
#                                  "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/revision/ltf.bed"),
#                        cex = 0.5
#                        )

## hp
plot_gviz_track(GEN = "mm9", COOR = "chr8:112098016-112104083",
                       BAMFILE = browser_files[grep("cebp.*Rep2|control|h3k27ac.*Rep2", browser_files$id),]$bam_files,
                       type = "a", maxlimit = "800,800,800,800,800,800,800,800,40,40,40,40,1500,1500,1500,1500,500,500,500,500",
                       BEDFILE=c("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/ALLPEAKS.BED.SIG.cebpe",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/revision/hp.bed"),
                       cex = 0.5
                       )

## ngp (chr9:110321510-110326317)
# plot_gviz_track(GEN = "mm9", COOR = "chr9:110321510-110326317",
#                        BAMFILE = browser_files[grep("cebp.*Rep2|control|h3k27ac.*Rep1", browser_files$id),]$bam_files,
#                        type = "a", maxlimit = "800,800,800,800,800,800,800,800,50,50,50,50",
#                        BEDFILE=c("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/ALLPEAKS.BED.SIG.cebpe",
#                                  "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/revision/ngp.bed"),
#                        cex = 0.5
#                        )

## dhrs7 (chr12:73683417-73823033, chr12:73742293-73774863)
plot_gviz_track(GEN = "mm9", COOR = "chr12:73742293-73774863",
                       BAMFILE = browser_files[grep("cebp.*Rep2|control|h3k27ac.*Rep1", browser_files$id),]$bam_files,
                       type = "a", maxlimit = "800,800,800,800,800,800,800,800,100,100,100,100,2000,2000,2000,2000,500,500,500,500",
                       BEDFILE=c("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/ALLPEAKS.BED.SIG.cebpe",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/revision/dhrs7.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/revision/dhrs7_promoter.bed"),
                       cex = 0.5
                       )
# ## Ifitm6 (chr7:148197387-148207115)
# plot_gviz_track(GEN = "mm9", COOR = "chr7:148197387-148207115",
#                        BAMFILE = browser_files[grep("cebp.*Rep2|control|h3k27ac.*Rep1", browser_files$id),]$bam_files,
#                        type = "a", maxlimit = "800,800,800,800,800,800,800,800,50,50,50,50",
#                        BEDFILE=c("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/ALLPEAKS.BED.SIG.cebpe"),
#                        cex = 0.5
#                        )

#^Card10$|^Hlx$|^Tgfbr2$|^Nt5e$|^Mmp8$|^Ltf$|^Hp$|^Ngp$|^Dhrs7$|^Camp$
df.melt <- melt(expr_data_norm_with_replicates[which(grepl("E2f", expr_data_norm_with_replicates$GeneID)),
                               c("GeneID",
                                 grep("lsk_.*wt", colnames(expr_data_norm_with_replicates), value = T),
                                 grep("pregm_.*wt", colnames(expr_data_norm_with_replicates), value = T),
                                 grep("gmp_.*wt|gmpG.*wt", colnames(expr_data_norm_with_replicates), value = T),
                                 grep("gmpG.*wt", colnames(expr_data_norm_with_replicates), value = T),
                                 grep("epm_.*wt", colnames(expr_data_norm_with_replicates), value = T),
                                 grep("lpm_.*wt", colnames(expr_data_norm_with_replicates), value = T),
                                 grep("my_.*wt", colnames(expr_data_norm_with_replicates), value = T),
                                 grep("mm_.*wt", colnames(expr_data_norm_with_replicates), value = T),
                                 grep("bc_.*wt", colnames(expr_data_norm_with_replicates), value = T),
                                 grep("pmn_.*wt", colnames(expr_data_norm_with_replicates), value = T),
                                 grep("gmpG_.*ko", colnames(expr_data_norm_with_replicates), value = T),
                                 grep("epm_.*ko", colnames(expr_data_norm_with_replicates), value = T),
                                 grep("lpm_.*ko", colnames(expr_data_norm_with_replicates), value = T),
                                 grep("mm_.*ko", colnames(expr_data_norm_with_replicates), value = T)
                                 )])
LEVELS <- unique(gsub("_[Rr]ep.*|_cebpa", "", df.melt$variable))
df.melt$variable <- gsub("_[Rr]ep.*|_cebpa", "", df.melt$variable)
df.melt$value <- df.melt$value
df.melt <- summarySE(df.melt, measurevar = "value", groupvars = c("GeneID", "variable"))
df.melt$variable <- factor(df.melt$variable, levels=LEVELS, ordered = T)

ggplot(df.melt, aes(variable, value)) +
  geom_errorbar(aes(ymin=value-se, ymax=value+se), width=.1, alpha=1) +
  geom_point(shape = 21, size = 2, aes(fill=GeneID)) +
  #scale_fill_manual(values = c(rep("#000000",12))) +
  geom_line(aes(group=1, color=GeneID), alpha=1) +
  theme_classic() + theme(legend.position="top", axis.text.x = element_text(angle = 45, hjust = 1)) + ylab("Expr.") +
  facet_wrap(~GeneID, scales="free")
dev.off()


##--------------------------------------------------------------------------##
## Gviz plots for genes shown in the paper (now with CEBPA ChIP-seq data)
##--------------------------------------------------------------------------##
pdf("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/pdf/gviz_revision.pdf")
browser_files <- read.table(pipe("grep -v '#\\|gmpG' /home/pundhir/project/chip-seq-analysis/analysis_cebpe/FILES_FOR_GENOME_BROWSER"))
colnames(browser_files) <- c("id", "bam_files")

cdk4 <- plot_gviz_track(GEN = "mm9", COOR = "chr10:126499725-126505396",
                       BAMFILE = browser_files[grep("cebp.*Rep1|myc_esc|e2f1_esc_Rep1", browser_files$id),]$bam_files,
                       type = "a", maxlimit = "150,150,150,150,150,150,150,150,40,150,700,700,700,700",
                       BEDFILE=c("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/E2f1_mm9_lifted.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/cMyc_mm9_lifted.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/nfyb_esc_ucsc.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.NFYB",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.LIN54")
                       )

myc <- plot_gviz_track(GEN = "mm9", COOR = "chr15:61811875-61826937",
                       BAMFILE = browser_files[grep("cebp.*Rep2|control|myc_esc|e2f1_esc_Rep1", browser_files$id),]$bam_files,
                       type = "a", maxlimit = "150,150,150,150,150,150,150,150,150,150,1500,1500,1500,1500",
                       BEDFILE=c("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/E2f1_mm9_lifted.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/cMyc_mm9_lifted.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/nfyb_esc_ucsc.bed"),
                       cex = 0.5
                       )

myc1 <- plot_gviz_track(GEN = "mm9", COOR = "chr15:61815896-61822916",
                       BAMFILE = browser_files[grep("cebp.*Rep2|control|myc_esc|e2f1_esc_Rep1", browser_files$id),]$bam_files,
                       type = "a", maxlimit = "30,30,30,30,30,30,30,30,100,100,100,100,100,100",
                       BEDFILE=c("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/E2f1_mm9_lifted.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/cMyc_mm9_lifted.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/nfyb_esc_ucsc.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.NFYB",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.LIN54",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/motifAna_selGenes/motifAna/myc/REGIONS_INTEREST.annoBed"),
                       cex = 0.5
                       )

ccnb1 <- plot_gviz_track(GEN = "mm9", COOR = "chr13:101547075-101557803",
                       BAMFILE = browser_files[grep("cebp.*Rep1|^nfyb|^e2f1_esc_Rep1", browser_files$id),]$bam_files,
                       type = "a", maxlimit = "20,20,20,20,20,20,20,20,200,30,100,100,100,100",
                       BEDFILE=c("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/E2f1_mm9_lifted.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/cMyc_mm9_lifted.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/nfyb_esc_ucsc.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.NFYB",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.LIN54",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.E2F1"
                                 )
                       )

cdk1 <- plot_gviz_track(GEN = "mm9", COOR = "chr10:68783105-68831938",
                       BAMFILE = browser_files[grep("cebp.*Rep1|^nfyb|^e2f1_esc_Rep1", browser_files$id),]$bam_files,
                       type = "a", maxlimit = "40,40,40,40,40,40,40,40,150,30,400,400,400,400",
                       BEDFILE=c("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/E2f1_mm9_lifted.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/cMyc_mm9_lifted.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/nfyb_esc_ucsc.bed",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.NFYB",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.LIN54",
                                 "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.E2F1"
                                 )
                       )

# CEBPE (+6 kb enhancer)
cebpe <- plot_gviz_track(GEN = "mm9", COOR = "chr14:55323682-55332011",
                        BAMFILE = browser_files[grep("cebp.*Rep1|myc_esc|max_esc|e2f1_esc_Rep1", browser_files$id),]$bam_files, 
                        type = "a", maxlimit = "500,500,500,500,500,500,500,500,100,20,20,6000,6000,6000,6000",
                        BEDFILE=c("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/E2f1_mm9_lifted.bed",
                                  "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/cMyc_mm9_lifted.bed",
                                  "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.E2F1"),
                        cex = 0.5)

# CEBPA (+37 kb enhancer)
cebpa <- plot_gviz_track(GEN = "mm9", COOR = "chr7:35891494-35950004",
                        BAMFILE = browser_files[grep("cebp.*Rep1|myc_esc|max_esc|e2f1_esc_Rep1", browser_files$id),]$bam_files, 
                        type = "a", maxlimit = "150,150,150,150,150,150,150,150,20,10,20,1000,1000,1000,1000",
                        BEDFILE=c("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/E2f1_mm9_lifted.bed",
                                  "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/cMyc_mm9_lifted.bed",
                                  "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.E2F1",
                                  "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/multiClassMotifAna/e2f_peaks/motifAna/REGIONS_INTEREST.annoBed.CEBPE",
                                  "/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/E2f1_mm9_source2.bed.ucsc"),
                       cex = 0.5)
dev.off()
```

## cell cycle genes analysis
```{r, eval=FALSE}

test <- genes_cebpe
genes_g1_s <- unique(as.vector(read.table(pipe("cut -f 2 /home/pundhir/project/genome_annotations/cell_cycle/genes_G1_S_phase_transition.txt"), sep="\t", header=T)))
dim(genes_g1_s)
genes_g2_m <- unique(as.vector(read.table(pipe("cut -f 2 /home/pundhir/project/genome_annotations/cell_cycle/genes_G2_M_phase_transition.txt"), sep="\t", header=T)))
dim(genes_g2_m)
test$cell_cycle <- "Unknown"
test[test$gene %in% genes_g2_m$Symbol,]$cell_cycle <- "G2_M_phase_transition"
test[test$gene %in% genes_g1_s$Symbol,]$cell_cycle <- "G1_S_phase_transition"
plot(Venn(list(g1_s=genes_g1_s$Symbol, g2_m=genes_g2_m$Symbol)))
attr(gplots::venn(list(g1_s=genes_g1_s$Symbol, g2_m=genes_g2_m$Symbol), show.plot=FALSE), "intersections")

freqPlot(melt(table(test[which(test$cluster_de!="F_rest" & test$cell_cycle!="Unknown"),c("cluster_de", "cell_cycle")])), "density") + theme_bw()

#-----------------------------------------------------------------#

df.melt <- melt(expr_data_norm_with_replicates[grep("^Cdkn1a$|^Cdkn3$|^Ccnd2$|^Cdk4$|^Cdkn2d$|^Cdkn2c$|^Ccna2$|^Ccnf$|^Ccnb1$|^Cdk1$|^Gadd45a$|^Trp53$", expr_data_norm_with_replicates$GeneID),
                                               c("GeneID",
                                                 grep("gmpG.*wt", colnames(expr_data_norm_with_replicates), value = T),
                                                 grep("epm_.*wt", colnames(expr_data_norm_with_replicates), value = T),
                                                 grep("lpm_.*wt", colnames(expr_data_norm_with_replicates), value = T),
                                                 grep("mm_.*wt", colnames(expr_data_norm_with_replicates), value = T),
                                                 grep("gmpG_.*ko", colnames(expr_data_norm_with_replicates), value = T),
                                                 grep("epm_.*ko", colnames(expr_data_norm_with_replicates), value = T),
                                                 grep("lpm_.*ko", colnames(expr_data_norm_with_replicates), value = T),
                                                 grep("mm_.*ko", colnames(expr_data_norm_with_replicates), value = T)
                                               )])

df.melt$class <- gsub("_.*", "", (df.melt$variable))
df.melt$variable <- gsub("^.*_", "", gsub("_[Rr]ep.*", "", df.melt$variable))
df.melt <- summarySE(df.melt, measurevar = "value", groupvars = c("class", "variable", "GeneID"))
df.melt$class <- factor(df.melt$class, levels=c("gmpG", "epm", "lpm", "mm"))
gXExprProfileCellCycle <- ggplot(df.melt, aes(class, value, color=variable)) +
  geom_errorbar(aes(ymin=value-se, ymax=value+se), width=.1, alpha=1) +
  geom_line(aes(group=factor(variable), color=factor(variable)), alpha=1) +
  geom_point(shape = 21, size = 2, fill="white") +
  scale_color_manual(values=c("#313695", "#a50026")) +
  theme_bw() + theme(legend.position="top") + 
  facet_wrap(~GeneID, scales="free")

gXExprProfileCellCycle

#------------------------------------------------------#
cell_cycle <- read.table("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/CELL_CYCLE_GENES")
colnames(cell_cycle) <- c("gene", "stage")
cell_cycle <- cell_cycle[cell_cycle$gene %in% genes_cebpe[which(genes_cebpe$de_count>0),]$gene,]
matrix2pheatmap(mat_raw[mat_raw$gene %in% cell_cycle$gene,c(2,3,5:(ncol(mat_raw)-1))], mat_raw[mat_raw$gene %in% cell_cycle$gene,]$cluster, "row", colorRampPalette(c("dodgerblue4", "#2171b5", "grey97", "#fc8d59", "sienna3"), bias=1)(256))[[4]]
table(genes_cebpe[genes_cebpe$gene %in% cell_cycle$gene,]$cluster_de)

mat <- as.matrix(rpClass[rpClass$name %in% cell_cycle[which(cell_cycle$stage=="G1-S"),]$gene,c(9:21)])
row.names(mat) <- rpClass[rpClass$name %in% cell_cycle[which(cell_cycle$stage=="G1-S"),]$gene,]$name
pheatmap(mat, Rowv = NA, Colv = NA)

mat <- as.matrix(rpClass[rpClass$name %in% cell_cycle[which(cell_cycle$stage=="G2-M"),]$gene,c(9:21)])
row.names(mat) <- rpClass[rpClass$name %in% cell_cycle[which(cell_cycle$stage=="G2-M"),]$gene,]$name
pheatmap(mat, Rowv = NA, Colv = NA)

#------------------------------------------------------#
library("pathview")
GL_epm <- c("Anapc5", "Atm", "Atr", "Ccnd2", "Cdc6", "Cdk4", "E2f5", "Mcm2", "Mcm3", "Mcm4", "Mcm5", 
            "Mcm6", "Mcm7", "Myc", "Pcna", "Prkdc", "Skp2", "Tgfb1", "Trp53", "Wee1")
GL_lpm <- c("Bub1", "Bub1b", "Ccna2", "Ccnb1", "Ccnb2", "Cdc20", "Cdc23", "Cdc25c", "Cdc45", "Cdk1", 
            "Chek2", "Dbf4", "Espl1", "Mad2l1", "Pkmyt1", "Rbl1", "Ttk")
pathview(gene.data  = GL_epm,
                     pathway.id = "mmu04110",
                     species    = "mmu", same.layer = F, gene.idtype = gene.idtype.list[1], out.suffix="EPM", kegg.native = F)
pathview(gene.data  = GL_lpm,
                     pathway.id = "mmu04110",
                     species    = "mmu", same.layer = F, gene.idtype = gene.idtype.list[1], out.suffix="LPM", kegg.native = F)

p1 <- ggplot(genes_cebpe[genes_cebpe$gene %in% GL_epm,], aes(myc_bound, log2FC_epm)) + geom_boxplot()
p2 <- ggplot(genes_cebpe[genes_cebpe$gene %in% GL_epm,], aes(e2f_bound, log2FC_mm)) + geom_boxplot()
p3 <- ggplot(genes_cebpe[genes_cebpe$gene %in% GL_epm,], aes(cebpe_bound, log2FC_mm)) + geom_boxplot()
p4 <- ggplot(genes_cebpe[which(genes_cebpe$gene %in% GL_lpm),], aes(peak_class, log2FC_mm)) + geom_boxplot()
genes_cebpe[which(genes_cebpe$gene %in% GL_epm & genes_cebpe$myc_bound=="Y"),]$gene
genes_cebpe[which(genes_cebpe$gene %in% GL_lpm & genes_cebpe$peak_class=="e2f1+cebpe+nfyb+lin54"),]$gene
gXCellCycleGenes <- ggarrange(p1, p2, p3, p4, nrow=2, ncol=2)

#------------------------------------------------------#

## most cell cycle genes in EPM and LPM cluster are up-regulated
table(genes_cebpe[,c("cluster_de", "cell_cycle")])
test <- genes_cebpe[which(genes_cebpe$de_mm=="up" & (genes_cebpe$cluster_de=="A_epm" | genes_cebpe$cluster_de=="B_lpm")),]
test$cell_cycle <- sprintf("%s_%s", test$cluster_de, test$cell_cycle)

ggplot(test, aes(cebpe_lpmmy, color=cell_cycle)) + stat_ecdf()
ggplot(test, aes(myc, color=cell_cycle)) + stat_ecdf()
ggplot(test, aes(e2f, color=cell_cycle)) + stat_ecdf()
freqPlot(melt(table(test[,c("cell_cycle", "peak_class")])), "density")

ggplot(genes_cebpe[which(genes_cebpe$de_mm=="up" & (genes_cebpe$cluster_de=="A_epm" | genes_cebpe$cluster_de=="B_lpm") & genes_cebpe$cell_cycle=="G1_S_phase_transition"),], aes(cluster_de, cebpe_lpmmy)) + geom_boxplot()
freqPlot(melt(table(genes_cebpe[which(genes_cebpe$cell_cycle=="G2_M_phase_transition"),c("de_mm", "peak_class")])), "density")

```

## plot results of cell cycle genes analysis
```{r}
pdf("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/pdf/analysis_cell_cycle.pdf")
gXExprProfileCellCycle
gXCellCycleGenes
dev.off()
```

## revision analysis
```{r}
xie_de_genes <- read.table(pipe("grep -vw GM ~/project/chip-seq-analysis/analysis_cebpe/revision/xie_de_genes.txt"))
xie_de_genes <- merge(xie_de_genes, genes_cebpe, by.x="V2", by.y="gene")
# xie_de_genes[which(xie_de_genes$cluster==3 | xie_de_genes$cluster==4 | xie_de_genes$cluster==5),]$cluster_de <- "A"
# xie_de_genes[which(xie_de_genes$cluster==6),]$cluster_de <- "B"
# xie_de_genes[which(xie_de_genes$cluster==10 | xie_de_genes$cluster==11),]$cluster_de <- "C"
# xie_de_genes[which(xie_de_genes$cluster==12),]$cluster_de <- "D"
# xie_de_genes[which(xie_de_genes$cluster==14 | xie_de_genes$cluster==15),]$cluster_de <- "E"

freqPlot(melt(table(xie_de_genes[,c("V1", "cluster")])), "density") + theme_bw()
# freqPlot(reshape2::melt(table(xie_de_genes[which(xie_de_genes$cluster_de!="F_rest"),c("V1", "cluster_de")])), "density") + theme_bw()
freqPlot(reshape2::melt(table(xie_de_genes[which(xie_de_genes$cluster_de!="F_rest" & !grepl("G5", xie_de_genes$V1)),
                                           c("cluster_de", "V1")])), "density") + theme_bw()

pheatmap(xie_de_genes[order(xie_de_genes$V1),grep("wt_norm", colnames(genes_cebpe), value=T)], scale="row", cluster_rows = F, cluster_cols = F)

##
df <- read.table("~/project/chip-seq-analysis/analysis_cebpe/analysis/genes_cebpe.bed.cebpaRevision")
colnames(df) <- c("chr", "start", "end", "name", "score", "strand", "cluster_deRefined", "peak_class", 
                  "input_my12_wt_Rep2", "input_my12_wt_Rep1", "input_my12_ko_Rep1", "input_mm_wt_Rep1", "input_mm_ko_Rep1",
                  "h3k27ac_my12_wt_Rep2", "h3k27ac_my12_wt_Rep1", "h3k27ac_my12_ko_Rep2", "h3k27ac_my12_ko_Rep1", 
                  "h3k27ac_mm_wt_Rep2", "h3k27ac_mm_wt_Rep1", "h3k27ac_mm_ko_Rep2", "h3k27ac_mm_ko_Rep1", 
                  "cebpa_my12_wt_Rep2", "cebpa_my12_wt_Rep1", "cebpa_my12_ko_Rep2", "cebpa_my12_ko_Rep1", 
                  "cebpa_mm_wt_Rep2", "cebpa_mm_wt_Rep1", "cebpa_mm_ko_Rep2", "cebpa_mm_ko_Rep1")


## cebpe (my12)
mat <- as.matrix(df[,grep("cebpa_my12", colnames(df))])
rownames(mat) <- df$name
colnames(mat)
mat <- matrix2deseq2(mat, "wt,wt,ko,ko")
mat$de <- "neutral"
mat[which(mat$pvalue<0.05 & mat$log2FoldChange<0),]$de <- "down"
mat[which(mat$pvalue<0.05 & mat$log2FoldChange>0),]$de <- "up"
table(mat$de)
colnames(mat) <- sprintf("%s_cebpa_my12", colnames(mat))
df <- merge(df, mat[,grep("gene|baseMean|log2FoldChange|de", colnames(mat), value=T)], by.x="name", by.y="gene_cebpa_my12")

## cebpe (mm)
mat <- as.matrix(df[,grep("cebpa_mm", colnames(df))])
rownames(mat) <- df$name
colnames(mat)
mat <- matrix2deseq2(mat, "wt,wt,ko,ko")
mat$de <- "neutral"
mat[which(mat$pvalue<0.05 & mat$log2FoldChange<0),]$de <- "down"
mat[which(mat$pvalue<0.05 & mat$log2FoldChange>0),]$de <- "up"
table(mat$de)
colnames(mat) <- sprintf("%s_cebpa_mm", colnames(mat))
df <- merge(df, mat[,grep("gene|baseMean|log2FoldChange|de", colnames(mat), value=T)], by.x="name", by.y="gene_cebpa_mm")

## h3k27ac (my12)
mat <- as.matrix(df[,grep("h3k27ac_my12", colnames(df))])
rownames(mat) <- df$name
colnames(mat)
mat <- matrix2deseq2(mat, "wt,wt,ko,ko")
mat$de <- "neutral"
mat[which(mat$pvalue<0.05 & mat$log2FoldChange<0),]$de <- "down"
mat[which(mat$pvalue<0.05 & mat$log2FoldChange>0),]$de <- "up"
table(mat$de)
colnames(mat) <- sprintf("%s_h3k27ac_my12", colnames(mat))
df <- merge(df, mat[,grep("gene|baseMean|log2FoldChange|de", colnames(mat), value=T)], by.x="name", by.y="gene_h3k27ac_my12")

## h3k27ac (mm)
mat <- as.matrix(df[,grep("h3k27ac_mm", colnames(df))])
rownames(mat) <- df$name
colnames(mat)
mat <- matrix2deseq2(mat, "wt,wt,ko,ko")
mat$de <- "neutral"
mat[which(mat$pvalue<0.05 & mat$log2FoldChange<0),]$de <- "down"
mat[which(mat$pvalue<0.05 & mat$log2FoldChange>0),]$de <- "up"
table(mat$de)
colnames(mat) <- sprintf("%s_h3k27ac_mm", colnames(mat))
df <- merge(df, mat[,grep("gene|baseMean|log2FoldChange|de", colnames(mat), value=T)], by.x="name", by.y="gene_h3k27ac_mm")

df.melt <- melt(df[, c("cluster_deRefined", grep("log2FoldChange", colnames(df), value=T))])
p1 <- ggplot(df.melt, aes(cluster_deRefined, value)) + 
  geom_boxplot() + facet_grid(.~variable) + theme_bw() + theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  geom_hline(yintercept = 0, linetype="dashed") + ylab("log2FoldChange in ChIP-seq signal")

df.melt <- df.melt[grepl("de", df.melt$cluster_deRefined),]
p2 <- ggplot(df.melt, aes(cluster_deRefined, value)) + 
  geom_boxplot(notch = 0.5) + facet_grid(.~variable) + theme_bw() + theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  geom_hline(yintercept = 0, linetype="dashed") + ylab("log2FoldChange in ChIP-seq signal")

pdf("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/pdf/analysis_cebpa_revision.pdf")
ggarrange(p1, p2, nrow=2, ncol=1)
dev.off()

# df.melt <- melt(df[which(df$de_cebpa_my12!="neutral" & df$de_cebpa_mm!="neutral"),
                   # c("peak_class", grep("log2FoldChange", colnames(df), value=T))])
# ggplot(df.melt, aes(peak_class, value)) + geom_boxplot(notch=0.5) + theme_bw() + facet_grid(.~variable)

# barplot(table(df[which(grepl("-de", df$cluster_deRefined) & df$peak_class=="e2f1+cebpe+nfyb+lin54" & df$de_cebpa_my12=="neutral" & df$de_cebpa_mm=="neutral" & 
                 # df$de_h3k27ac_my12!="down" & df$de_h3k27ac_mm!="down"),]$cluster_deRefined)/table(df[which(grepl("-de", df$cluster_deRefined) & df$peak_class=="e2f1+cebpe+nfyb+lin54"),]$cluster_deRefined), las=2)

# df[which(grepl("B_lpm-de", df$cluster_deRefined) & df$peak_class=="e2f1+cebpe+nfyb+lin54" & df$de_cebpa_my12=="neutral" & df$de_cebpa_mm=="neutral" & 
                 # df$de_h3k27ac_my12!="down" & df$de_h3k27ac_mm!="down"),]$name

# barplot(table(df[which(grepl("-de", df$cluster_deRefined) & df$de_cebpa_my12=="neutral" & df$de_cebpa_mm=="neutral" & 
                 # df$de_h3k27ac_my12=="down" & df$de_h3k27ac_mm=="down"),]$cluster_deRefined)/table(df$cluster_deRefined), las=2)

# df[which(grepl("-de", df$cluster_deRefined) & df$de_cebpa_my12=="neutral" & df$de_cebpa_mm=="neutral" & 
                 # df$de_h3k27ac_my12=="down" & df$de_h3k27ac_mm=="down"),]$name

test <- merge(df, genes_cebpe[,c("gene", "log2FC_lpm", "log2FC_mm")], by.x="name", by.y="gene")
cor(test$log2FC_mm, test$log2FoldChange_h3k27ac_mm, use="complete.obs", method="spearman")
```



















#### analyze transcription factor binding profile at proximal and distal regions to interesting gene clusters
```{r}




## add motif (cebpe, e2f, myc etc) information to genes_cebpe df

ggplot(melt(genes_cebpe[grep("^Cebpa|Cebpe|E2f1|Myc$|Nfyb$|Lin54$", genes_cebpe$gene),c(4,28:29,31:37,43:46)]), aes(variable, (value), fill=variable)) + geom_bar(stat="identity") + theme(axis.text.x = element_text(angle = 45, hjust = 1)) + facet_wrap(~gene, scales="free")

t.melt <- melt(genes_cebpe[,c("pu1_lsk", "pu1_pregm", "pu1_gmp","cluster_de")])
# t.melt$value <- log(t.melt$value)
t.melt <- summarySE(t.melt, measurevar = "value", groupvars = c("cluster_de", "variable"))
p1 <- ggplot(t.melt, aes(variable, value, color=cluster_de)) +
  geom_errorbar(aes(ymin=value-se, ymax=value+se), width=.1, alpha=1) +
  geom_line(aes(group=factor(cluster_de), color=factor(cluster_de)), alpha=1) +
  geom_point(shape = 21, size = 2, fill="white") +
  #scale_color_manual(values=c("#313695", "#a50026")) +
  theme(legend.position="top") + theme_bw() + ylab("PU.1 - promoters (TPM)")

t.melt <- melt(genes_cebpe[,c("cebpe_lpmmy", "cebpe_mm", "cebpe_bc", "cebpe_pmn", "cluster_de"),])
# t.melt$value <- log(t.melt$value)
t.melt <- summarySE(t.melt, measurevar = "value", groupvars = c("cluster_de", "variable"))
p2 <- ggplot(t.melt, aes(variable, value, color=cluster_de)) +
  geom_errorbar(aes(ymin=value-se, ymax=value+se), width=.1, alpha=1) +
  geom_line(aes(group=factor(cluster_de), color=factor(cluster_de)), alpha=1) +
  geom_point(shape = 21, size = 2, fill="white") +
  #scale_color_manual(values=c("#313695", "#a50026")) +
  theme(legend.position="top") + theme_bw() + ylab("CEBPA/E - promoters (TPM)")

t.melt <- summarySE(melt(genes_cebpe[,c("pu1_lsk_dis", "pu1_pregm_dis", "pu1_gmp_dis", "cluster_de")]), measurevar = "value", groupvars = c("cluster_de", "variable"))
p3 <- ggplot(t.melt, aes(variable, value, color=cluster_de)) +
  geom_errorbar(aes(ymin=value-se, ymax=value+se), width=.1, alpha=1) +
  geom_line(aes(group=factor(cluster_de), color=factor(cluster_de)), alpha=1) +
  geom_point(shape = 21, size = 2, fill="white") +
  #scale_color_manual(values=c("#313695", "#a50026")) +
  theme(legend.position="top") + theme_bw() + ylab("PU.1 - distal (TPM)")

t.melt <- summarySE(melt(genes_cebpe[,c("cebpe_lpmmy_dis", "cebpe_mm_dis", "cebpe_bc_dis", "cebpe_pmn_dis", "cluster_de")]), measurevar = "value", groupvars = c("cluster_de", "variable"))
p4 <- ggplot(t.melt, aes(variable, value, color=cluster_de)) +
  geom_errorbar(aes(ymin=value-se, ymax=value+se), width=.1, alpha=1) +
  geom_line(aes(group=factor(cluster_de), color=factor(cluster_de)), alpha=1) +
  geom_point(shape = 21, size = 2, fill="white") +
  #scale_color_manual(values=c("#313695", "#a50026")) +
  theme(legend.position="top") + theme_bw() + ylab("CEBPA/E - distal (TPM)")

t.melt <- genes_cebpe[,c("cluster_de", "pu1_pregm", "cebpa_gmp", "cebpe_lpmmy")]
t.melt[,c(2:ncol(t.melt))] <- scale_df(as.matrix(t.melt[,c(2:ncol(t.melt))]), "qu")
t.melt <- summarySE(melt(t.melt), measurevar = "value", groupvars = c("cluster_de", "variable"))

p5 <- ggplot(t.melt, aes(x = cluster_de, y = variable, label = value)) +
  geom_point(aes(size = value, colour = value, alpha=.02)) + 
  #geom_text(hjust = 1, size = 2) +
  scale_size(range = c(1,10)) +
  theme_bw() + scale_y_discrete(position="right") + ylab("") + 
  scale_color_gradientn(colours = rev(colorRampPalette(brewer.pal(9, "RdBu"), bias=1)(256))) + xlab("") +
  theme(legend.position="none", legend.text = element_text(size=6), axis.text.y = element_text(angle = -90, hjust = 1))

t.melt <- genes_cebpe[,c("cluster_de", "pu1_pregm_dis", "cebpa_gmp_dis", "cebpe_lpmmy_dis")]
t.melt[,c(2:ncol(t.melt))] <- scale_df(as.matrix(t.melt[,c(2:ncol(t.melt))]), "qu")
t.melt <- summarySE(melt(t.melt), measurevar = "value", groupvars = c("cluster_de", "variable"))

p6 <- ggplot(t.melt, aes(x = cluster_de, y = variable, label = value)) +
  geom_point(aes(size = value, colour = value, alpha=.02)) + 
  #geom_text(hjust = 1, size = 2) +
  scale_size(range = c(1,10)) +
  theme_bw() + scale_y_discrete(position="right") + ylab("") + 
  scale_color_gradientn(colours = rev(colorRampPalette(brewer.pal(9, "RdBu"), bias=1)(256))) + xlab("") +
  theme(legend.position="none", legend.text = element_text(size=6), axis.text.y = element_text(angle = -90, hjust = 1))

gDEWithTF <- ggarrange(p1, p2, p3, p4, p5, p6, ncol = 2, nrow=3)

t.melt <- summarySE(melt(genes_cebpe[,c("enhancer_count", "cluster_de")]), measurevar = "value", groupvars = c("cluster_de", "variable"))
p1 <- ggplot(t.melt, aes(x = cluster_de, y = variable, label = value)) +
  geom_point(aes(size = value, colour = value, alpha=.02)) + 
  #geom_text(hjust = 1, size = 2) +
  scale_size(range = c(1,10)) +
  scale_y_discrete(position="right") + ylab("") + 
  scale_color_gradientn(colours = rev(colorRampPalette(brewer.pal(9, "RdBu"), bias=1)(256))) + xlab("") +
  theme_bw() +
  theme(legend.position="none", legend.text = element_text(size=6), axis.text.x = element_text(angle = 45, hjust = 1))

p2 <- ggplot(melt(genes_cebpe[,c("cluster", "enhancer_count")]), aes(log(value), color=cluster)) + stat_ecdf() +
  scale_color_manual(values=colorRampPalette(brewer.pal(11, "PRGn"), bias=1)(15)) + xlab("# enhancers") + ylab("Density")

t.melt <- summarySE(melt(genes_cebpe[which(genes_cebpe$cluster_de!="F_rest"),c("pu1_pregm", "cebpa_gmp", "cebpe_lpmmy", "cluster_de")]), measurevar = "value", groupvars = c("cluster_de", "variable"))
p3 <- ggplot(t.melt, aes(cluster_de, value, color=variable)) +
  geom_errorbar(aes(ymin=value-se, ymax=value+se), width=.1, alpha=1) +
  geom_line(aes(group=factor(variable), color=factor(variable)), alpha=1) +
  geom_point(shape = 21, size = 2, fill="white") +
  #scale_color_manual(values=c("#313695", "#a50026")) +
  theme(legend.position="top", axis.text.x = element_text(angle = 45, hjust = 1))  + ylab("proximal (TPM)") +
  facet_wrap(~variable, scales="free") + theme_bw()

# test <- genes_cebpe
# test$pu1_pregm_dis <- test$pu1_pregm_dis/test$enhancer_count
# test$cebpa_gmp_dis <- test$cebpa_gmp_dis/test$enhancer_count
# test$cebpe_lpmmy_dis <- test$cebpe_lpmmy_dis/test$enhancer_count

t.melt <- summarySE(melt(genes_cebpe[which(genes_cebpe$cluster_de!="F_rest"),c("pu1_pregm_dis", "cebpa_gmp_dis", "cebpe_lpmmy_dis", "cluster_de")]), measurevar = "value", groupvars = c("cluster_de", "variable"))
p4 <- ggplot(t.melt, aes(cluster_de, value, color=variable)) +
  geom_errorbar(aes(ymin=value-se, ymax=value+se), width=.1, alpha=1) +
  geom_line(aes(group=factor(variable), color=factor(variable)), alpha=1) +
  geom_point(shape = 21, size = 2, fill="white") +
  #scale_color_manual(values=c("#313695", "#a50026")) +
  theme(legend.position="top", axis.text.x = element_text(angle = 45, hjust = 1)) + ylab("distal (TPM)") +
  facet_wrap(~variable, scales="free")

gDEWithTFRefined <- ggarrange(p1, p2, p3, p4, ncol = 2, nrow=2)

# geneSet[[1]]$fc <- geneSet[[1]]$lpm_ko/geneSet[[1]]$lpm_wt
# t.melt <- summarySE(melt(geneSet[[1]][,c("lpm_wt", "lpm_ko", "fc")]), measurevar = "value", groupvars = c("fc", "variable"))
# ggplot(t.melt, aes(fc, value, color=variable)) +
#   geom_errorbar(aes(ymin=value-se, ymax=value+se), width=.1, alpha=1) +
#   geom_line(aes(group=factor(variable), color=factor(variable)), alpha=1) +
#   geom_point(shape = 21, size = 2, fill="white") +
#   scale_color_manual(values=c("#a50026", "#313695")) +
#   theme(legend.position="top") + theme_bw()

ggplot(melt(genes_cebpe[,c("cluster_de","cebpe_lpmmy","cebpe_lpmmy_dis")]), aes(cluster_de, log(value))) + geom_boxplot() + facet_grid(.~variable)
#df <- read.table("/home/pundhir/project/chip-seq-analysis/data/gene/cebpe/t")
#df$enhancers_count <- unlist(lapply(df$V9, function(x) length(unlist(strsplit(as.character(x),",")))))
#ggplot(melt(df[,c(7,10)]), aes(V7, log(value))) + geom_boxplot()

rpClass <- read.table("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/analysis/allGenes.RPClass", header=T)
rpClass <- merge(rpClass, genes_cebpe[,c(4,27,99)], by.x="name", by.y="gene")
rpClass <- rpClass[,c(2:4,1,5:ncol(rpClass))]
rpClass$cluster_de <- factor(rpClass$cluster_de, levels=sort(unique(rpClass$cluster_de)), ordered = T)

## plot percentage of genes from each class that are bound by specific TFs
rpClass[,c(9:21,23:35)] <- apply(rpClass[,c(9:21,23:35)], 2, function(x) ifelse(is.na(x), 0, 1))

#y="e2f"; barplot(unlist(lapply(levels(rpClass$cluster_de), function(x) (sum(ifelse(rpClass[which(rpClass$cluster_de==x),c(y)]=="Y", 1, 0))*100)/length(rpClass[which(rpClass$cluster_de==x),c(y)]))), names.arg =  levels(rpClass$cluster_de), las=2)

rpClassSummary <- as.data.frame(lapply(colnames(rpClass[,c(9:21)]), function(y) unlist(lapply(levels(rpClass$cluster_de), function(x) (sum(ifelse(rpClass[which(rpClass$cluster_de==x),c(y)]==1, 1, 0))*100)/length(rpClass[which(rpClass$cluster_de==x),c(y)])))))
colnames(rpClassSummary) <- colnames(rpClass[,c(9:21)])
rownames(rpClassSummary) <- levels(rpClass$cluster_de)
rpClassSummary$cluster_de <- levels(rpClass$cluster_de)

#ggplot(melt(rpClassSummary), aes(cluster_de, value)) + geom_bar(stat="identity") + facet_grid(.~variable)
pheatmap(as.matrix(rpClassSummary[,c(1:ncol(rpClassSummary)-1)]), scale = "column")
```












#### analyze genes cluster for enrichment of e2f, pu1, cebpa, cebpe and cell cycle genes
```{r, eval=FALSE}
p1 <- ggplot(melt(genes_cebpe[,c(28:30,27)]), aes(variable, log(value))) + geom_boxplot(notch=TRUE, notchwidth = 0.3) + facet_grid(.~cluster) + ylim(c(-5,5)) + geom_hline(yintercept = 0) + theme_bw() + ylab("PU.1 - promoters (log; TPM)")
p2 <- ggplot(melt(genes_cebpe[,c(31:33,27)]), aes(variable, log(value))) + geom_boxplot(notch=TRUE, notchwidth = 0.3) + facet_grid(.~cluster) + ylim(c(-3,3)) + geom_hline(yintercept = 0) + theme_bw() + ylab("CEBPA - promoters (log; TPM)")
p3 <- ggplot(melt(genes_cebpe[,c(34:37,27)]), aes(variable, log(value))) + geom_boxplot(notch=TRUE, notchwidth = 0.3) + facet_grid(.~cluster) + ylim(c(-3,3)) + geom_hline(yintercept = 0) + theme_bw() + ylab("CEBPE - promoters (log; TPM)")
p4 <- ggplot(test, aes(cluster, per_e2f_binding)) + geom_bar(stat="identity") + theme_bw()
p5 <- ggplot(test, aes(cluster, per_e2f_binding_distal)) + geom_bar(stat="identity") + theme_bw()
p6 <- ggplot(test, aes(cluster, impGenes)) + geom_bar(stat="identity") + theme_bw()
p7 <- ggplot(melt(genes_cebpe[,c(39:41,27)]), aes(variable, log(value))) + geom_boxplot(notch=TRUE, notchwidth = 0.3) + facet_grid(.~cluster) + ylim(c(-5,5)) + geom_hline(yintercept = 0) + theme_bw() + ylab("PU.1 - distal (log; TPM)")
p8 <- ggplot(melt(genes_cebpe[,c(42:44,27)]), aes(variable, log(value))) + geom_boxplot(notch=TRUE, notchwidth = 0.3) + facet_grid(.~cluster) + ylim(c(-3,3)) + geom_hline(yintercept = 0) + theme_bw() + ylab("CEBPA - distal (log; TPM)")
p9 <- ggplot(melt(genes_cebpe[,c(45:48,27)]), aes(variable, log(value))) + geom_boxplot(notch=TRUE, notchwidth = 0.3) + facet_grid(.~cluster) + ylim(c(-3,3)) + geom_hline(yintercept = 0) + theme_bw() + ylab("CEBPE - distal (log; TPM)")

# ggplot(melt(genes_cebpe[,c(8:28,29)]), aes(variable, log(value))) + geom_boxplot() + facet_grid(.~cluster) + geom_hline(yintercept = 0) + theme_bw()
gTF <- ggarrange(p1, p2, p3, p4, p5, p6, p7, p8, p9, ncol = 3, nrow=3, labels = c("A", "B", "C", "D", "E", "F", "G", "H", "I"))

heatmap_lst <- list()
heatmap_lst[[1]] <- pheatmap(cor(rp[apply(rp[,c(8:27)], 1, function(x) ifelse(sd(x, na.rm=TRUE)>0, T, F)),c(8:27)]))[[4]]
heatmap_lst[[2]] <- pheatmap(cor(rp[apply(rp[,c(29:48)], 1, function(x) ifelse(sd(x, na.rm=TRUE)>0, T, F)),c(29:48)]))[[4]]
heatmap_lst[[3]] <- pheatmap(cor(rp[apply(rp[,c(50:69)], 1, function(x) ifelse(sd(x, na.rm=TRUE)>0, T, F)),c(50:69)]))[[4]]

heatmap_lst[[4]] <- pheatmap(cor(rp_mean[apply(rp_mean[,c(8:17)], 1, function(x) ifelse(sd(x, na.rm=TRUE)>0, T, F)),c(8:17)]))[[4]]
heatmap_lst[[5]] <- pheatmap(cor(rp_mean[apply(rp_mean[,c(19:28)], 1, function(x) ifelse(sd(x, na.rm=TRUE)>0, T, F)),c(19:28)]))[[4]]
heatmap_lst[[6]] <- pheatmap(cor(rp_mean[apply(rp_mean[,c(30:39)], 1, function(x) ifelse(sd(x, na.rm=TRUE)>0, T, F)),c(30:39)]))[[4]]

gTFCorr <- grid.arrange(arrangeGrob(grobs= heatmap_lst,ncol=3))
```

#### analyze genes cluster for enrichment of e2f, pu1, cebpa, cebpe and cell cycle genes
```{r, eval=FALSE}
customCol <- c(rep("#4575b4", 2), rep("#91bfdb", 2), rep("#e0f3f8", 2), rep("#ffffbf", 3), rep("#fee090", 2), rep("#fc8d59", 2), rep("#d73027", 2))

## BOXPLOTS
df <- melt(genes_cebpe[,c(29,27)])
df$median <- as.factor((ave(melt(genes_cebpe[,c(29,27)])$value, as.factor(melt(genes_cebpe[,c(29,27)])$cluster), FUN = median)))
p1 <- ggplot(df, aes(cluster, log(value), fill=median)) + geom_boxplot(notch=TRUE, notchwidth = 0.3) + ylim(c(-3,5)) + 
  geom_hline(yintercept = median(log(df$value))) + theme_bw() + stat_summary(fun.y=median,geom="line",lwd=0.5,aes(group=1)) +
  scale_fill_manual(values=customCol) + theme(legend.position = "none") + ylab("PU.1 - promoters")
df <- melt(genes_cebpe[,c(33,27)])
df$median <- as.factor((ave(melt(genes_cebpe[,c(33,27)])$value, as.factor(melt(genes_cebpe[,c(33,27)])$cluster), FUN = median)))
p2 <- ggplot(df, aes(cluster, log(value), fill=median)) + geom_boxplot(notch=TRUE, notchwidth = 0.3) + ylim(c(-3,5)) + 
  geom_hline(yintercept = median(log(df$value))) + theme_bw() + stat_summary(fun.y=median,geom="line",lwd=0.5,aes(group=1)) +
  scale_fill_manual(values=customCol) + theme(legend.position = "none") + ylab("CEBPA - promoters")
df <- melt(genes_cebpe[,c(34,27)])
df$median <- as.factor((ave(melt(genes_cebpe[,c(34,27)])$value, as.factor(melt(genes_cebpe[,c(34,27)])$cluster), FUN = median)))
p3 <- ggplot(df, aes(cluster, log(value), fill=median)) + geom_boxplot(notch=TRUE, notchwidth = 0.3) + ylim(c(-3,5)) + 
  geom_hline(yintercept = median(log(df$value))) + theme_bw() + stat_summary(fun.y=median,geom="line",lwd=0.5,aes(group=1)) +
  scale_fill_manual(values=customCol) + theme(legend.position = "none") + ylab("CEBPE - promoters")
p4 <- ggplot(test, aes(cluster, per_e2f_binding, fill=as.factor(per_e2f_binding))) + geom_bar(stat="identity") + theme_bw() + 
  scale_fill_manual(values=customCol) + theme(legend.position = "none") + ylab("Promoters bound by E2F (%)")
p5 <- ggplot(test, aes(cluster, per_e2f_binding_distal, fill=as.factor(per_e2f_binding_distal))) + geom_bar(stat="identity") + theme_bw() +
  scale_fill_manual(values=customCol) + theme(legend.position = "none") + ylab("Enhancers bound by E2F (%)")
p6 <- ggplot(test, aes(cluster, impGenes, fill=as.factor(impGenes))) + geom_bar(stat="identity") + theme_bw() +
  scale_fill_manual(values=customCol) + theme(legend.position = "none") + ylab("# Imp. genes")
gTFMean <- ggarrange(p1, p2, p3, p4, p5, p6, ncol = 3, nrow=2, labels = c("A", "B", "C", "D", "E", "F"))

## HEATMAPS
df <- melt(genes_cebpe[,c(29,27)])
df$median <- as.factor((ave(melt(genes_cebpe[,c(29,27)])$value, as.factor(melt(genes_cebpe[,c(29,27)])$cluster), FUN = median)))
p1 <- ggplot(df, aes(1, cluster)) + geom_raster(aes(fill=as.factor(median))) + scale_fill_manual(values=customCol) + theme(legend.position = "none") + theme_bw()
df <- melt(genes_cebpe[,c(33,27)])
df$median <- as.factor((ave(melt(genes_cebpe[,c(33,27)])$value, as.factor(melt(genes_cebpe[,c(33,27)])$cluster), FUN = median)))
p2 <- ggplot(df, aes(1, cluster)) + geom_raster(aes(fill=as.factor(median))) + scale_fill_manual(values=customCol) + theme(legend.position = "none") + theme_bw()
df <- melt(genes_cebpe[,c(34,27)])
df$median <- as.factor((ave(melt(genes_cebpe[,c(34,27)])$value, as.factor(melt(genes_cebpe[,c(34,27)])$cluster), FUN = median)))
p3 <- ggplot(df, aes(1, cluster)) + geom_raster(aes(fill=as.factor(median))) + scale_fill_manual(values=customCol) + theme(legend.position = "none") + theme_bw()
p4 <- ggplot(test, aes(1, cluster)) + geom_raster(aes(fill=as.factor(per_e2f_binding))) + scale_fill_manual(values=customCol) + theme(legend.position = "none") + theme_bw()
p5 <- ggplot(test, aes(1, cluster)) + geom_raster(aes(fill=as.factor(per_e2f_binding_distal))) + scale_fill_manual(values=customCol) + theme(legend.position = "none")+ theme_bw()
p6 <- ggplot(test, aes(1, cluster)) + geom_raster(aes(fill=as.factor(impGenes))) + scale_fill_manual(values=customCol) + theme(legend.position = "none") + theme_bw()
gTFMeanHeatmap <- ggarrange(p1, p2, p3, p4, p5, p6, ncol = 3, nrow=2, labels = c("A", "B", "C", "D", "E", "F"))

# df.melt <- data.frame(cluster=as.factor(sprintf("C%s", (row.names(genes_cebpe)))), value=genes_cebpe$pu1_pregm)
# ggplot(df.melt, aes(cluster, 1, fill=log(value))) + geom_tile() + 
#   scale_fill_gradientn(colours = rev(colorRampPalette(brewer.pal(11, "RdBu"), bias=0.6)(256)))

```

#### plot PU.1, CEBPA, CEBPE and E2F signal plots for different gene clusters
```{r, eval=FALSE}
pdf("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/pdf/tfSignal_gene_expression.pdf", width=10)
gTF
gTFCorr
gTFMean
gTFMeanHeatmap
dev.off()
```

#### define cluster based on proximal pu1, cebpa, cebpe peaks
```{r, eval=FALSE}
mat_peaks <- normalize.quantiles(as.matrix(genes_cebpe[,c(30:39)]))
cluster_info_matrix(mat_peaks[apply(mat_peaks, 1, function(x) ifelse(sd(x, na.rm=TRUE)>0, T, F)),])
set.seed(1)
# KC_CEBPE_PEAKS <- cluster_matrix(mat_peaks[apply(mat_peaks, 1, function(x) ifelse(sd(x, na.rm=TRUE)>0, T, F)),],10)
# mat_peaks <- as.data.frame(mat_peaks)
colnames(mat_peaks) <- colnames(genes_cebpe[,c(30:39)])
mat_peaks$cluster <- 0
mat_peaks[apply(mat_peaks[,c(1:ncol(mat_peaks)-1)], 1, function(x) ifelse(sd(x, na.rm=TRUE)>0, T, F)),]$cluster <- KC_CEBPE_PEAKS$cluster
mat_peaks$cluster <- sprintf("CP%d", mat_peaks$cluster)
genes_cebpe$cluster_peaks <- mat_peaks$cluster

# p1 <- ggplot(melt(genes_cebpe[,c(30:32,63)]), aes(variable, log(value))) + geom_boxplot() + facet_grid(.~cluster_peaks) + ylim(c(-5,5)) + geom_hline(yintercept = 0)
# p2 <- ggplot(melt(genes_cebpe[,c(33:35,63)]), aes(variable, log(value))) + geom_boxplot() + facet_grid(.~cluster_peaks) + ylim(c(-3,3)) + geom_hline(yintercept = 0)
# p3 <- ggplot(melt(genes_cebpe[,c(36:39,63)]), aes(variable, log(value))) + geom_boxplot() + facet_grid(.~cluster_peaks) + ylim(c(-3,3)) + geom_hline(yintercept = 0)
# g <- ggarrange(p1, p2, p3, ncol = 3, nrow=1, labels = c("A", "B", "C"))
```

#### define cluster based on distal pu1, cebpa, cebpe peaks
```{r, eval=FALSE}
mat_peaks_distal <- normalize.quantiles(as.matrix(genes_cebpe[,c(41:50)]))
cluster_info_matrix(mat_peaks_distal[apply(mat_peaks_distal, 1, function(x) ifelse(sd(x, na.rm=TRUE)>0, T, F)),])
set.seed(1)
# KC_CEBPE_PEAKS_DISTAL <- cluster_matrix(mat_peaks_distal[apply(mat_peaks_distal, 1, function(x) ifelse(sd(x, na.rm=TRUE)>0, T, F)),],10)
# mat_peaks_distal <- as.data.frame(mat_peaks_distal)
colnames(mat_peaks_distal) <- colnames(genes_cebpe[,c(41:50)])
mat_peaks_distal$cluster <- 0
mat_peaks_distal[apply(mat_peaks_distal[,c(1:ncol(mat_peaks_distal)-1)], 1, function(x) ifelse(sd(x, na.rm=TRUE)>0, T, F)),]$cluster <- KC_CEBPE_PEAKS_DISTAL$cluster
mat_peaks_distal$cluster <- sprintf("CP%d", mat_peaks_distal$cluster)
genes_cebpe$cluster_peaks_distal <- mat_peaks_distal$cluster

# p1 <- ggplot(melt(genes_cebpe[,c(41:43,64)]), aes(variable, log(value))) + geom_boxplot() + facet_grid(.~cluster_peaks_distal) + ylim(c(-5,5)) + geom_hline(yintercept = 0)
# p2 <- ggplot(melt(genes_cebpe[,c(44:46,64)]), aes(variable, log(value))) + geom_boxplot() + facet_grid(.~cluster_peaks_distal) + ylim(c(-3,3)) + geom_hline(yintercept = 0)
# p3 <- ggplot(melt(genes_cebpe[,c(47:50,64)]), aes(variable, log(value))) + geom_boxplot() + facet_grid(.~cluster_peaks_distal) + ylim(c(-3,3)) + geom_hline(yintercept = 0)
# g <- ggarrange(p1, p2, p3, ncol = 3, nrow=1, labels = c("A", "B", "C"))
```

#### analyze relationship between gene expression dynamics and proximal and distal chromatin environment (peaks)
```{r, eval=FALSE}
matrix2pheatmap(mat_peaks[order(mat_peaks$cluster),c(1:ncol(mat_peaks)-1)],
                mat_peaks[order(mat_peaks$cluster),]$cluster, "row", 
                colorRampPalette(c("dodgerblue4", "#2171b5", "grey97", "#fc8d59", "sienna3"), bias=1)(256))

matrix2pheatmap(genes_cebpe[order(genes_cebpe$cluster_peaks),c(30:39)],
                genes_cebpe[order(genes_cebpe$cluster_peaks),]$cluster_peaks, "row", 
                colorRampPalette(c("dodgerblue4", "#2171b5", "grey97", "#fc8d59", "sienna3"), bias=1)(256))

genes_cebpe[which(genes_cebpe$cluster_peaks=="CP1" | 
                    genes_cebpe$cluster_peaks=="CP2" | 
                    genes_cebpe$cluster_peaks=="CP3" | 
                    genes_cebpe$cluster_peaks=="CP6"),]$cluster_peaks <- "cebpa"
genes_cebpe[which(genes_cebpe$cluster_peaks=="CP4"),]$cluster_peaks <- " pu1"
genes_cebpe[which(genes_cebpe$cluster_peaks=="CP5"),]$cluster_peaks <- "cebpa_cebpe"
genes_cebpe[which(genes_cebpe$cluster_peaks=="CP7"),]$cluster_peaks <- "cebpe"
genes_cebpe[which(genes_cebpe$cluster_peaks=="CP8"),]$cluster_peaks <- "pu1_cebpe"
genes_cebpe[which(genes_cebpe$cluster_peaks=="CP9" | genes_cebpe$cluster_peaks=="CP10"),]$cluster_peaks <- "pu1_cebpa"


freqPlot(melt((table(genes_cebpe[,c(29,63)]))), "density")
freqPlot(melt((table(genes_cebpe[,c(29,64)]))), "density")

matrix2pheatmap(mat_peaks_distal[order(mat_peaks_distal$cluster),c(1:ncol(mat_peaks_distal)-1)], 
                mat_peaks_distal[order(mat_peaks_distal$cluster),]$cluster, "row", 
                colorRampPalette(c("dodgerblue4", "#2171b5", "grey97", "#fc8d59", "sienna3"), bias=1)(256))

matrix2pheatmap(genes_cebpe[order(genes_cebpe$cluster_peaks),c(41:50)], 
                genes_cebpe[order(genes_cebpe$cluster_peaks),]$cluster_peaks_distal, "row", 
                colorRampPalette(c("dodgerblue4", "#2171b5", "grey97", "#fc8d59", "sienna3"), bias=1)(256))

```

## gene set classification based on taking up and down-DE genes separately
```{r, eval=FALSE}

-------------------------------------------------------------------------------------------------

rpClass$pu1_bound <- "N"
rpClass[grep("pu1", apply(rpClass[,c(9:21)], 1, function(x) names(which(!is.na(x))))),]$pu1_bound <- "Y"

rpClass$cebpa_bound <- "N"
rpClass[grep("cebpa", apply(rpClass[,c(9:21)], 1, function(x) names(which(!is.na(x))))),]$cebpa_bound <- "Y"

rpClass$cebpe_bound <- "N"
rpClass[grep("cebpe", apply(rpClass[,c(9:21)], 1, function(x) names(which(!is.na(x))))),]$cebpe_bound <- "Y"

rpClass$e2f_bound <- "N"
rpClass[grep("e2f", apply(rpClass[,c(9:21)], 1, function(x) names(which(!is.na(x))))),]$e2f_bound <- "Y"

rpClass$pu1_bound_dis <- "N"
rpClass[grep("pu1", apply(rpClass[,c(23:35)], 1, function(x) names(which(!is.na(x))))),]$pu1_bound_dis <- "Y"

rpClass$cebpa_bound_dis <- "N"
rpClass[grep("cebpa", apply(rpClass[,c(23:35)], 1, function(x) names(which(!is.na(x))))),]$cebpa_bound_dis <- "Y"

rpClass$cebpe_bound_dis <- "N"
rpClass[grep("cebpe", apply(rpClass[,c(23:35)], 1, function(x) names(which(!is.na(x))))),]$cebpe_bound_dis <- "Y"

rpClass$e2f_bound_dis <- "N"
rpClass[grep("e2f", apply(rpClass[,c(23:35)], 1, function(x) names(which(!is.na(x))))),]$e2f_bound_dis <- "Y"

-------------------------------------------------------------------------------------------------

## expression profile (see section below for original cluster classification based on down and up-DE genes separately)
geneSet <- list()

geneSet[["A_early"]] <- genes_cebpe[which((genes_cebpe$cluster==2) &
                                    (genes_cebpe$de_lpm=="up" | genes_cebpe$de_mm=="up" | genes_cebpe$de_gmpG=="down" | genes_cebpe$de_epm=="down")),]
geneSet[["A_early"]]$cluster_de <- "A_early"

geneSet[["B_gmp_down"]] <- genes_cebpe[which((genes_cebpe$cluster==3 | genes_cebpe$cluster==4 | genes_cebpe$cluster==5) & 
                                    (genes_cebpe$de_lpm=="down" | genes_cebpe$de_mm=="down")),]
geneSet[["B_gmp_down"]]$cluster_de <- "B_gmp_down"

geneSet[["B_gmp_up"]] <- genes_cebpe[which((genes_cebpe$cluster==3 | genes_cebpe$cluster==4 | genes_cebpe$cluster==5) & 
                                    (genes_cebpe$de_lpm=="up" | genes_cebpe$de_mm=="up")),]
geneSet[["B_gmp_up"]]$cluster_de <- "B_gmp_up"

geneSet[["C_lpm_down"]] <- genes_cebpe[which((genes_cebpe$cluster==6) & 
                                    (genes_cebpe$de_lpm=="down" | genes_cebpe$de_mm=="down")),]
geneSet[["C_lpm_down"]]$cluster_de <- "C_lpm_down"

geneSet[["C_lpm_up"]] <- genes_cebpe[which((genes_cebpe$cluster==6) & (genes_cebpe$de_mm=="up")),]
geneSet[["C_lpm_up"]]$cluster_de <- "C_lpm_up"

geneSet[["D_gran_down"]] <- genes_cebpe[which((genes_cebpe$cluster==8 | genes_cebpe$cluster==9 | genes_cebpe$cluster==10 | genes_cebpe$cluster==11) & 
                                    (genes_cebpe$de_gmpG=="up" | genes_cebpe$de_epm=="up" | genes_cebpe$de_lpm=="down" | genes_cebpe$de_mm=="down")),]
geneSet[["D_gran_down"]]$cluster_de <- "D_gran_down"

geneSet[["D_gran_up"]] <- genes_cebpe[which((genes_cebpe$cluster==12) &
                                    (genes_cebpe$de_lpm=="up" | genes_cebpe$de_mm=="up")),]
geneSet[["D_gran_up"]]$cluster_de <- "D_gran_up"

geneSet[["E_mono"]] <- genes_cebpe[which((genes_cebpe$cluster==14 | genes_cebpe$cluster==15) & (genes_cebpe$de_lpm=="up" | genes_cebpe$de_mm=="up")),]
geneSet[["E_mono"]]$cluster_de <- "E_mono"

geneSet[["F_rest"]] <- genes_cebpe[!genes_cebpe$gene %in% do.call(rbind, geneSet)$gene,]
geneSet[["F_rest"]]$cluster_de <- "F_rest"

# plot(Venn(lapply(geneSet[2:5], function(x) x$gene)))
input <- attr(gplots::venn(lapply(geneSet[1:length(geneSet)], function(x) x$gene), show.plot=FALSE), "intersections")
# unique(input$"B_gmp_down:B_gmp_up")
# unique(input$"C_lpm_down:C_lpm_up")

## define gene clusters based upon the stage at which they are differentially expressed
#unlist(lapply(grep(":", names(input), value=T, invert = T), function(x) length(input[[x]])))
genes_cebpe$cluster_de <- "none"
#for(i in grep(":", names(input), value=T, invert = T)) { genes_cebpe[genes_cebpe$gene %in% input[[i]],]$cluster_de <- i; }
for(i in names(input)) { genes_cebpe[genes_cebpe$gene %in% input[[i]],]$cluster_de <- i; }
genes_cebpe[which(genes_cebpe$cluster_de=="B_gmp_down:B_gmp_up"),]$cluster_de <- "B_gmp_down"
genes_cebpe[which(genes_cebpe$cluster_de=="C_lpm_down:C_lpm_up"),]$cluster_de <- "C_lpm_up"
table(genes_cebpe$cluster_de)
# genes_cebpe <- do.call(rbind, geneSet)

## old

geneSet <- list()

geneSet[[1]] <- genes_cebpe[which((genes_cebpe$cluster==2 | genes_cebpe$cluster==3) & (genes_cebpe$de_lpm=="up" | genes_cebpe$de_mm=="up")),]
geneSet[[1]]$cluster_de <- "A_UP"

geneSet[[2]] <- genes_cebpe[which((genes_cebpe$cluster==2 | genes_cebpe$cluster==3) & (genes_cebpe$de_gmpG=="down" | genes_cebpe$de_epm=="down")),]
geneSet[[2]]$cluster_de <- "A_DOWN"

geneSet[[3]] <- genes_cebpe[which((genes_cebpe$cluster==4 | genes_cebpe$cluster==5) & (genes_cebpe$de_lpm=="up" | genes_cebpe$de_mm=="up")),]
geneSet[[3]]$cluster_de <- "B_UP"

geneSet[[4]] <- genes_cebpe[which((genes_cebpe$cluster==4 | genes_cebpe$cluster==5) & (genes_cebpe$de_gmpG=="down" | genes_cebpe$de_epm=="down")),]
geneSet[[4]]$cluster_de <- "B_DOWN"

# plot(Venn(list(A_up=geneSet[[1]]$gene, A_down=geneSet[[2]]$gene, B_up=geneSet[[3]]$gene, B_down=geneSet[[4]]$gene)))
# plot(Venn(input))

geneSet[[5]] <- genes_cebpe[which((genes_cebpe$cluster==6) & (genes_cebpe$de_mm=="up")),]
geneSet[[5]]$cluster_de <- "C_UP"

geneSet[[6]] <- genes_cebpe[which((genes_cebpe$cluster==10 | genes_cebpe$cluster==11) & (genes_cebpe$de_gmpG=="up" | genes_cebpe$de_epm=="up")),]
geneSet[[6]]$cluster_de <- "D_UP"

geneSet[[7]] <- genes_cebpe[which((genes_cebpe$cluster==10 | genes_cebpe$cluster==11) & (genes_cebpe$de_lpm=="down" | genes_cebpe$de_mm=="down")),]
geneSet[[7]]$cluster_de <- "D_DOWN"

geneSet[[8]] <- genes_cebpe[which((genes_cebpe$cluster==14 | genes_cebpe$cluster==15) & (genes_cebpe$de_lpm=="up" | genes_cebpe$de_mm=="up")),]
geneSet[[8]]$cluster_de <- "E_UP"

geneSet[[9]] <- genes_cebpe[!genes_cebpe$gene %in% do.call(rbind, geneSet)$gene,]
geneSet[[9]]$cluster_de <- "F_rest"
```

## Kim's plots
```{r}
pdf("/home/pundhir/project/chip-seq-analysis/analysis_cebpe/pdf/kim_plots_points.pdf")
df <- as.data.frame(rbind(c(0, 0,	4), 
                    c(5, 15, 8), 
                    c(100,	100, 100),
                    c(NA, 74, 72),
                    c(42, 68, 61),
                    c(28, 59, 52),
                    c(18,	55, 41)))
colnames(df) <- c("rep1", "rep2", "rep3")
df$id <- c("MOCK", "neg.control", "pos.control", "10e-p32", "30e-p33", "100e-p32", "250e-p32")
df$id <- factor(df$id, levels=df$id, ordered = T)

df.melt <- melt(df)
ggbarplot(df.melt, x="id", y="value", add = c("mean_se", "jitter"), fill=c("gray"), point.color="gray")

df <- as.data.frame(rbind(c(0.2070, 0.1210,	1.2500,	1.2800,	2.6300,	1.3300,	3.1500),
            c(0.2240,	0.1610,	1.4200,	1.1600,	2.2800,	1.5200,	2.7600),
            c(0.2230,	0.1620,	1.9500,	1.4500,	3.8100,	1.2600,	4.4100),
            c(0.3920,	0.3570,	0.7460,	0.0186,	3.7900,	0.0000,	0.0016),
            c(0.3420,	0.2980,	0.9480,	0.0211,	1.2900,	0.0002,	0.0002),
            c(0.2360,	0.3230,	0.8410,	0.0183,	1.6500,	0.0000,	0.0001)))
colnames(df) <- c("GP", "EPM", "LPM", "MY", "MM", "BC", "PMN")
df$class <- c("wt", "wt", "wt", "ko", "ko", "ko")
df.melt <- melt(df)
df.melt$class <- factor(df.melt$class, levels=c("wt", "ko"), ordered=T)
df.melt$variable <- factor(df.melt$variable, levels=c("GP", "EPM", "LPM", "MY", "MM", "BC", "PMN"), ordered=T)
ggbarplot(df.melt, x="variable", y="value", add = c("mean_se", "jitter"), palette = c("#d52221", "#2a7ab9"), point.color="gray", color="class", position = position_dodge())

df <- as.data.frame(rbind(c(0.3970,	0.9920,	1.9400,	1.2300),
                          c(0.3960,	0.9400,	2.2000,	1.0700),
                          c(0.2780,	0.9750,	1.7000,	0.8110),
                          c(0.7670,	2.9600,	6.2500,	1.9800),
                          c(0.5060,	2.1400,	8.9300,	2.0300),
                          c(0.6150,	2.5800,	8.1300,	2.0700)))
colnames(df) <- c("MP", "PMO12", "MO1", "MO2")
df$class <- c("wt", "wt", "wt", "ko", "ko", "ko")
df.melt <- melt(df)
df.melt$class <- factor(df.melt$class, levels=c("wt", "ko"), ordered=T)
df.melt$variable <- factor(df.melt$variable, levels=c("MP", "PMO12", "MO1", "MO2"), ordered=T)
ggbarplot(df.melt, x="variable", y="value", add = c("mean_se", "jitter"), palette = c("#d52221", "#2a7ab9"), point.color="gray", color="class", position = position_dodge())

df <- as.data.frame(rbind(c(30.6,	17.5,	44.4,	49.9,	3.77,	2.8,	1.36),
                          c(32.2,	17.6,	38.5,	40.6,	1.82,	0.728,	0.489),
                          c(36.1,	22.5,	50.2,	58.7,	2.76,	1.14,	0.618),
                          c(34.6,	24.7,	38.4,	49.2,	21.1,	0, 0),
                          c(35.4,	25.3,	53.2,	69.7,	23.6,	0, 0),
                          c(38.2,	19.5,	50.3,	69.4,	26.2,	0, 0)))
colnames(df) <- c("GP", "PM", "MY1", "MY2", "MM", "BC", "GR")
df$class <- c("wt", "wt", "wt", "ko", "ko", "ko")
df.melt <- melt(df)
df.melt$class <- factor(df.melt$class, levels=c("wt", "ko"), ordered=T)
df.melt$variable <- factor(df.melt$variable, levels=c("GP", "PM", "MY1", "MY2", "MM", "BC", "GR"), ordered=T)
ggbarplot(df.melt, x="variable", y="value", add = c("mean_se", "jitter"), palette = c("#d52221", "#2a7ab9"), point.color="gray", color="class", position = position_dodge())

df <- as.data.frame(rbind(c(2.50, 12.25, 13.72),
                          c(2.50,	12.50, 18.75),
                          c(2.50,	13.50, 21.60),
                          c(1.00,	8.75,	100.63),
                          c(1.00,	8.00,	112.00),
                          c(1.00,	7.75,	94.94),
                          c(1.00,	10.75, 123.625),
                          c(1.00,	10.25, 107.625),
                          c(1.00,	8.75, 105)
                    ))
colnames(df) <- c("d0", "d2", "d4")
df$class <- c("ewt11", "ewt11", "ewt11", "ertm3", "ertm3", "ertm3", "ebrm56", "ebrm56", "ebrm56")
df.melt <- melt(df)
df.melt$class <- factor(df.melt$class, levels=c("ewt11", "ertm3", "ebrm56"), ordered=T)
df.melt$variable <- factor(df.melt$variable, levels=c("d0", "d2", "d4"), ordered=T)
ggline(df.melt, x="variable", y="value", add = c("mean_se", "jitter"), color="class", palette=c("#1b9e77", "#d95f02", "#7570b3"), point.color="gray")

df <- as.data.frame(rbind(c(37.45, 40.83,	6.81,	12.1),
                          c(40.9,	43.5,	8.4, 3.5),
                          c(41.5,	43.9,	8.1, 3.1),
                          c(47.44, 32.47,	6.03,	12.43),
                          c(45.24, 33.96,	7.22,	11.03),
                          c(48.09, 32.27,	7.14,	10.2),
                          c(34.03, 16.58,	5.11,	42.4),
                          c(34.46, 17.57,	4.57,	41.71),
                          c(33.24, 15.73,	4.4, 45.19)
                    ))
colnames(df) <- c("G1", "S", "G2M", "ltG1")
df$class <- c("d0", "d0", "d0", "d2", "d2", "d2", "d4", "d4", "d4")
df.melt <- melt(df)
df.melt$class <- factor(df.melt$class, levels=c("d0", "d2", "d4"), ordered=T)
df.melt$variable <- factor(df.melt$variable, levels=c("G1", "S", "G2M", "ltG1"), ordered=T)
p1 <- ggbarplot(df.melt, x="variable", y="value", add = c("mean_se", "jitter"), palette=c("#e41a1c", "#377eb8", "#4daf4a"), point.color="gray", color="class", position = position_dodge())

df <- as.data.frame(rbind(c(33.68, 49.09,	8.15,	3.53),
                          c(46.25, 39.30,	7.18,	2.31),
                          c(46.7,	42.1,	8.0,	2.00),
                          c(31.90,	50.45,	9.73,	2.87),
                          c(32.24,	50.65,	9.57,	2.32),
                          c(32.83,	48.51,	10.33,	2.81),
                          c(32.91,	49.11,	8.49,	2.87),
                          c(33.01,	48.74,	9.15,	2.30),
                          c(34.34,	47.62,	9.07,	2.75)
                    ))
colnames(df) <- c("G1", "S", "G2M", "ltG1")
df$class <- c("d0", "d0", "d0", "d2", "d2", "d2", "d4", "d4", "d4")
df.melt <- melt(df)
df.melt$class <- factor(df.melt$class, levels=c("d0", "d2", "d4"), ordered=T)
df.melt$variable <- factor(df.melt$variable, levels=c("G1", "S", "G2M", "ltG1"), ordered=T)
p2 <- ggbarplot(df.melt, x="variable", y="value", add = c("mean_se", "jitter"), palette=c("#e41a1c", "#377eb8", "#4daf4a"), point.color="gray", color="class", position = position_dodge())

df <- as.data.frame(rbind(c(38.94, 43.88,	9.33,	3.20),
                          c(47.5,	42.1,	6.2,	1.1),
                          c(47.6,	41.4,	7.9,	1.1),
                          c(34.44,	42.82,	12.50, 4.76),
                          c(33.76,	45.59,	10.75,	5.06),
                          c(32.90,	46.74,	10.47,	4.21),
                          c(31.76,	46.07,	11.48,	5.51),
                          c(30.98,	47.84,	10.31,	5.23),
                          c(32.21,	44.87,	11.02,	6.44)
                    ))
colnames(df) <- c("G1", "S", "G2M", "ltG1")
df$class <- c("d0", "d0", "d0", "d2", "d2", "d2", "d4", "d4", "d4")
df.melt <- melt(df)
df.melt$class <- factor(df.melt$class, levels=c("d0", "d2", "d4"), ordered=T)
df.melt$variable <- factor(df.melt$variable, levels=c("G1", "S", "G2M", "ltG1"), ordered=T)
p3 <- ggbarplot(df.melt, x="variable", y="value", add = c("mean_se", "jitter"), palette=c("#e41a1c", "#377eb8", "#4daf4a"), point.color="gray", color="class", position = position_dodge())
ggarrange(p1, p2, p3, nrow=3, ncol=1)
dev.off()
```

